<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en" data-bs-theme="auto">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<title>FAIR in (biological) practice: All in One View</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<script src="../assets/themetoggle.js"></script><link rel="stylesheet" type="text/css" href="../assets/styles.css">
<script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../favicons/incubator/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicons/incubator/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="../favicons/incubator/favicon-16x16.png">
<link rel="manifest" href="../favicons/incubator/site.webmanifest">
<link rel="mask-icon" href="../favicons/incubator/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">
<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">
</head>
<body>
    <header id="top" class="navbar navbar-expand-md top-nav incubator"><svg xmlns="http://www.w3.org/2000/svg" class="d-none"><symbol id="check2" viewbox="0 0 16 16"><path d="M13.854 3.646a.5.5 0 0 1 0 .708l-7 7a.5.5 0 0 1-.708 0l-3.5-3.5a.5.5 0 1 1 .708-.708L6.5 10.293l6.646-6.647a.5.5 0 0 1 .708 0z"></path></symbol><symbol id="circle-half" viewbox="0 0 16 16"><path d="M8 15A7 7 0 1 0 8 1v14zm0 1A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"></path></symbol><symbol id="moon-stars-fill" viewbox="0 0 16 16"><path d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"></path><path d="M10.794 3.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387a1.734 1.734 0 0 0-1.097 1.097l-.387 1.162a.217.217 0 0 1-.412 0l-.387-1.162A1.734 1.734 0 0 0 9.31 6.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387a1.734 1.734 0 0 0 1.097-1.097l.387-1.162zM13.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732l-.774-.258a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L13.863.1z"></path></symbol><symbol id="sun-fill" viewbox="0 0 16 16"><path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"></path></symbol></svg><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-8">
      <div class="large-logo">
        <img id="incubator-logo" alt="Lesson Description" src="../assets/images/incubator-logo.svg"><span class="badge text-bg-info">
          <abbr title="This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.">
            <a href="https://cdh.carpentries.org/the-lesson-life-cycle.html#polishing-beta-stage" class="external-link alert-link">
              <i aria-hidden="true" class="icon" data-feather="alert-circle" style="border-radius: 5px"></i>
              Beta
            </a>
            <span class="visually-hidden">This lesson is in the beta phase, which means that it is ready for teaching by instructors outside of the original author team.</span>
          </abbr>
        </span>

      </div>
    </div>
    <div class="selector-container">
      <div id="theme-selector">
        <li class="nav-item dropdown" id="theme-button-list">
          <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
            <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
<li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                Light
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                Dark
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
            <li>
              <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                Auto
                <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
            </li>
          </ul>
</li>
      </div>

      <div class="dropdown" id="instructor-dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
<li><button class="dropdown-item" type="button" onclick="window.location.href='../aio.html';">Learner View</button></li>
        </ul>
</div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl bottom-nav incubator" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Lesson Description" src="../assets/images/incubator-logo-sm.svg">
</div>
    <div class="lesson-title-md">
      FAIR in (biological) practice
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="Search the All In One page" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
<li class="nav-item">
          <span class="lesson-title">
            FAIR in (biological) practice
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown">
<li><a class="dropdown-item" href="code-good-practices.html"></a></li>
<li><a class="dropdown-item" href="feedback_fair-bio-practice_workshop_1.html"></a></li>
<li><a class="dropdown-item" href="learners.html">Learners Profiles</a></li>
<hr>
<li><a class="dropdown-item" href="discuss.html">Discussion</a></li>
<li><a class="dropdown-item" href="reference.html">Glossary</a></li>
          </ul>
</li>
      </ul>
</div>
    <!--
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled>
      <input class="form-control me-2 searchbox" type="search" placeholder="" aria-label="">
        <button class="btn btn-outline-success tablet-search-button"  type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="Search the All In One page"></i>
        </button>
      </fieldset>
    </form>
    -->
    <a id="search-button" class="btn btn-primary" href="../instructor/aio.html" role="button" aria-label="Search the All In One page">Search the All In One page</a>
  </div>
<!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  FAIR in (biological) practice
</div>

<aside class="col-md-12 lesson-progress"><div style="width: %" class="percentage">
    %
  </div>
  <div class="progress incubator">
    <div class="progress-bar incubator" role="progressbar" style="width: %" aria-valuenow="" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row" id="theme-row-mobile">
            <div class="col" id="theme-selector">
              <li class="nav-item dropdown" id="theme-button-list">
                <button class="btn btn-link nav-link px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (auto)">
                  <svg class="bi my-1 theme-icon-active"><use href="#circle-half"></use></svg><span class="d-lg-none ms-1" id="bd-theme-text">Toggle Theme</span>
                </button>
                <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="bd-theme-text">
<li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#sun-fill"></use></svg>
                      Light
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center" data-bs-theme-value="dark" aria-pressed="false">
                      <svg class="bi me-2 theme-icon"><use href="#moon-stars-fill"></use></svg>
                      Dark
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                  <li>
                    <button type="button" class="btn dropdown-item d-flex align-items-center active" data-bs-theme-value="auto" aria-pressed="true">
                      <svg class="bi me-2 theme-icon"><use href="#circle-half"></use></svg>
                      Auto
                      <svg class="bi ms-auto d-none"><use href="#check2"></use></svg></button>
                  </li>
                </ul>
</li>
            </div>
          </div>
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../aio.html">Learner View</a>
                      </div>
                    </div>
                  </div>
<!--/div.accordion-item-->
                </div>
<!--/div.accordion-flush-->
              </div>
<!--div.sidenav-view-selector -->
            </div>
<!--/div.col -->

            <hr>
</div>
<!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-welcome.html">1. Welcome</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush3">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading3">
        <a href="02-os-introduction.html">2. Introduction to Open Science</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-being-fair.html">3. Being FAIR</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04-IP-licensing.html">4. Intellectual Property, Licensing and Openness</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05-intro-to-metadata.html">5. Introduction to metadata</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush7">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading7">
        <a href="06-being-precise.html">6. Being precise</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush8">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading8">
        <a href="07-data-in-excel.html">7. (Meta)data in Excel</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush9">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading9">
        <a href="08-laboratory-records.html">8. Laboratory records</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush10">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading10">
        <a href="09-files-organization.html">9. Working with files</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush11">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading11">
        <a href="10-reusable-analysis.html">10. Reusable analysis</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush12">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading12">
        <a href="11-version-control.html">11. Version control</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush13">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading13">
        <a href="12-templates.html">12. Templates for consistency</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush14">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading14">
        <a href="13-data-repositories.html">13. Public repositories</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush15">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading15">
        <a href="14-data_management_planning.html">14. It's all about planning</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush16">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading16">
        <a href="15-putting-all-together.html">15. Putting it all together</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush17">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading17">
        <a href="episode_template.html">16. Template</a>
    </div>
<!--/div.accordion-header-->

  </div>
<!--/div.accordion-item-->
</div>
<!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width">
<div class="accordion accordion-flush lesson-resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="lesson-resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul>
<li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <li><a href="code-good-practices.html"></a></li>
<li><a href="feedback_fair-bio-practice_workshop_1.html"></a></li>
<li><a href="learners.html">Learners Profiles</a></li>
<hr>
<li><a class="dropdown-item" href="discuss.html">Discussion</a></li>
<li><a class="dropdown-item" href="reference.html">Glossary</a></li>
                    </ul>
</div>
                </div>
              </div>
            </div>
            <hr class="half-width lesson-resources">
<a href="../instructor/aio.html">See all in one page</a>


            <hr class="d-none d-sm-block d-md-none">
<div class="d-grid gap-1">

            </div>
          </div>
<!-- /div.accordion -->
        </div>
<!-- /div.sidebar-inner -->
      </nav>
</div>
<!-- /div.sidebar -->
  </div>
<!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-extra.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <main id="main-content" class="main-content"><div class="container lesson-content">
        
        
<section id="aio-01-welcome"><p>Content from <a href="01-welcome.html">Welcome</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/01-welcome.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 15 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Who are we and what are we going to learn?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Introduce ourselves and the course</li>
<li>Setup our learning platform</li>
<li>Better research by better sharing</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="introductions">Introductions<a class="anchor" aria-label="anchor" href="#introductions"></a>
</h2>
<hr class="half-width">
<div id="testimonial1" class="callout testimonial">
<div class="callout-square">
<i class="callout-icon" data-feather="heart"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Testimonial</h3>
<div class="callout-content">
<p>Introductions set the stage for learning.</p>
<p>-– Tracy Teal, Former Executive Director, The Carpentries</p>
</div>
</div>
</div>
<p>Hello everyone, and welcome to the <strong>FAIR in biological
practice</strong> workshop. We are very pleased to have you with us.</p>
<div id="todays-trainers" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="todays-trainers" class="callout-inner">
<h3 class="callout-title">Today’s Trainers</h3>
<div class="callout-content">
<p>To begin the class, each Trainer should give a brief introduction of
themselves.</p>
</div>
</div>
</div>
<p>Now we would like to get to know all of you.</p>
<div id="who-are-you-and-what-are-your-expectations-from-the-workshop" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="who-are-you-and-what-are-your-expectations-from-the-workshop" class="callout-inner">
<h3 class="callout-title">Who are you and what are your expectations from the workshop</h3>
<div class="callout-content">
<p>Please introduce yourself shortly and tell us:</p>
<ul>
<li>Why are you taking this course?</li>
<li>What goals do you have for the follwing days?</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="better-research-by-better-sharing">Better research by better sharing<a class="anchor" aria-label="anchor" href="#better-research-by-better-sharing"></a>
</h2>
<hr class="half-width">
<p>For many of us, data management or output sharing in general are
considered a burden rather than a useful activity. Part of the problem
is our bad timing and lack of planning.</p>
<p><strong>Data management is a continuous process</strong></p>
<p><img src="../fig/01-sharing_as_part_of_workflow.jpg" alt="Figure 5.2. Sharing as part of the workflow" class="figure"><em>Figure credits:
Tomasz Zielinski and Andrés Romanowski</em></p>
<p>When should you engage in data sharing and open practices?</p>
<ul>
<li>Data management should be done throughout the duration of your
project.</li>
<li>If you wait till the end, it will take a massive effort on your side
and will be more of a burden than a benefit.</li>
<li>The overheads of making data suitable for sharing at each stage of
data life-cycle are small comparing to the other actions involved</li>
<li>Taking the time to do effective data management will help you
understand your data better and make it easier to find when you need it
(for example when you need to write a manuscript or a thesis!).</li>
<li>All the practices that enable others to access and use your outcomes
directly benefit you and your group</li>
</ul>
<p>In this workshop we will show you how you can plan and do your
research in a way that makes your outputs readily available for re-use
by others.</p>
<p>Our agenda:</p>
<ul>
<li><p><strong>Day 1</strong> We will start with explaining Open Science
principles and what the benefits are of being open for you and society.
Then we will talk about FAIR principles which define steps you/we should
take so that our “shared” outputs are of value. We will finish with a
brief introduction to metadata and ontologies, and how to be precise to
deliver FAIR outputs.</p></li>
<li><p><strong>Day 2</strong> We will show you how to use metadata in
Excel, and the benefits of using online records for documenting
experiments. We will talk about working and organizing files and
appropriate file naming conventions using Excel or CSV tables to store
and document data.</p></li>
<li><p><strong>Day 3</strong> We will introduce Jupyter notebooks as a
tool for reusable data analysis. We will talk about Version Control. We
will teach how to describe your projects using simple text files or
customized templates.</p></li>
<li><p><strong>Day 4</strong> We will show you how public repositories
make your outputs accessible and reusable. We will consolidate our
knowledge of FAIR ready data management and what other tools can help
you during your research.</p></li>
</ul></section><section><h2 class="section-heading" id="online-workshop-specifics">Online workshop specifics<a class="anchor" aria-label="anchor" href="#online-workshop-specifics"></a>
</h2>
<hr class="half-width">
<div id="our-learning-tools" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="our-learning-tools" class="callout-inner">
<h3 class="callout-title">Our learning tools</h3>
<div class="callout-content">
<p>Before we begin let’s explain how to use the tools:</p>
<ul>
<li>Raising hands</li>
<li>Yes/No sticker</li>
<li>Chatroom for links not for jokes</li>
<li>Breakout rooms, leaving and rejoining</li>
<li>using pad, answering questions in pad</li>
<li>where to find things If needed, check the pre workshop setup, ask to
report problems and help at a break or after the session.</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Do not be shy</li>
<li>Be nice</li>
<li>Remember, you can do better research if you plan to share your
outputs!</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-02-os-introduction"><p>Content from <a href="02-os-introduction.html">Introduction to Open Science</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/02-os-introduction.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 55 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is Open Science?</li>
<li>How can I benefit from Open Science?</li>
<li>Why has Open Science become a hot topic?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Identify parts of the Open Science movement, their goals and
motivations</li>
<li>Explain the main benefits of Open Science</li>
<li>Recognize the barriers and risks in the adoption of Open Science
practices</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(16 min teaching)</p>
<p>Science works best by exchanging ideas and building on them. Most
efficient science involves both questions and experiments being made as
fully informed as possible, which requires the free exchange of data and
information.</p>
<p>All practices that make knowledge and data <strong>freely
available</strong> fall under the umbrella-term of <strong>Open
Science/Open Research</strong>. It makes science <strong>more
reproducible, transparent, and accessible</strong>. As science becomes
more open, the way we conduct and communicate science changes
continuously.</p>
<div id="what-is-open-science" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="what-is-open-science" class="callout-inner">
<h3 class="callout-title">What is Open Science</h3>
<div class="callout-content">
<p>Open science is the movement to make scientific research (including
publications, data, physical samples, and software) and its
dissemination <strong>accessible to all levels</strong> of an inquiring
society, amateur or professional.</p>
<p>Open Science represents a new approach to the scientific process
based on <strong>cooperative work</strong> and new ways of diffusing
knowledge by <strong>using digital technologies</strong> and new
collaborative tools</p>
<p>Open science is <strong>transparent and accessible knowledge</strong>
that is shared and developed through collaborative networks.</p>
<p>Characteristics:</p>
<ul>
<li>Using web-based tools to facilitate information exchange and
scientific collaboration</li>
<li>Transparency in experimental methodology, observation, and
collection of data</li>
<li>Public availability and reusability of scientific data, methods and
communications</li>
</ul>
</div>
</div>
</div>
<div class="section level3">
<h3 id="what-is-the-open-science-movement">What is the Open Science movement?<a class="anchor" aria-label="anchor" href="#what-is-the-open-science-movement"></a>
</h3>
<p>Sharing of information is fundamental for science. This began at a
significant scale with the invention of scientific journals in 1665. At
that time this was the best available alternative to critique &amp;
disseminate research, and foster communities of like-minded
researchers.</p>
<p>Whilst this was a great step forward, the journal-driven system of
science has led to a culture of ‘closed’ science, where knowledge or
data is unavailable or unaffordable to many.</p>
<p>The distribution of knowledge has always been subject to improvement.
Whilst the internet was initially developed for military purposes, it
was hijacked for communication between scientists, which provided a
viable route to change the dissemination of science.</p>
<p>The momentum has built up with a change in the way science is
communicated to reflect what research communities are calling for –
solutions to the majority of problems (e.g. impact factors, data
reusability, reproducibility crisis, trust in the public science sector
etc…) that we face today.</p>
<p>Open Science is the movement to increase transparency and
reproducibility of research, through using the open best practices.</p>
<figure><img src="../fig/OpenScienceBuildingBlocks.jpg" alt="Figure 1. Open Science Building Blocks" class="figure mx-auto d-block"></figure><p><em>After <a href="https://www.fosteropenscience.eu/content/what-open-science-introduction" class="external-link">Gema
Bueno de la Fuente</a></em></p>
</div>
<div class="section level3">
<h3 id="open-science-building-blocks">Open Science Building Blocks<a class="anchor" aria-label="anchor" href="#open-science-building-blocks"></a>
</h3>
<ul>
<li><p><strong>Open Access:</strong> Research outputs hosted in a way
that make them accessible for everyone. Traditionally Open Access
referred to journal articles, but now includes books, chapters or
images.</p></li>
<li><p><strong>Open Data:</strong> Data freely and readily available to
access, reuse, and share. Smaller data sets were often accessible as
supplemental materials by journals alongside articles themselves.
However, they should be hosted in dedicated platforms for more
convenient and better access.</p></li>
<li><p><strong>Open Software:</strong> Software where the source code is
made readily available; others are free to use, change, and share. Some
examples of these including the coding language and supporting software
R and RStudio, as well as image analysis software such as
Fiji/ImageJ.</p></li>
<li><p><strong>Open Notebooks:</strong> Lab &amp; notebooks hosted
online, readily accessible to all. These are popular among some of the
large funding bodies and allow anyone to comment on any stage of the
experimental record.</p></li>
<li><p><strong>Open Peer Review:</strong> A system where peer review
reports are published alongside the body of work. This can include
reviewers’ reports, correspondence between parties involved, rebuttals,
editorial decisions etc…</p></li>
<li><p><strong>Citizens Science:</strong> Lay people become involved in
scientific research, most commonly in data collection or image analysis.
Platforms such as <a href="https://www.zooniverse.org/" class="external-link">zooniverse.org</a> help connect
projects with lay people interested in playing an active role in
research, which can help generate and/or process data which would
otherwise be unachievable by one single person.</p></li>
<li><p><strong>Scientific social networks:</strong> Networks of
researchers, which often meet locally in teams, but are also connected
online, foster open discussions on scientific issues. Online, many
people commonly use traditional social media platforms for this, such as
Twitter, Instagram, various sub-reddits, discussion channels on
Slack/Discord etc…, although there are also more dedicated spaces such
as <a href="https://www.researchgate.net/" class="external-link">researchgate.net</a>.</p></li>
<li><p><strong>Open Education resources:</strong> Educational materials
that are free for anyone to access and use to learn from. These can be
anything from talks, instructional videos, and explanations posted on
video hosting websites (e.g. YouTube), to entire digital textbooks
written and then published freely online.</p></li>
<li><p><strong>Citizen science:</strong> Citizen participation of
various stages of research process from project funding to collecting
and analysing data.</p></li>
</ul>
<div id="exercise-1-benefits-of-open-science-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-benefits-of-open-science-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Benefits of Open Science (5 min)</h3>
<div class="callout-content">
<p>Being open has other outcomes/consequences beyond giving the free
access to information. For example, Open educational resources:</p>
<ul>
<li>enables collaborative development of courses</li>
<li>improves teachers/instructors skills by sharing ideas</li>
</ul>
<p>Select one or two of the following OS parts:</p>
<ul>
<li>Open Access</li>
<li>Open Data</li>
<li>Open Software</li>
<li>Open Notebooks</li>
<li>Open Peer Review</li>
</ul>
<p>and discuss what are the benefits or what problems are solved by
adaption of those Open initiatives.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>Possible benefits and consequences for each part:</p>
<p><strong>Open Access</strong></p>
<ul>
<li>speed of knowledge distribution</li>
<li>leveling field for underfunded sites which otherwise wouldn’t be
able to navigate the paywall</li>
<li>prevent articles being paid for ‘thrice’ (first to produce, second
to publish, third to access) by institutions.</li>
<li>greater access to work by others, increasing chance for exposure
&amp; citations</li>
<li>access to work by lay audiences, thus increases social exposure of
research</li>
</ul>
<p><strong>Open Data</strong></p>
<ul>
<li>ensures data isn’t lost overtime - reusability</li>
<li>acceleration of scientific discovery rate</li>
<li>value for money/reduced redundancy</li>
<li>permits statistical re-analysis of the data to validate
findings</li>
<li>gives access to datasets which were not published as papers
(e.g. negative results, large screening data sets)</li>
<li>provides an avenue to generate new hypotheses</li>
<li>permits combination of multiple data sources to address questions,
provides greater power than a single data source</li>
</ul>
<p><strong>Open Software</strong></p>
<ul>
<li>great source to learn programming skills</li>
<li>the ability to modify creates a supportive community of users and
rapid innovation</li>
<li>saves time</li>
<li>faster bug fixes</li>
<li>better error scrutiny</li>
<li>use of the same software/code allows better reproducibility between
experiments</li>
<li>need funds to maintain and update software</li>
</ul>
<p><strong>Open Notebooks</strong></p>
<ul>
<li>100% transparent science, allowing input from others at early stages
of experiments</li>
<li>source of learning about the process of how science is actually
conducted</li>
<li>allows access to experiments and data which otherwise never get
published</li>
<li>provides access to ‘negative’ results and failed experiments</li>
<li>anyone, anywhere around the world, at any time, can check in on
projects, including many users simultaneously</li>
<li>possibility of immediate feedback</li>
<li>thorough evidence of originality of ideas and experiments, negating
effect of ‘scooping’</li>
</ul>
<p><strong>Open Peer Review</strong></p>
<ul>
<li>visibility leads to more constructive reviews</li>
<li>mitigates against editorial conflicts of interest and/or biases</li>
<li>mitigates against reviewers conflicts of interest and/or biases</li>
<li>allows readers to learn/benefit from comments of the reviewers</li>
</ul>
<p><strong>Open Educational Materials</strong></p>
<ul>
<li>Foster collaboration between educators/others</li>
<li>Show clearly how method was taught (e.g. Carpentries materials)
which can be reproduces anywhere, anytime</li>
<li>protects materials from becoming technologically obsolete</li>
<li>authors preparing the material or contribute all earn credit
(e.g. GitHub)</li>
<li>recycle animations and material that is excellent (why reinvent the
wheel?)</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="motivation-money-8-min-teaching">Motivation: Money (8 min teaching)<a class="anchor" aria-label="anchor" href="#motivation-money-8-min-teaching"></a>
</h3>
<p>One has to consider the moral objectives that accompany the
research/publication process: charities/taxpayers pay to fund research,
these then pay again to access the research they already funded.</p>
<p>From an economic point of view, scientific outputs generated by
public research are a public good that everyone should be able to use at
no cost.</p>
<p>According to EU report <a href="https://op.europa.eu/en/publication-detail/-/publication/d375368c-1a0a-11e9-8d04-01aa75ed71a1" class="external-link">“Cost-benefit
analysis for FAIR research data”</a>, €10.2bn is lost every year because
of not accessible data (plus additional 16bn if accounting for re-use
and research quality).</p>
<p>The goals of Open Science is to make research and research data
available to e.g. charities/taxpayers who funded this research.</p>
<p><strong>COAlition S</strong>, a group of national research funding
organisations backed by the European Commission and the European
Research Council, is a big driver trying to get rid of the paywalls that
our research is sat behind. They announced Plan S, an initiative to make
research publications fully free at the point of access, meaning that
all research funded by public funding bodies must be published Open
Access from 2021 onwards.</p>
<div id="open-access-a-successful-example" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="open-access-a-successful-example" class="callout-inner">
<h3 class="callout-title">Open Access (a successful example)</h3>
<div class="callout-content">
<p>The majority of larger UK and other countries’ funding bodies are now
making Open Access publication conditional upon funding.</p>
<p>The initiative is known as Plan S, which requires <em>“resulting
publications available immediately (without embargoes) and under open
licences, either in quality Open Access platforms or journals or through
immediate deposit in open repositories that fulfil the necessary
conditions.”</em></p>
<p>Exact requirements differ between funding bodies, with the minimum
requirement being that a copy be deposited with your home
institution.</p>
<p>Details of funding bodies and their involvement and requirements can
be found at <a href="https://www.coalition-s.org/plan-s-funders-implementation/" class="external-link">Plan
S/cOAlition S</a>. There is also a <a href="https://www.coalition-s.org/blog/unboxing-the-journal-checker-tool/" class="external-link">cOAlition
S journal checker tool</a> to assess compliance being developed. The <a href="https://doaj.org/" class="external-link">Directory of Open Access Journals (DOAJ)</a> is
a tool to find which journals are Open Access.</p>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="motivation-reproducibility">Motivation: Reproducibility<a class="anchor" aria-label="anchor" href="#motivation-reproducibility"></a>
</h3>
<p>The inherited transparency of Open Science and the easy access to
data, methods and analysis details naturally help to address the
Reproducibility crisis. The openness of scientific communications and of
the actual process of evaluation of the research (Open Peer Review)
increase confidence in the research findings.</p>
</div>
<div class="section level3">
<h3 id="personal-motivators">Personal motivators<a class="anchor" aria-label="anchor" href="#personal-motivators"></a>
</h3>
<p>Open Science is advantageous to many parties involved in science
(including researcher community, funding bodies, the public even
journals), which is leading to a push for the widespread adoption of
Open Science practices.</p>
<p>Large UK funding bodies such as The Wellcome Trust are big supporters
of Open Science. We can see with the example of Open Access, that once
enforced by funders (<em>the stick</em>) there is a wide adoption. But
what about the personal motivators, <em>the carrots</em>.</p>
<div id="exercise-2-personal-benefits-of-being-open-4-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-2-personal-benefits-of-being-open-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Personal benefits of being “open” (4 min)</h3>
<div class="callout-content">
<p>Below are some personal benefits to adopting Open Science practices.
Read through them which of them are the strongest motivators for you.
Select two the most important/attractive for you and mark them with +1,
select the two least important for you and mark them with 0</p>
<ul>
<li>receive higher citations</li>
<li>complying with funders’ policies</li>
<li>get extra value from your work (e.g. collaborators, reuse by
modellers, ML specialists)</li>
<li>demonstrate research impact</li>
<li>save own time (reproducibility but also communication overhead)</li>
<li>become pioneers</li>
<li>distinguish yourself from the crowd</li>
<li>plan successful research proposals</li>
<li>gain valuable experience</li>
<li>form community</li>
<li>increased speed and/or ease of writing papers</li>
<li>speed up and help with peer review</li>
<li>build reputation and presence in the science community</li>
<li>evidence of your scientific rigour and work ethic</li>
<li>avoid embarrassment/disaster when you cannot reproduce your
results</li>
</ul>
<p>Can you think of other benefits? How personal benefits of Open
Science compare to the benefits for the (scientific) society?</p>
</div>
</div>
</div>
<p>(3 min teaching)</p>
<p>The main difference between the public benefits of Open Science
practices and the personal motivators of outputs creators, that the
public can benefit almost instantly from the open resources. However,
the advantages for data creator comes with a delay, typically counted in
years. For example, building reputation will not happen with one
dataset, the re-use also will lead to citations/collaboration after the
next research cycle.</p>
</div>
<div class="section level3">
<h3 id="barriers-and-risks-of-os-movement">Barriers and risks of OS movement:<a class="anchor" aria-label="anchor" href="#barriers-and-risks-of-os-movement"></a>
</h3>
<div id="exercise-3-why-we-are-not-doing-open-science-already-4-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-3-why-we-are-not-doing-open-science-already-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 3: Why we are not doing Open Science already (4 min)</h3>
<div class="callout-content">
<p>Discuss Open Science barriers, mention the reasons for not already
being open:</p>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<ul>
<li>sensitive data (anonymising data from administrative health records
can be difficult)</li>
<li>IP</li>
<li>misuse (fake news)</li>
<li>lack of confidence (the fear of critics)</li>
<li>lack of expertise</li>
<li>the costs in $ and in time</li>
<li>novelty of data</li>
<li>it is not mandatory</li>
<li>lack of credit (publishing negative results is of little benefit to
you)</li>
</ul>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<p>(9 min teaching)</p>
<p>It may seem obvious that we should adopt open science practices, but
there are associated challenges with doing so.</p>
<p>Sensitivity of data is sometimes considered a barrier. Shared data
needs to be compliant with data privacy laws, leading many to shy away
from hosting it publicly. Anonymising data to desensitise it can help
overcome this barrier.</p>
<p>The potential for intellectual property on research can dissuade some
from adopting open practices. Again, much can be shared if the data is
filtered carefully to protect anything relating to intellectual
property.</p>
<p>Another risk could be seen with work on Covid19: pre-prints. A
manuscript hosted publicly prior to peer review, may accelerate access
to knowledge, but can also be misused and/or misunderstood. This can
result in political and health decision making based on faulty data,
which is counter to societies’ best interest.</p>
<p>One concern is that opening up ones data to the scientific community
can lead to the identification of errors, which may lead to feelings of
embarrassment. However, this could be considered an upside - we should
seek for our work to be scrutinized and errors to be pointed out, and is
the sign of a competent scientist. One should rather have errors pointed
out rather than risking that irreproducible data might cause even more
embarrassment and disaster.</p>
<p>One of <strong>the biggest barriers are the costs</strong> involved
in “being Open”. Firstly, making outputs readily available and usable to
others takes time and significant effort. Secondly, there are costs of
hosting and storage. For example, microscopy datasets reach sizes in
terabytes, making such data accessible for 10 years involves serious
financial commitment.</p>
</div>
<div class="section level3">
<h3 id="get-involved">Get involved<a class="anchor" aria-label="anchor" href="#get-involved"></a>
</h3>
<p>Thankfully, incentive structures are beginning to support Open
Science practices:</p>
<ul>
<li>Universities signing up to the Declaration on Research Assessment
(DORA).</li>
<li>Wellcome Trust funding proposals that increase Open Science</li>
<li>Wellcome Trust asked for description of Open Science activities in
the grant application</li>
</ul>
<p>You do not want to be left behind!</p>
</div>
<section><h2 class="section-heading" id="where-to-next">Where to next<a class="anchor" aria-label="anchor" href="#where-to-next"></a>
</h2>
<hr class="half-width">
<blockquote>
<p>Further reading/links:</p>
<ul>
<li><a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000246" class="external-link">Challenges
&amp; benefits of OS</a></li>
<li><a href="https://www.cos.io/" class="external-link">Centre for Open Science</a></li>
<li><a href="https://www.youtube.com/watch?v=c-bemNZ-IqA" class="external-link">Ted talk
supporting OS</a></li>
</ul>
</blockquote>
<div id="exercise-4-open-science-quiz-5-min-runs-over-break" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-open-science-quiz-5-min-runs-over-break" class="callout-inner">
<h3 class="callout-title">Exercise 4: Open Science Quiz (5 min + runs over break)</h3>
<div class="callout-content">
<p>Which of the following statements about the OS movement are
true/false?</p>
<ul>
<li>Open Science relies strongly on the Internet</li>
<li>Open Access eliminates publishing costs</li>
<li>Open Data facilitates re-use</li>
<li>Open Data can increases confidence in research findings</li>
<li>In Open Peer Review, readers vote on publication acceptance</li>
<li>Open Notebooks improve reproducibility</li>
<li>Open Notebooks can create patenting issues</li>
<li>Open Access permits the whole society to benefit from scientific
findings</li>
<li>Citizen Science engages the public in the research process</li>
<li>Citizen Science can help get ecological data quickly</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<ul>
<li>Open Science relies strongly on the Internet T</li>
<li>Open Access eliminates publishing costs F</li>
<li>Open Data facilitates re-use T</li>
<li>Open Data increases confidence in research findings T</li>
<li>In Open Peer Review, readers vote on publication acceptance F</li>
<li>Open Notebooks improve reproducibility T</li>
<li>Open Notebooks can create patenting issues T*</li>
<li>Open Access permits the whole society to benefit from scientific
findings T</li>
<li>Citizen Science engages the public in the research process T</li>
<li>Citizen Science can help get ecological data quickly T</li>
</ul>
</div>
</div>
</div>
</div>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was adapted from:</p>
<ul>
<li>Wiki <a href="https://en.wikipedia.org/wiki/Open_science" class="external-link">Open
Science</a>
</li>
<li><a href="https://www.eosc-hub.eu/open-science-info" class="external-link">European Open
Science Cloud</a></li>
<li>
<a href="https://portlandpress.com/biochemist/article/42/3/58/225220/Science-is-necessarily-collaborative" class="external-link">Science
is necessarily collaborative - The Biochemist article</a>.</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Open Science increases transparency in research</li>
<li>Publicly funded science should be publicly available</li>
<li>While both You and the research community benefit from open
practices, they are costs involved in making outputs open</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-03-being-fair"><p>Content from <a href="03-being-fair.html">Being FAIR</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/03-being-fair.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to get more value from your own data?</li>
<li>What are the FAIR guidelines?</li>
<li>Why being FAIR matters?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Recognize typical issues that prevent data re-use</li>
<li>Understand FAIR principles</li>
<li>Know steps for achieving FAIR data</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(7 min teaching)</p>
<p>We have seen how Open practices can benefit both scientific community
as a whole and individual practitioner. The wide adoption of Open Access
principles has resulted in an easy access to recent biomedical
publications. Unfortunately, the same cannot be said about data and
software that accompanies those publications.</p>
<div id="what-is-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="what-is-data" class="callout-inner">
<h3 class="callout-title">What is data</h3>
<div class="callout-content">
<p>Although scientific data is a very broad term, we still encounter
groups who (wrongly) believe they do not have data! Data does not mean
Excel files with recorded measurements from a machine. Data also
includes:</p>
<ul>
<li>images, not only from microscopes</li>
<li>information about biological materials, like strain or patient
details</li>
<li>biological models</li>
<li>recipes, laboratory and measurement protocols</li>
<li>scripts, analysis procedures, and custom software can also be
considered data However, there are specific recommendations on how to
deal with code.</li>
</ul>
</div>
</div>
</div>
<p>Let’s have a look how challenging it can be to access and use data
from published biological papers.</p>
<div id="exercise-1-impossible-protocol-4-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-impossible-protocol-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Impossible protocol (4 min)</h3>
<div class="callout-content">
<p>You need to do a western blot to identify Titin proteins, the largest
proteins in the body, with a molecular weight of 3,800 kDa. You found an
antibody sold by Sigma Aldrich that has been validated in western blots
and immunofluorescence. Sigma Aldrich lists the <a href="https://doi.org/10.1002/acn3.50831" class="external-link">Yu et al., 2019</a> paper as
reference.</p>
<p>Find details of how to separate and transfer this large protein in
the reference paper.</p>
<ul>
<li>Hint 1: Methods section has a Western blot analysis subsection.</li>
<li>Hint 2: Follow the references.</li>
</ul>
<p>Would you say that the methods was Findable? Accessible?
Reusable?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<ul>
<li>Ref 17 will lead you to <a href="https://doi.org/10.1002/ana.24102" class="external-link">this paper</a>, which first of
all is not Open Access</li>
<li>Access the paper through your institutions (if you can) and find the
‘Western Blotting’ protocol on page 232 which will show the following
(Screenshot from the methods section from <a href="https://doi.org/10.1002/ana.24102" class="external-link">Evilä et al 2014</a>):</li>
<li><img src="../fig/impossible_protocol.png" alt="Figure 1. Impossible Protocol" class="figure"></li>
<li>“Western blotting were performed according to standard methods.” -
with no further reference to these standard methods, describing these
methods, or supplementary material detailing these methods</li>
<li>This methodology is unfortunately a true dead end and we thus can’t
easily continue our experiments!</li>
</ul>
</div>
</div>
</div>
</div>
<div id="impossible-numbers" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="impossible-numbers" class="callout-inner">
<h3 class="callout-title">Impossible numbers</h3>
<div class="callout-content">
<p><a href="https://doi.org/10.1093/jxb/err244" class="external-link">Ikram 2014</a> paper
contains data about various metabolites in different accessions
(genotypes) of Arabidopsis plant. You would like to calculate average
nitrogen content in plants grown under normal and nitrogen limited
conditions. Please calculate the average (over genotypes) nitrogen
content for the two experimental conditions.</p>
<ul>
<li>Hint 1. Data are in Supplementary data</li>
<li>Hint 2. Search for nitrogen in paper text to identify the correct
data column.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ul>
<li>Finding the right table and column containing the relevant data is
already problematic as the headers are obscured so they need to decoded
using manuscript</li>
<li>Data in pdf table so they cannot be readily used in
calculations</li>
<li>Depending on the software used to open (and the way the pdf was
created), the local machine international settings, copying the data
into Excel can bring unexpected results <img src="../fig/03-average_to_excel1.png" alt="Figure 2. Pdf data copied to Excel" class="figure"><br><em>Data needs parsing after coping to Excel</em> <img src="../fig/03-average_to_excel1.png" alt="Figure 2. The same data copied to Excel with polish locale" class="figure"><br><em>The same data copied to Excel with polish locale has been converted
to dates</em>
</li>
<li>In general pdf tables cannot be read programaticaly from R or
Python.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="impossible-numbers-1" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="impossible-numbers-1" class="callout-inner">
<h3 class="callout-title">Impossible numbers</h3>
<div class="callout-content">
<p>Systems biologists usually require raw numerical data to build their
models. Take a look at the following example: Try to find the numerical
data behind the graph shown in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC166576/figure/F6/" class="external-link">Figure
6</a> which demonstrates changes in levels of phytochrome proteins from
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC166576/" class="external-link">Sharrock
RA and Clack T, 2002</a>.</p>
<ul>
<li>Hint 1: Materials and methods describe quantification procedure</li>
<li>Hint 2: Supporting Information or Supplementary Materials sections
often contain data files.</li>
</ul>
<p>How easy it was?</p>
</div>
</div>
</div>
<div id="impossible-resourcelink" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="impossible-resourcelink" class="callout-inner">
<h3 class="callout-title">Impossible resource/link</h3>
<div class="callout-content">
<p>RNA-seq (transcriptomics) data is usually deposited in online
repositories such as <a href="https://www.ncbi.nlm.nih.gov/sra" class="external-link">SRA</a>
or <a href="https://www.ebi.ac.uk/arrayexpress/" class="external-link">ArrayExpress</a>. Your
task is to find the link to the repository of the raw RNA-seq data in <a href="https://genesdev.cshlp.org/content/26/8/785.long" class="external-link">Li et al., Genes
Dev. 2012</a>. Can you find it anywhere?</p>
</div>
</div>
</div>
<p>(29 min teaching)</p>
<p>The above examples illustrate the typical challenges in accessing
research data and software. Firstly, data/protocols/software often do
not have an identity of their own, but only accompany a publication.
Second, they are not easily accessible or reusable, for example, all the
details are inside one supporting information PDF file. Such file
includes “printed” numerical table or even source code, both of which
need to be “re-typed” if someone would like to use them. Data are shared
in proprietary file format specific to a particular vendor and not
accessible if one does not have a particular software that accompanies
the equipment. Finally, data files are provided without detailed
description other than the whole article text.</p>
<p>In our examples, the protocol was difficult to <strong>find</strong>
(the loops), difficult to <strong>access</strong> (pay wall), and not
<strong>reusable</strong> as it lacked the necessary details (dead-end).
In the second example the data were not <strong>interoperable</strong>
and <strong>reusable</strong> as their were only available as a figure
graph.</p>
<p>To avoid such problems FAIR principles were designed.</p>
<p><img src="../fig/FAIR.png" alt="Figure 2. FAIR principles" class="figure"> After <a href="https://commons.wikimedia.org/wiki/File:FAIR_data_principles.jpg" class="external-link">SangyaPundir</a></p>
<div id="fair-principles" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="fair-principles" class="callout-inner">
<h3 class="callout-title">FAIR Principles</h3>
<div class="callout-content">
<p>In 2016, the <a href="https://www.nature.com/articles/sdata201618" class="external-link">FAIR Guiding
Principles for scientific data management and stewardship</a> were
published in Scientific Data. The original guideline focused on
“machine-actionability” - the ability of computer systems to operate on
data with minimal human intervention. However, now the focus has shifted
to making data accessible from a human perspective, and not an automated
one (mostly due to the lack of user friendly tools that could help deal
with standards and structured metadata).</p>
<p><strong>Findable</strong>: Easy to find data and metadata for both
humans and computers. Automatic and reliable discovery of datasets and
services depends on machine-readable persistent identifiers (PIDs) and
metadata.</p>
<p><strong>Accessible</strong>: (Meta)data should be retrievable by
their identifier using a standardized and open communications protocol
(including authentication and authorisation). Metadata should be
available even when the data are no longer available.</p>
<p><strong>Interoperable</strong>: Data should be able to be combined
with and used with other data or tools. The format of the data should be
open and interpretable for various tools. It applies both to data and
metadata, (meta)data should use vocabularies that follow FAIR
principles.</p>
<p><strong>Re-usable</strong>: FAIR aims at optimizing the reuse of
data. Metadata and data should be well-described so that they can be
replicated and/or combined in different settings. The reuse of
(meta)data should be stated with clear and accessible license(s).</p>
</div>
</div>
</div>
<section><h2 class="section-heading" id="fair-in-biological-practice">FAIR in biological practice<a class="anchor" aria-label="anchor" href="#fair-in-biological-practice"></a>
</h2>
<hr class="half-width">
<div class="section level4">
<h4 id="findable-accessible">Findable &amp; Accessible<a class="anchor" aria-label="anchor" href="#findable-accessible"></a>
</h4>
<p>Deposit data to an external, reputable public repository.</p>
<p>Repositories provide persistent identifiers (PIDs), catalogue
options, advanced metadata searching, and download statistics. Some
repositories can also host private data or provide embargo periods,
meaning access to all data can be delayed.</p>
<p>There are general “data agnostic” repositories, for example:</p>
<ul>
<li>
<a href="https://datadryad.org" class="external-link">Dryad</a>,</li>
<li>
<a href="https://zenodo.org" class="external-link">Zenodo</a>,</li>
<li>
<a href="https://figshare.com" class="external-link">FigShare</a>,</li>
<li>
<a href="https://thedata.org" class="external-link">Dataverse</a>.</li>
</ul>
<p>Or domain specific, for example:</p>
<ul>
<li>
<a href="https://www.uniprot.org/" class="external-link">UniProt</a> protein data,</li>
<li>
<a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a> sequence
data,</li>
<li>
<a href="https://www.ebi.ac.uk/metabolights/" class="external-link">MetaboLights</a>
metabolomics data</li>
<li>
<a href="https://github.com/" class="external-link">GitHub</a> for code.</li>
</ul>
<p><em>We will cover repositories in more details in a later
episode.</em></p>
<div id="what-are-persistent-identifiers-pids" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="what-are-persistent-identifiers-pids" class="callout-inner">
<h3 class="callout-title">What are persistent identifiers (PIDs)</h3>
<div class="callout-content">
<p>A persistent identifier is a long-lasting reference to a digital
resource. Typically it has two components:</p>
<ul>
<li>a service that locates the resource over time even when its location
changes</li>
<li>and a unique identifier (that distinguishes the resource or concept
from others).</li>
</ul>
<p>Persistent identifiers aim to solve the problem of the persistence of
accessing cited resource, particularly in the field of academic
literature. All too often, web addresses (links) changes over time and
fail to take you to the referenced resource you expected.</p>
<p>There are several services and technologies (schemes) that provide
PIDs for objects (whether digital, physical or abstract). One of the
most popular is <strong>Digital Object Identifier <a href="https://www.doi.org/" class="external-link">(DOI)</a></strong>, recognizable by the
prefix doi.org in the web links. For example: <a href="https://doi.org/10.1038/sdata.2016.18" class="external-link uri">https://doi.org/10.1038/sdata.2016.18</a> resolves to the
location of the paper that describes FAIR principles.</p>
<p>Public repositories often maintain web addresses of their content in
a stable form which follow the convention <a href="https://repository.adress/identifier" class="external-link">http://repository.adress/identifier</a>;
these are often called permalinks. For well establish services,
permalinks can be treated as PIDs.</p>
<p>For example: <a href="https://identifiers.org/SO:0000167" class="external-link">http://identifiers.org/SO:0000167</a>
resolves to a page defining promoter role, and can be used to annotate
part of a DNA sequence as performing such a role during
transcription.</p>
</div>
</div>
</div>
</div>
<div class="section level4">
<h4 id="interoperable">Interoperable<a class="anchor" aria-label="anchor" href="#interoperable"></a>
</h4>
<ul>
<li>Use common file formats (can be domain specific)</li>
<li>Always use .csv or .xls files for numerical data.
<strong>Never</strong> share data tables as word or pdf,</li>
<li>Provide underlying numerical data for all plots and graphs</li>
<li>Convert proprietary binary formats to the open ones. For example
convert Snapgene to Genbank, microscopy multistack images to
OME-TIFF</li>
</ul>
</div>
<div class="section level4">
<h4 id="reusable">Reusable<a class="anchor" aria-label="anchor" href="#reusable"></a>
</h4>
<ol style="list-style-type: decimal">
<li>Describe your data well / provide good metadata</li>
</ol>
<ul>
<li>write README file describing the data</li>
<li>user descriptive column headers for the data tables</li>
<li>tidy data tables, make them analysis friendly</li>
<li>provide as many details as possible (prepare good metadata)</li>
<li>use (meta)data formats (e.g. SBML, SBOL)</li>
<li>follow Minimum Information Standards</li>
</ul>
<p><em>Describing data well is the most challenging part of the data
sharing process. We will cover this in more detail later on.</em></p>
<ol start="2" style="list-style-type: decimal">
<li>Attach license files. Licenses explicitly declare conditions and
terms by which data and software can be re-used. Here, we
recommend:</li>
</ol>
<ul>
<li>for data <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">Creative Commons
Attribution (CC BY)</a> license,</li>
<li>for code a permissive open source license such as the <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a>, <a href="https://opensource.org/licenses/BSD-2-Clause" class="external-link">BSD</a>, or <a href="https://www.apache.org/licenses/" class="external-link">Apache license</a>.</li>
</ul>
<div id="copyright-and-data" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="copyright-and-data" class="callout-inner">
<h3 class="callout-title">Copyright and data</h3>
<div class="callout-content">
<p>Software code (the text) automatically gets the default copyright
protection which prevents others from copying or modifying it. Only by
adding the explicit licence you can permit re-use by others.</p>
<p>Data, being factual, cannot be copyrighted. <strong>So why, do we
need a license?</strong></p>
<p>While the data itself cannot be copyrighted, the way how it is
presented can be. The extend to which it is protected needs ultimately
to be settled by the court.</p>
<p>The “good actors” will restrain from using your data to avoid “court”
risks. The “bad actors” will either ignore the risk or can afford the
lawyers fees.</p>
</div>
</div>
</div>
<div id="exercise-2-example-of-fair-data-4-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-example-of-fair-data-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Example of FAIR data (4 min)</h3>
<div class="callout-content">
<p>Zenodo is general data repository. Have a look at the dataset record
with COVID-19 data: <a href="https://doi.org/10.5281/zenodo.6339631" class="external-link uri">https://doi.org/10.5281/zenodo.6339631</a></p>
<p>Identify how each of F.A.I.R principles has been met.<br><em>Hint: navigate to linked github record to easily access the README
file</em></p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<ul>
<li><ol start="6" style="list-style-type: upper-alpha">
<li>The dataset is identified by a PID (doi). It can be found by its ID.
It has human accessible description and keywords, both suitable for
discovery.</li>
</ol></li>
<li><ol style="list-style-type: upper-alpha">
<li>Data can be downloaded using standard browser.</li>
</ol></li>
<li><ol style="list-style-type: upper-roman">
<li>Dataset entries are in common formats: csv, R, jpg</li>
</ol></li>
<li><ol style="list-style-type: upper-roman">
<li>Dataset is linked to publication, github record and project
website</li>
</ol></li>
<li><ol start="18" style="list-style-type: upper-alpha">
<li>The record contains rich metadata in README file, including files
structure and the detailed tables formats.</li>
</ol></li>
<li><ol start="18" style="list-style-type: upper-alpha">
<li>Data are released under open Creative Commons Attribution
Licence</li>
</ol></li>
</ul>
</div>
</div>
</div>
</div>
<div id="exercise-2-tooadvanced-example-of-fair-data-4-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-tooadvanced-example-of-fair-data-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 2 (too)Advanced: Example of FAIR data (4 min)</h3>
<div class="callout-content">
<p><em>Depending on the audience this version of the excercise may
present information overload.</em></p>
<p>Uniprot is a high-quality and freely accessible resource of protein
sequence and functional information. Have a look at the record of the
GFP protein: <a href="https://www.uniprot.org/uniprot/P42212" class="external-link uri">https://www.uniprot.org/uniprot/P42212</a></p>
<p>Identify how each of F.A.I.R principles has been met.</p>
<p><em>Alternative records to check:</em> <em><a href="https://www.ncbi.nlm.nih.gov/nuccore/AH002877.2" class="external-link uri">https://www.ncbi.nlm.nih.gov/nuccore/AH002877.2</a> <a href="https://synbiohub.org/public/bsu/SubtilinReceiver_spaRK_separated/1" class="external-link uri">https://synbiohub.org/public/bsu/SubtilinReceiver_spaRK_separated/1</a></em></p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>All entries are uniquely identified by a stable URL (‘F’), that
provides access to the record in a variety of formats including a web
page, plain-text, FASTA, and GENBANK (‘A’, ‘I’). The record contains
rich metadata (‘R’) that is both human-readable (HTML) and
machine-readable (text) (‘I’). There search options that uses both
record IDs and the rich metadata (‘F’). The graphical pane shows how the
data are re-usable. Metadata uses ontological terms (e.g. taxonomy) and
shared vocabularies (e.g. genbank features) (‘I’). Interlinking with
other databases: GenBank, Pubmed (‘I’, ‘R’) enabling automated retrieval
of records and cross-referencing information.</p>
</div>
</div>
</div>
</div>
<div id="exercise-3-fair-and-you-3-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-fair-and-you-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 3: FAIR and You (3 min)</h3>
<div class="callout-content">
<p>The FAIR acronym is sometimes accompanied with the following
labels:</p>
<ul>
<li>Findable - Citable</li>
<li>Accessible - Trackable and countable</li>
<li>Interoperable - Intelligible</li>
<li>Reusable - Reproducible</li>
</ul>
<p>Using those labels as hints discuss how FAIR principles directly
benefit you as the data creators.</p>
</div>
</div>
</div>
<div id="accordionSolution5" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution5" aria-expanded="false" aria-controls="collapseSolution5">
  <h4 class="accordion-header" id="headingSolution5"> Show me the solution </h4>
</button>
<div id="collapseSolution5" class="accordion-collapse collapse" data-bs-parent="#accordionSolution5" aria-labelledby="headingSolution5">
<div class="accordion-body">
<ul>
<li>Findable data have their own identity, so they can be easily cited
and secure the credits to the authors</li>
<li>Data accessibility over the Internet using standard protocols can be
easily monitored (for example using Google analytics). This results in
metrics on data popularity or even geo-locations of data users.</li>
<li>Interoperable data can benefit the future you, for example you will
be able to still read your data even when you no longer have access to
the specialized, vendor specific software with which you worked with
them before. Also the future you may not remember abreviations and
ad-hoc conventions you used before (Intelligible).</li>
<li>Well documented data should contain all the details necessary to
reproduce the experiments, helping the future you or someone taking over
from you in the laboratory.</li>
<li>Saves time and money.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="fair-vs-open-science-3-min-teaching">FAIR vs Open Science (3 min teaching)<a class="anchor" aria-label="anchor" href="#fair-vs-open-science-3-min-teaching"></a>
</h2>
<hr class="half-width">
<p><strong>FAIR does not mean Open</strong>. Actually, FAIR guideline
only requires that the metadata record is always accessible. For
example, the existence of the data can be known (their metadata), the
data can have easy to use PID to reference them, but the actual data
files can only be downloaded after the login and authorization.</p>
<p>However, if data are already in the FAIR form, i.e. accessible over
the internet, in interoperable format and well documented, then it is
almost effortless to “open” the dataset and make it available to the
whole public. The data owner can do it any time when he no longer
perceives oppening as a risk.</p>
<p>At the same time, Open data which does not follow FAIR guidelines
have little value. If they are not well described, not in open formats
then they are not going to be re-used even if they were made “open” by
posting them on some website.</p>
<div id="exercise-4-fair-quiz-5-min---run-through-break" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-fair-quiz-5-min---run-through-break" class="callout-inner">
<h3 class="callout-title">Exercise 4: FAIR Quiz (5 min - run through break)</h3>
<div class="callout-content">
<p>Which of the following statements is true/false (T or F).</p>
<ul>
<li>F in FAIR stands for free.</li>
<li>Only figures presenting results of statistical analysis need
underlying numerical data.</li>
<li>Sharing numerical data as a .pdf in Zenodo is FAIR.</li>
<li>Sharing numerical data as an Excel file via Github is not FAIR.</li>
<li>Group website is a good place to share your data.</li>
<li>Data from failed experiments are not re-usable.</li>
<li>Data should always be converted to Excel or .csv files in order to
be FAIR.</li>
<li>A DOI of a dataset helps in getting credit.</li>
<li>FAIR data are peer reviewed.</li>
<li>FAIR data accompany a publication.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution6" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution6" aria-expanded="false" aria-controls="collapseSolution6">
  <h4 class="accordion-header" id="headingSolution6"> Show me the solution </h4>
</button>
<div id="collapseSolution6" class="accordion-collapse collapse" data-bs-parent="#accordionSolution6" aria-labelledby="headingSolution6">
<div class="accordion-body">
<ul>
<li>F in FAIR stands for free. F</li>
<li>Only figures presenting results of statistical analysis need
underlying numerical data. F</li>
<li>Sharing numerical data as a .pdf in Zenodo is FAIR. F</li>
<li>Sharing numerical data as an Excel file via Github is not FAIR.
F</li>
<li>Group website is a good place to share your data. F</li>
<li>Data from failed experiments are not re-usable. F</li>
<li>Data should always be converted to Excel or .csv files in order to
be FAIR. F</li>
<li>A DOI of a dataset helps in getting credit. T</li>
<li>FAIR data are peer reviewed. F</li>
<li>FAIR data accompany a publication. F</li>
</ul>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>FAIR stands for Findable Accessible Interoperable Reusable</li>
<li>FAIR assures easy reuse of data underlying scientific findings</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-04-IP-licensing"><p>Content from <a href="04-IP-licensing.html">Intellectual Property, Licensing and Openness</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/04-IP-licensing.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 20 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is intellectual property?</li>
<li>Why should I consider IP in Open Science?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Timeline matters for legal protection</li>
<li>Understand what can and cannot be patented</li>
<li>Understand what licenses to use for re-use of data and software</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(13 min teaching)</p>
<section><h2 class="section-heading" id="open-science-and-intellectual-property">Open Science and Intellectual property<a class="anchor" aria-label="anchor" href="#open-science-and-intellectual-property"></a>
</h2>
<hr class="half-width">
<div id="callout1" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout</h3>
<div class="callout-content">
<p><em>This section is tailored to the EU and UK.</em></p>
<p>Intellectual property (IP) is something that you create using your
mind - for example, a story, an invention, an artistic work or a
symbol.</p>
<p>The timeline of “opening” matters when one seeks legal protection for
his IP.</p>
<p>For example, patents are granted only for inventions that are
<strong>new and were not known to the public</strong> in any form.
Publishing in a journal or presenting in a conference information
related to the invention completely <strong>prevents the inventor from
getting a patent</strong>!</p>
<p>In our opinion, you are more likely to benefit from new
collaborations, industrial partnerships, consultations which are
acquired by openness, than from patent related royalties.</p>
</div>
</div>
</div>
<div id="optional-intellectual-property-protection" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="optional-intellectual-property-protection" class="callout-inner">
<h3 class="callout-title">(Optional) Intellectual property protection</h3>
<div class="callout-content">
<p><em>This section is tailored to the EU and UK.</em></p>
<p>You can use a patent to protect your (<em>technical</em>) invention.
It gives you the right to take legal action against anyone who makes,
uses, sells or imports it without your permission.</p>
<p>Discoveries, mathematical methods, computer programs and business
methods as such are not regarded as inventions. Surgical and therapeutic
procedures, diagnostic methods and new plant or animal varieties are
completely excluded from patentability.</p>
<p>Patents are granted only for inventions that are <strong>new and were
not known to the public</strong> in any form. Publishing in a journal or
presenting in a conference information related to the invention
completely <strong>prevents the inventor from getting a
patent</strong>!</p>
<p>In principle, software cannot be patented. However, it is a “no but
yes” situation, and a software patent are being granted. It is usually,
settled by the court for each case.</p>
<p><strong>Software code is copyrighted</strong>. Copyright prevents
people from:</p>
<ul>
<li>copying your code</li>
<li>distributing copies of it, whether free of charge or for sale.</li>
</ul>
<p>Data cannot be patented, and in principle, it cannot be copyrighted.
<strong>It is not possible to copyright facts!</strong></p>
<p>However, how data are collated and presented (especially if it is a
database), can have a layer of copyright protection. Deciding what data
needs to be included in a database, how to organize the data, and how to
relate different data elements are all creative decisions that may
receive copyright protection. Again, it is often a case by case
situation and may come down to who has better lawyers.</p>
<p>After: UK Government, <a href="https://www.gov.uk/browse/business/intellectual-property" class="external-link">Intellectual
Property</a> European Patent Office, <a href="https://www.epo.org/learning/materials/inventors-handbook.html" class="external-link">Inventor’s
Handbook</a></p>
</div>
</div>
</div>
<div id="exercise-1-checking-common-licenses-5-min-discussion" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-checking-common-licenses-5-min-discussion" class="callout-inner">
<h3 class="callout-title">Exercise 1: Checking common licenses (5 min + discussion)</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li><p>Open CC BY license summary <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link uri">https://creativecommons.org/licenses/by/4.0/</a> is it clear
how you can use the data under this licence and why it is popular in
academia?</p></li>
<li><p>Check the MIT license wording: <a href="https://opensource.org/licenses/MIT" class="external-link uri">https://opensource.org/licenses/MIT</a> is it clear what you
can do with software code under this licence?</p></li>
<li><p>Compare the full wording of CC BY <a href="https://creativecommons.org/licenses/by/4.0/legalcode" class="external-link uri">https://creativecommons.org/licenses/by/4.0/legalcode</a>
can you guess why the MIT licence is currently one of the most popular
for open source code?</p></li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>CC BY license states material can be reproduced, shared, in whole or
in part, unless where exceptions and limitations are stated.
Attributions must be made to the Licensor.</li>
<li>MIT license states that Software can by used without restriction (to
copy, modify, publish, distribute etc…)</li>
<li>The MIT license is short, to the point and optimised for software
developers as it offers flexibility.</li>
</ol>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>A license is a promise not to sue - therefore attach license
files</li>
<li>For data use Creative Commons Attribution (CC BY) license (CC0 is
even more permissive)</li>
<li>For code use open source licenses such as MIT, BSD, or Apache
license</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-05-intro-to-metadata"><p>Content from <a href="05-intro-to-metadata.html">Introduction to metadata</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/05-intro-to-metadata.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 25 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is metadata?</li>
<li>What do we use metadata for?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Recognise what metadata is</li>
<li>Distinguish different types of metadata</li>
<li>Understand what makes metadata interoperable</li>
<li>Know how to decide what to include in metadata</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(5 min teaching)</p>
<section><h2 class="section-heading" id="what-is-or-are-metadata">What is (or are) metadata?<a class="anchor" aria-label="anchor" href="#what-is-or-are-metadata"></a>
</h2>
<hr class="half-width">
<p>Simply put, metadata is data about the data. Sound confusing? Lets
clarify: metadata is the <em>description of data</em>. It allows deeper
understanding of data and provides insight for its interpretation.
Hence, your metadata should be considered as important as your data.
Further, metadata plays a very important role in making your data FAIR.
It should be continuously added to your research data (not just at the
beginning or end of a project!). Metadata can be produced in an
automated way (e.g. when you capture a microscopy image usually the
accompanying software saves metadata as part of it) or manually.</p>
<ins>
Let’s take a look at an example:
</ins>
<p>This is a confocal microscopy image of a <em>C. elegans</em> nematode
strain used as a proteostasis model (Pretty! Isn’t it?). The image is
part of the raw data associated to <a href="https://doi.org/10.1016/j.celrep.2019.12.078" class="external-link">Goya et al.,
2020</a>, which was deposited in a Public Omero Server<br><a href="https://publicomero.bio.ed.ac.uk/webclient/?show=project-58" class="external-link">Project</a><br><a href="https://publicomero.bio.ed.ac.uk/webclient/?show=dataset-231" class="external-link">Figure1
set</a></p>
<p><img src="../fig/04-microscopy_with_metadata.png" alt="nematode_confocal_microscopy_image" width="600" class="figure"><em>Figure credits: María Eugenia Goya</em></p>
<p>. What information <strong>can you get</strong> from the image,
without the associated description (metadata)?</p>
<p>Let’s see the associated metadata of the image and the dataset to
which it belongs:</p>
<div id="image-metadata" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="image-metadata" class="callout-inner">
<h3 class="callout-title">Image metadata</h3>
<div class="callout-content">
<p>Name: OP50 D10Ad_06.czi Image ID: 3485 Owner: Maria Eugenia Goya
ORCID: 0000-0002-5031-2470</p>
<p>Acquisition Date: 2018-12-12 17:53:55 Import Date: 2020-04-30
22:38:59 Dimensions (XY): 1344 x 1024 Pixels Type: uint16 Pixels Size
(XYZ) (µm): 0.16 x 0.16 x 1.00 Z-sections/Timepoints: 56 x 1 Channels:
TL DIC, TagYFP ROI Count: 0</p>
<p>Tags: time course; day 10; adults; food switching; E. coli OP50;
NL5901; C. elegans</p>
<div class="section level2">
<h2 id="dataset-metadata">Dataset metadata<a class="anchor" aria-label="anchor" href="#dataset-metadata"></a>
</h2>
<p>Name: Figure2_Figure2B Dataset ID: 263 Owner: Maria Eugenia Goya
ORCID: 0000-0002-5031-2470</p>
<p>Description: The datasets contains a time course of α-syn aggregation
in NL5901 C. elegans worms after a food switch at the L4 stage:</p>
<p>E. coli OP50 to OP50 Day 01 adults Day 03 adults Day 05 adults Day 07
adults Day 10 adults Day 13 adults</p>
<p>E. coli OP50 to B. subtilis PXN21 Day 01 adults Day 03 adults Day 05
adults Day 07 adults Day 10 adults Day 13 adults</p>
<p>Images were taken at 6 developmental timepoints (D1Ad, D3Ad, D5Ad,
D7Ad, D10Ad, D13Ad)</p>
<p>* Some images contain more than one nematode.</p>
<p>Each image contains ~30 (or more) Z-sections, 1 µmeters apart. The
TagYFP channel is used to follow the alpha-synuclein particles. The TL
DIC channel is used to image the whole nematode head.</p>
<p>These images were used to construct Figure 2B of the Cell Reports
paper (<a href="https://doi.org/10.1016/j.celrep.2019.12.078" class="external-link uri">https://doi.org/10.1016/j.celrep.2019.12.078</a>).</p>
<p>Creation date: 2020-04-30 22:16:39</p>
<p>Tags: protein aggregation; time course; E. coli OP50 to B. subtilis
PXN21; food switching; E. coli OP50; 10.1016/j.celrep.2019.12.078;
NL5901; C. elegans</p>
</div>
</div>
</div>
</div>
<p>This is a lot of information!</p>
</section><section><h2 class="section-heading" id="types-of-metadata">Types of metadata<a class="anchor" aria-label="anchor" href="#types-of-metadata"></a>
</h2>
<hr class="half-width">
<p>According to <a href="https://howtofair.dk/" class="external-link">How to FAIR</a> we can
distinguish between three main types of metadata:</p>
<ul>
<li>
<strong>Administrative metadata:</strong> data about a project or
resource that are relevant for managing it; E.g. project/resource owner,
principal investigator, project collaborators, funder, project period,
etc. They are usually assigned to the data, before you collect or create
them.</li>
<li>
<strong>Descriptive or citation metadata:</strong> data about a
dataset or resource that allow people to discover and identify it; E.g.
authors, title, abstract, keywords, persistent identifier, related
publications, etc.</li>
<li>
<strong>Structural metadata:</strong> data about how a dataset or
resource came about, but also how it is internally structured. E.g. the
unit of analysis, collection method, sampling procedure, sample size,
categories, variables, etc. Structural metadata have to be gathered by
the researchers according to best practice in their research community
and will be published together with the data.</li>
</ul>
<p>Descriptive and structural <strong>metadata should be added
continuously</strong> throughout the project.</p>
<div id="exercise-1-identifying-metadata-types-4-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-identifying-metadata-types-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Identifying metadata types (4 min)</h3>
<div class="callout-content">
<p>Here we have an excel spreadsheet that contains project metadata for
a made-up experiment of plant metabolites <img src="../fig/04-metadatafull_spreadsheet.png" alt="Metadata in data table example" class="figure"><em>Figure credits: Tomasz
Zielinski and Andrés Romanowski</em></p>
<p>In groups, identify different types of metadata (administrative,
descriptive, structural) present in this example.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<ul>
<li>Administrative metadata marked in blue</li>
<li>Descriptive metadata marked in orange</li>
<li>Structural metadata marked in green <img src="../fig/04-metadatafull_spreadsheet_solution.png" alt="metadata-full-spreadhseet" class="figure"><em>Figure credits: Tomasz Zielinski
and Andrés Romanowski</em>
</li>
</ul>
</div>
</div>
</div>
</div>
<p>(6 min teaching)</p>
<div id="where-does-data-end-and-metadata-start" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="where-does-data-end-and-metadata-start" class="callout-inner">
<h3 class="callout-title">Where does data end and metadata start?</h3>
<div class="callout-content">
<p>What is “data” and what is “metadata” can be a matter of perspective:
Some researchers’ metadata can be other researchers’ data.</p>
<p>For example, a funding body is categorised as typical administrative
metadata, however, it can be used to calculate numbers of public
datasets per funder and then used to compare effects of different
funders’ policies on open practices.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="adding-metadata-to-your-experiments">Adding metadata to your experiments<a class="anchor" aria-label="anchor" href="#adding-metadata-to-your-experiments"></a>
</h2>
<hr class="half-width">
<p>Good metadata are crucial for assuring re-usability of your outcomes.
Adding metadata is also a very time-consuming process if done manually,
so collecting metadata should be done incrementally during your
experiment.</p>
<p>As we saw metadata can take many forms from as simple as including a
ReadMe.txt file, by embedding them inside the Excel files, to using
domain specific metadata standards and formats.</p>
<p>But,</p>
<ul>
<li><strong>What should be included in metadata?</strong></li>
<li><strong>What terms should be used in descriptions?</strong></li>
</ul>
<p>For many assay methods and experiment types, there are defined
recommendations and guidelines called <strong>Minimal Information
Standards</strong>.</p>
<div id="minimal-information-standard" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="minimal-information-standard" class="callout-inner">
<h3 class="callout-title">Minimal Information Standard</h3>
<div class="callout-content">
<p>The minimum information standard is a set of guidelines for reporting
data derived by relevant methods in biosciences. If followed, it ensures
that the data can be easily verified, analysed and clearly interpreted
by the wider scientific community. Keeping with these recommendations
also facilitates the foundation of structuralized databases, public
repositories and development of data analysis tools. Individual minimum
information standards are brought by the communities of
cross-disciplinary specialists focused on issues of the specific method
used in experimental biology.</p>
<p>Minimum Information for Biological and Biomedical Investigations <a href="https://fairsharing.org/collection/MIBBI" class="external-link">(MIBBI)</a> is the
collection of the most known standards.</p>
<p><a href="https://fairsharing.org/standards/" class="external-link">FAIRSharing</a> offers
excellent search service for finding standards</p>
</div>
</div>
</div>
<div id="exercise-2-minimal-information-standard-example-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-minimal-information-standard-example-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Minimal information standard example (5 min)</h3>
<div class="callout-content">
<p>Look at <strong><em>Minimum Information about a Neuroscience
Investigation (MINI) Electrophysiology</em></strong> <a href="https://www.nature.com/articles/npre.2008.1720.1.pdf" class="external-link">Gibson, F.
et al. Nat Prec (2008)</a>. which contains recommendations for reporting
the use of electrophysiology in a neuroscience study.<br>
(<em>Neuroscience (or neurobiology) is the scientific study of the
nervous system</em>).</p>
<p>Scroll to <em>Reporting requirement</em> and decide which of the
points 1-8 are:</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>important for understanding and reuse of data</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>important for technical replication</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>could be applied to other experiments in neuroscience</li>
</ol></li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Possible answers:</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>2, 3, 4, 5, 6, 8a-b</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>3, 7</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>2, 3, 4, 5, 6</li>
</ol></li>
</ul>
</div>
</div>
</div>
</div>
<p>What if there are no metadata standards defined for your data / field
of research?</p>
<p>Think about the minimum information that someone else (from your lab
or from any other lab in the world) would need to know to be able to
work with your dataset without any further input from you.</p>
<p><strong>Think as a consumer</strong> of your data not the
producer!</p>
<div id="exercise-3-what-to-include---discussion-4-minutes" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-what-to-include---discussion-4-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 3: What to include - discussion (4 minutes)</h3>
<div class="callout-content">
<p>Think of the data you generate in your projects, and imagine you are
going to share them.</p>
<p>What information would another researcher need to understand or
reproduce your data (the structural metadata)?</p>
<p>For example, we believe that any dataset should have:</p>
<ul>
<li>a name/title</li>
<li>its purpose or experimental hypothesis</li>
</ul>
<p>Write down and compare your proposals, can we find some common
elements?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>Some typical elements are:</p>
<ul>
<li>biological material, e.g. Species, Genotypes, Tissue type, Age,
Health conditions</li>
<li>biological context, e.g. speciment growth, entrainment, samples
preparation</li>
<li>experimental factors and conditions, e.g. drug treatments, stress
factors</li>
<li>primers, plasmid sequences, cell line information, plasmid
construction</li>
<li>specifics of data acquisition</li>
<li>specifics of data processing and analysis</li>
<li>definition of variables</li>
<li>accompanying code, software used (version nr), parameters applied,
statistical tests used, seed for randomisation</li>
<li>LOT numbers</li>
</ul>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="metadata-and-fair-guidelines">Metadata and FAIR guidelines<a class="anchor" aria-label="anchor" href="#metadata-and-fair-guidelines"></a>
</h2>
<hr class="half-width">
<p>Metadata provides extremely valuable information for us and others to
be able to interpret, process, reuse and reproduce the research data it
accompanies.</p>
<p>Because metadata are data about data, all of the FAIR principles i.e.
Findable, Accessible, Interoperable and Reusable apply to metadata.</p>
<p>Ideally, metadata should not only be machine-readable, but also
interoperable so that they can interlink or be reasoned about by
computer systems.</p>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was adapted from:</p>
<ul>
<li>
<a href="https://escience-academy.github.io/Lesson-FAIR-Data-Climate/metadata/index.html" class="external-link">Metadata
- FAIR data for climate sciences</a>.</li>
<li><a href="https://howtofair.dk/how-to-fair/metadata/" class="external-link">Metadata - How
to FAIR</a></li>
<li><a href="https://fairsharing.org/collection/MIBBI" class="external-link">MIBBI</a></li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Metadata provides contextual information so that other people can
understand the data.</li>
<li>Metadata is key for data reuse and complying with FAIR
guidelines.</li>
<li>Metadata should be added incrementally through out the project</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-06-being-precise"><p>Content from <a href="06-being-precise.html">Being precise</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/06-being-precise.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 35 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to make my metadata interoperable?</li>
<li>How to avoid disambiguation?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Using public identifiers</li>
<li>Understand difference between close vocabulary and ontology</li>
<li>Finding ontology terms</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(16 min teaching)</p>
<section><h2 class="section-heading" id="being-precise">Being precise<a class="anchor" aria-label="anchor" href="#being-precise"></a>
</h2>
<hr class="half-width">
<p>If the metadata purpose is to help understand the data, it has to be
done in a precise and “understandable” way i.e. it has to be
interoperable. To be interoperable metadata should use a
<strong><em>formal, accessible, shared, and broadly applicable
terms/language</em></strong> for knowledge representation.</p>
<p>One of the easiest examples is the problem of author
disambiguation.</p>
<figure><img src="../fig/04-names-example.jpg" alt="Why we need ORCID" class="figure mx-auto d-block"></figure><p><em>After <a href="https://librarycarpentry.org/lc-fair-research/" class="external-link">Libarary Carpentry
FAIR Data</a></em></p>
<div id="open-researcher-and-contributor-id-orcid" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="open-researcher-and-contributor-id-orcid" class="callout-inner">
<h3 class="callout-title">Open Researcher and Contributor ID (ORCID)</h3>
<div class="callout-content">
<p>Have you ever searched yourself in pubmed and found that you have a
<em>doppelganger</em>? So how can you uniquely associate something you
created to just yourself and no other researcher with the same name?</p>
<p><a href="https://orcid.org/" class="external-link">ORCID</a> is a free, unique, persistent
identifier that you own and control—forever. It distinguishes you from
every other researcher across disciplines, borders, and time.</p>
<p>ORCIDs of authors of this episode are:</p>
<ul>
<li><a href="https://orcid.org/0000-0002-0194-5706" class="external-link">0000-0002-0194-5706</a></li>
<li><a href="https://orcid.org/0000-0003-0737-2408" class="external-link">0000-0003-0737-2408</a></li>
</ul>
<p>You can connect your iD with your professional
information—affiliations, grants, publications, peer review, and more.
You can use your iD to share your information with other systems,
ensuring you get recognition for all your contributions, saving you time
and hassle, and reducing the risk of errors.</p>
<p>If you do not have an <a href="https://orcid.org/" class="external-link">ORCID</a>, you
should register to get one!</p>
</div>
</div>
</div>
<div id="exercise-1-public-id-in-action-3-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-public-id-in-action-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Public ID in action (3 min)</h3>
<div class="callout-content">
<p>The Wellcome Open Research journal uses ORCID to identify
authors.</p>
<ul>
<li><p>Open one of our papers <a href="https://doi.org/10.12688/wellcomeopenres.15341.2" class="external-link">doi.org/10.12688/wellcomeopenres.15341.2</a>
and have a look how public IDs such as ORCID can be used to interlink
information.</p></li>
<li><p>If you have not done so yet, register yourself at ORCID*</p></li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<p>ORCID is used to link to authors profiles which list their other
publications.</p>
</div>
</div>
</div>
</div>
<p>ORCID provides the registry of researchers, so they can be precisely
identified. Similarly, there are other registries that can be used to
identify many of biological concepts and entities:</p>
<ul>
<li>species e.g. <a href="https://www.ncbi.nlm.nih.gov/Taxonomy" class="external-link">NCBI
taxonomy</a>
</li>
<li>chemicals e.g. <a href="https://www.ebi.ac.uk/chebi" class="external-link">ChEBI</a>
</li>
<li>proteins e.g. <a href="https://www.uniprot.org/" class="external-link">UniProt</a>
</li>
<li>genes e.g. <a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a>
</li>
<li>metabolic reactions, enzymes e.g <a href="https://www.genome.jp/kegg/" class="external-link">KEGG</a>
</li>
</ul>
<p><a href="https://bioportal.bioontology.org/" class="external-link">BioPortal</a> or <a href="https://www.ncbi.nlm.nih.gov/" class="external-link">NCBI</a><br>
are good places to start searching for a registry or a term.</p>
<div id="exercise-2-public-id-in-action-2-3-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-public-id-in-action-2-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Public ID in action 2 (3 min)</h3>
<div class="callout-content">
<ul>
<li>
<p>The second metadata example (the Excel table): contains two other
types of public IDs.<br><img src="../fig/04-metadatafull_spreadsheet.png" alt="Metadata in data table example" class="figure"><em>Figure credits: Tomasz
Zielinski and Andrés Romanowski</em></p>
<ul>
<li>Can you find the public IDs?</li>
<li>Can you find the meaning behind those Ids?</li>
</ul>
</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" aria-labelledby="headingSolution2" data-bs-parent="#accordionSolution2">
<div class="accordion-body">
<p>The metadata example contains genes IDs from The Arabidopsis
Information Resource <a href="https://www.arabidopsis.org" class="external-link">TAIR</a> and
metabolites IDs from <a href="https://www.genome.jp/kegg/compound/" class="external-link">KEGG</a></p>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="disambiguation-7-min-teaching">Disambiguation (7 min teaching)<a class="anchor" aria-label="anchor" href="#disambiguation-7-min-teaching"></a>
</h2>
<hr class="half-width">
<p>In academic disciplines we quickly run into problems of naming
standards e.g.:</p>
<ul>
<li>Escherichia coli</li>
<li>EColi</li>
<li>E. coli</li>
<li>E. Coli</li>
<li>Kanamycin A</li>
<li>Kanamycin</li>
<li>Kanam.</li>
<li>Kan.</li>
</ul>
<p>In order to prevent such ambiguities applications provide interface
that constraints users to pre-defined options, it controls the available
vocabulary.</p>
<p><img src="../fig/06-controlled_voc_UI.jpg" alt="UI with controlled vocabulary" class="figure"><em>Example of graphical user
interfaces with controlled vocabularies</em></p>
<div id="controlled-vocabulary" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="controlled-vocabulary" class="callout-inner">
<h3 class="callout-title">Controlled Vocabulary</h3>
<div class="callout-content">
<p><strong>Definition:</strong> Any closed prescribed
<strong><em>list</em></strong> of terms</p>
<p><strong>Key Features:</strong></p>
<ul>
<li>Terms are not usually defined</li>
<li>Relationships between the terms are not usually defined</li>
<li>the simplest form is a list</li>
</ul>
<p><strong>Example:</strong></p>
<ul>
<li>E. coli</li>
<li>Drosophila melanogaster</li>
<li>Homo sapiens</li>
<li>Mus musculus</li>
<li>Salmonella</li>
</ul>
</div>
</div>
</div>
<p>Use of controlled vocabulary (a list) can be organised hierarchically
into a taxonomy, a system we know mostly from our species taxonomy.</p>
<div id="taxonomy" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="taxonomy" class="callout-inner">
<h3 class="callout-title">Taxonomy</h3>
<div class="callout-content">
<p><strong>Definition:</strong> Any controlled vocabulary that is
arranged in a <strong><em>hierarchy</em></strong></p>
<p><strong>Key Features:</strong></p>
<ul>
<li>Terms are not usually defined</li>
<li>Relationships between the terms are not usually defined</li>
<li>Terms are arranged in a hierarchy</li>
</ul>
<p><strong>Example:</strong></p>
<ul>
<li>Bacteria</li>
<li><ul>
<li>E. coli</li>
</ul></li>
<li><ul>
<li>Salmonella</li>
</ul></li>
<li>Eucariota</li>
<li><ul>
<li>Mammalia</li>
</ul></li>
<li><ul>
<li><ul>
<li>Homo sapiens</li>
</ul></li>
</ul></li>
<li><ul>
<li><ul>
<li>Mus musculus</li>
</ul></li>
</ul></li>
<li>Insecta</li>
<li><ul>
<li>Drosophila melanogaster</li>
</ul></li>
</ul>
</div>
</div>
</div>
<p>Ontologies add a further dimension to controlled vocabularies and
taxonomy. They allow us to conceptualise relationships between the
established hierarchy which helps with more sophisticated data queries
and metadata searches.</p>
<div id="ontology" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="ontology" class="callout-inner">
<h3 class="callout-title">Ontology</h3>
<div class="callout-content">
<p><strong>Definition:</strong> A formal
<strong><em>conceptualisation</em></strong> of a specified domain</p>
<p><strong>Key Features:</strong></p>
<ul>
<li>Terms are DEFINED</li>
<li>Relationships between the terms are DEFINED, allowing logical
inference and sophisticated data queries</li>
<li>Terms are arranged in a hierarchy</li>
<li>expressed in a knowledge representation language such as RDFS, OBO,
or OWL</li>
</ul>
<p><strong>Example:</strong></p>
<ul>
<li>Bacteria</li>
<li><ul>
<li>E. coli</li>
</ul></li>
<li><ul>
<li>Salmonella</li>
</ul></li>
<li>Eucariota — has_part —&gt; nucleas</li>
<li><ul>
<li>Mammalia — has_part —&gt; placenta</li>
</ul></li>
<li><ul>
<li><ul>
<li>Homo sapiens</li>
</ul></li>
</ul></li>
<li><ul>
<li><ul>
<li>Mus musculus</li>
</ul></li>
</ul></li>
<li>Insecta</li>
<li><ul>
<li>Drosophila melanogaster</li>
</ul></li>
</ul>
</div>
</div>
</div>
<p>Ontologies represent a standardised, formal naming system and define
categories, properties and relationships between data. Ontologies allow
to describe properties of a subject area and how they are related
(e.g. taxonomy).</p>
<figure><img src="../fig/06-ontologies.png" alt="ontology-example" width="600" class="figure mx-auto d-block"></figure><p>Ontologies allows automatic reasoning about the data, using the
relations between the described terms. For example, in the picture above
we can “deduce” that <strong>hippocampal astrocyte</strong>:</p>
<ul>
<li>it is type of cell as it one of its predecessor in the knowledge
tree is <strong>cell (CL_0000000)</strong>
</li>
<li>it is located in the <strong>brain (UBERON_000955)</strong>
</li>
<li>it can perform <strong>axon regeneration (GO_0031103)</strong>
</li>
</ul>
<p>We could follow the term <em>axon regeneration</em> to find its
definition and what it means, to have a better understanding of
<em>hippocampal astrocyte</em>.</p>
<p>Ontologies are crucial for aggregation of information and finding
suitable data. Imagine you are interested in gene expersions in glial
cells. Using a free text search you would need to hope that data from
astrocytes will be also tagged as glial cell, or you would need to
specify all suptypes of glial cells like oligodendrocytes astrocytes.
Ontologies permits to automatically expands such searches that they
contain suitable domain.</p>
<p>The above figure presents also another aspect of using ontologies.
The terms are often referred to as:<br><strong>PREFIX_ID</strong> e.g. CL_0002604<br>
Prefix (CL, UB, UBERON, GO) will identify the ontology which defined the
term (<a href="https://obofoundry.org/ontology/cl.html" class="external-link">Cell
Ontology</a> for CL).<br>
ID refers to particular term in the ontology.</p>
<p>Ontologies are encoded in standard, interoperable formats, which
permitted creation of re-usable interfaces to access the described terms
or browsing different ontologies. For example: <a href="https://bioportal.bioontology.org/%5D%5C(https://bioportal.bioontology.org/%5C)" class="external-link uri">https://bioportal.bioontology.org/]\(https://bioportal.bioontology.org/\)</a>
permits searching for suitable terms definitions in hundreds of
ontologies.</p>
<p><a href="https://www.obofoundry.org/" class="external-link">Obo foundry</a> is another
useful tool that lists recommended ontologies.</p>
<div id="exercise-3-exploring-ontologies-with-instructor" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-3-exploring-ontologies-with-instructor" class="callout-inner">
<h3 class="callout-title">Exercise 3: Exploring ontologies (with instructor)</h3>
<div class="callout-content">
<p>Check the example of ontology records:</p>
<ul>
<li><a href="https://bioportal.bioontology.org/ontologies/ZFA/?p=classes&amp;conceptid=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FZFA_0001109#details" class="external-link">oocyte</a></li>
<li><a href="https://bioportal.bioontology.org/ontologies/CL?p=classes&amp;conceptid=CL:0000129#details" class="external-link">microglial
cell</a></li>
<li><a href="https://bioportal.bioontology.org/ontologies/SO?p=classes&amp;conceptid=http://purl.obolibrary.org/obo/SO_0000167#details" class="external-link">promoter</a></li>
</ul>
<p>Check what type of information is available, explore the terms
hierarchy, the visualization, the interlinked terms, mapping between
different ontologies.</p>
</div>
</div>
</div>
<div id="exercise-4-ontology-tests" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-ontology-tests" class="callout-inner">
<h3 class="callout-title">Exercise 4: Ontology tests</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>The prefix CL stands for:</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>Class ontology:</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>Cell ontology:</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>Cell line ontology</li>
</ol></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>The recommended ontology for chemical compounds is:</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>cheminf</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>chmo</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>chebi</li>
</ol></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Which terms captures both Alzheimer’s and Huntington’s diseases</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>DOID_680</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>DOID_1289</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>DOID_0060090</li>
</ol></li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" aria-labelledby="headingSolution3" data-bs-parent="#accordionSolution3">
<div class="accordion-body">
<p>3 b (hint try to see the tree in bioportal)</p>
</div>
</div>
</div>
</div>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was adapted from:</p>
<ul>
<li>BD2K Open Educational Resources: BDK14 Ontologies 101 <a href="https://github.com/OHSUBD2K/BDK14-Ontologies-101" class="external-link">Nicole
Vasilevsky</a>
</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Public identifiers and ontologies are key to knowledge
discovery</li>
<li>Automatic data aggregations needs standardised metadata formats and
values</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-07-data-in-excel"><p>Content from <a href="07-data-in-excel.html">(Meta)data in Excel</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/07-data-in-excel.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 55 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How to represent data in tables</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Do’s and don’ts in Excel</li>
<li>How good table organization can helps to re-use data</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(2 min teaching)</p>
<p>Tables are one of the best forms of storing and representing
information. That is why we find them almost everywhere, from a menu in
a restaurant, a bank statement, to professional databases. No surprise
then that we biologists love tables and we tend to predominantly use
Excel.</p>
<p>Excel is easy to use, flexible and powerful, however, it often gives
us too much freedom which leads to bad practices and difficult to re-use
data and metadata.</p>
<div id="exercise-1-what-can-go-wrong-with-data-in-excel-4-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-what-can-go-wrong-with-data-in-excel-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: What can go wrong with data in Excel (4 min)</h3>
<div class="callout-content">
<p>Have a look at the example <a href="../files/04-bad-metadata.xlsx">data-file</a> in Excel.</p>
<figure><img src="../fig/bad-metadata.png" alt="bad-metadata" class="figure mx-auto d-block"></figure><p>It contains data similar to the presented data before from
experiments on plants in different light conditions. Imagine you need to
work with this file.</p>
<ol style="list-style-type: decimal">
<li>What do you find confusing?</li>
<li>What would you try to clarify with the author before doing anything
with the file?</li>
<li>What will be the issues with calculation of: average biomas, biomas
per genotype?</li>
<li>Typically, more advanced data analysis is done programmatically,
which requires either conversion to text format as csv, tsv format or
using a library that reads Excel files and “kind of makes this
conversion on the fly”. Save this file in a text format, close Excel and
reopen the saved files. What has changed?</li>
</ol>
<p>Have you seen similar tables? Do you believe this example is
realistic and replicates real-life issues?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>This file hopefully unrealistically exacerbates typical bad practices
in Excel. 1+2. Some things that may be confusing:</p>
<ul>
<li>Why are there two tables, are the period measurement related to the
metabolics i.e. same samples?</li>
<li>Do colors in the period table have the same meaning? Seems no.</li>
<li>Why can row 22 be read, whilst row 13 says error?</li>
<li>What is the meaning of values in the media column?</li>
<li>Are the genotypes the same in different blocks or not?</li>
<li>What is the meaning behind bold text in the table?</li>
<li>What is the definition of the terms/why are units
missing/inconsistent?</li>
</ul>
<ol start="3" style="list-style-type: decimal"><li>
</li></ol>
<ul>
<li>Before averaging the biomas weight, they need to get converted the
same unit and the text needs to get replaced by the unit.</li>
<li>Averaging per genotype needs manual selection of suitable
entries</li>
</ul>
<ol start="4" style="list-style-type: decimal"><li>
</li></ol>
<ul>
<li>Information about light conditions is completely lost.</li>
<li>Header columns are scrambled.</li>
<li>The update date may change its meaning depending on the location
(switch year with day).</li>
</ul>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="common-spreadsheet-errors-25-min-teaching">Common Spreadsheet Errors (25 min teaching)<a class="anchor" aria-label="anchor" href="#common-spreadsheet-errors-25-min-teaching"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="tables">1. Using multiple tables<a class="anchor" aria-label="anchor" href="#tables"></a>
</h3>
<p>Multiple data tables within one spreadsheet confuse both human and
computers.</p>
<ul>
<li>For humans they prompt finding associations where they are not
present.</li>
<li>For computers each spreadsheet row typically is treated as one
observation. And you are potentially using the same field name in
multiple places.</li>
</ul>
<p>In our example both column A and K named <code>sample</code>
represent different information, and values in row 6 from both parts are
not related.</p>
</div>
<div class="section level3">
<h3 id="tabs">2. Using multiple tabs<a class="anchor" aria-label="anchor" href="#tabs"></a>
</h3>
<p>That seems like an easy way to organize data, right? Well, yes and
no.</p>
<ul>
<li>tabs make it more difficult to do programmatic analysis</li>
<li>tabs can be ignored (under the radar) when opening files, for
example the recently used tab is shown first, other user may not notice
that there are many other tabs before it (we have seen it a lot when
importing data)</li>
</ul>
<p>However,</p>
<ul>
<li>tabs seems like a perfect place to add administrative, descriptive
metadata so they are next to data while not “polluting” the data
table.</li>
<li>having secondary data next to primary in subsequent tabs permits
easy inspection of all without a need to always send a collection of
files.</li>
</ul>
<p>Our recommendation: use with caution; the more automatic analysis you
do, the less frequently you should use tabs.</p>
</div>
<div class="section level3">
<h3 id="zeros">3. Not filling in zeros<a class="anchor" aria-label="anchor" href="#zeros"></a>
</h3>
<p>It might be that when you’re measuring something, it’s usually a
zero, say the number of times a rabbit is observed in the survey. Why
bother writing in the number zero in that column, when it’s mostly
zeros?</p>
<p>However, there’s a difference between a zero and a blank cell in a
spreadsheet. To the computer, a zero is actually data. You measured or
counted it. A blank cell means that it wasn’t measured and the computer
will interpret it as an unknown value (otherwise known as a null
value).</p>
<p>The spreadsheets or statistical programs will likely mis-interpret
blank cells that you intend to be zeros.</p>
<p>Because of this, it’s very important to record zeros as zeros and
truly missing data as nulls.</p>
</div>
<div class="section level3">
<h3 id="null">4. Using problematic null values<a class="anchor" aria-label="anchor" href="#null"></a>
</h3>
<p>Example: using -999 or -1 or 0 (other numerical values) to represent
missing data.</p>
<p>Other times different null values are used to convey different
reasons why the data isn’t there. For example, -1 not recorded, -2
contamination etc.</p>
<p>This is important information to capture, but is in effect using one
column to capture two pieces (real values and comment or status). It
would be good here to create a new column like ‘data_missing’ and use
that column to capture the different reasons.</p>
<p>Whatever the reason, it’s a problem if unknown or missing data is
recorded as -999, 999, or 0. Many statistical programs will not
recognize that these are intended to represent missing (null)
values.</p>
<p>It is essential to use a clearly defined and consistent null
indicator. Blanks (most applications) and NA (for R) are good
choices.</p>
<p>White et al, 2013, explain good choices for indicating null values
for different software applications in their article: <a href="https://peerj.com/preprints/7/" class="external-link">Nine simple ways to make it easier
to (re)use your data.</a> Ideas in Ecology and Evolution.</p>
<figure><img src="../fig/null_values_table_1.jpg" alt="White et al." class="figure mx-auto d-block"></figure>
</div>
<div class="section level3">
<h3 id="formatting">5. Using formatting to convey information organizing data<a class="anchor" aria-label="anchor" href="#formatting"></a>
</h3>
<p>Never highlight cells, rows or columns that should be excluded from
an analysis, or to mark particular properties/conditions.</p>
<p>In our example file information about light conditions is only
encoded as a color. Formatting information is not available to analysis
software and almost certainly will be lost during processing.</p>
<p>All the information should be encoded as a field value, for example
in columns like: condition, calibrated, validated, omitted etc.</p>
<p>You can still use colors/fonts to help with readability (just make
sure they help and don’t distract) but no information should be lost if
data is exported to plain text.</p>
<p>If you are not careful, formatting a worksheet to be more
aesthetically pleasing can compromise your computer’s ability to see
associations in the data.</p>
<p><strong>Never merge cells, as it will make your data unreadable by
statistics software</strong>. Don’t leave blank rows to indicate
separations in data</p>
</div>
<div class="section level3">
<h3 id="units">6. Placing comments or units in cells<a class="anchor" aria-label="anchor" href="#units"></a>
</h3>
<p>Most analysis software can’t see Excel or LibreOffice comments, and
would be confused by comments placed within your data cells.</p>
<p>As described above for formatting, create another field if you need
to add notes to cells.</p>
<p>Similarly, don’t include units in cells: ideally, all the
measurements you place in one column should have the same unit, but if
for some reason they don’t, create another field and specify the unit. A
field value <code>0.123 g</code> will need to be parsed by a script into
<code>0.123</code> and <code>g</code> in order to be used for
calculations, don’t add extra work for others.</p>
</div>
<div class="section level3">
<h3 id="info">7. Entering more than one piece of information in a cell<a class="anchor" aria-label="anchor" href="#info"></a>
</h3>
<p>Don’t include more than one piece of information in a cell (like
<code>M30</code>, <code>F25</code>, for male age 30 and female 25, or
<code>Red light + Ampicilin + High nitrogen</code>).</p>
<p>This will limit the ways in which you can analyze your data. Design
your data sheet to include this information. For example, include one
column for age of individuals and a separate column for sex.</p>
</div>
<div class="section level3">
<h3 id="consistency">8. Inconsistency in used values<a class="anchor" aria-label="anchor" href="#consistency"></a>
</h3>
<p>Using “synonyms” or alternatives to describe the same values.</p>
<p>For example: <code>E. Coli</code>, <code>EColi</code>,
<code>Escherichia coli</code> in the same table</p>
</div>
<div class="section level3">
<h3 id="using-problematic-field-names-field_name">9. Using problematic field names {#field_name}<a class="anchor" aria-label="anchor" href="#using-problematic-field-names-field_name"></a>
</h3>
<p>Choose descriptive field names, but be careful not to include spaces,
numbers, or <strong>special characters</strong> of any kind (including
national characters: óęłńöüä). Spaces can be misinterpreted by parsers
that use whitespaces as delimiters and some programs don’t like field
names that are text strings starting with numbers (e.g. Excel).</p>
<p>Underscores (<code>_</code>) are a good alternative to spaces.
Consider writing names in camel case (like this: ExampleFileName) to
improve readability.</p>
<p>Remember that abbreviations that make sense at the moment may not be
so obvious in 6 months, but don’t overdo it with names that are
excessively long.</p>
<p>Including the units in the field names avoids confusion and enables
others to readily interpret your fields. Or add a separate row to help
automatic unit conversions.</p>
<p><strong>Examples</strong></p>
<table align="left" style="width =50%; border: 1px solid black;" class="table">
<tr>
<td>
<b>Good Name</b>
</td>
<br><td>
<b>Good Alternative </b>
</td>
<br><td>
<b>Avoid </b>
</td>
<br>
</tr>
<tr>
<td>
Max_temp_C
</td>
<td>
MaxTemp
</td>
<td>
Maximum Temp (°C)
</td>
</tr>
<tr>
<td>
Precipitation_mm
</td>
<td>
Precipitation
</td>
<td>
precmm
</td>
</tr>
<tr>
<td>
Mean_year_growth
</td>
<td>
MeanYearGrowth
</td>
<td>
Mean growth/year
</td>
</tr>
<tr>
<td>
sex
</td>
<td>
sex
</td>
<td>
M/F
</td>
</tr>
<tr>
<td>
weight
</td>
<td>
weight
</td>
<td>
w.
</td>
</tr>
<tr>
<td>
cell_type
</td>
<td>
CellType
</td>
<td>
Cell Type
</td>
</tr>
<tr>
<td>
Observation_01
</td>
<td>
first_observation
</td>
<td>
1st Obs
</td>
</tr>
</table>
</div>
<div class="section level3">
<h3 id="special">10. Using special characters in data<a class="anchor" aria-label="anchor" href="#special"></a>
</h3>
<p>For example, when writing longer text in a cell, people often include
line breaks, em-dashes, etc in their spreadsheet. Also, when copying
data from applications such as Word, formatting and ‘fancy’ non-standard
characters (such as left- and right-aligned quotation marks) are
included. When exporting this data into a coding/statistical environment
or into a relational database, dangerous things may occur, such as lines
being cut in half and encoding errors being thrown.</p>
<p>General best practice is to avoid adding characters such as new
lines, tabs, vertical tabs and sadly, national characters. In other
words, treat a text cell as if it were a simple web form that can only
contain text and spaces.</p>
</div>
<div class="section level3">
<h3 id="metadata">11. Values without field labels<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h3>
<p>It may be obvious for us that a cell
<code>Ann Smart and Daniele Hardwork</code> contains authors and
<code>Starch content in Arabidopis T. under different light conditions</code>
is a title, but this is not obvious for a computer program.</p>
<p>Always label values as the column header, or as a first field on the
left:</p>
<pre><code>Author  |  Ann Smart  |  Daniele Hardwork
Title   |  Starch content in Arabidopis T. under different light conditions</code></pre>
<div id="exercise-2-spotting-problems-4-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-spotting-problems-4-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Spotting problems (4 min)</h3>
<div class="callout-content">
<p>Look at the following rows and columns in the problematic table:</p>
<ul>
<li>Row 5</li>
<li>Row 2</li>
<li>Column C</li>
<li>Column E</li>
<li>Column L <img src="../fig/bad-metadata.png" alt="bad-metadata" class="figure">
</li>
</ul>
<p>which of the problems discussed above can you spot in these rows and
column? Here, we list them again:</p>
<ol style="list-style-type: decimal">
<li>Using multiple tables</li>
<li>Using multiple tabs</li>
<li>Not filling in zeros</li>
<li>Using problematic null values</li>
<li>Using formatting to convey information and organizing data</li>
<li>Placing comments or units in cells</li>
<li>Entering more than one piece of information in a cell</li>
<li>Inconsistency in used values</li>
<li>Using problematic field names</li>
<li>Using special characters in data</li>
<li>Values without field labels</li>
</ol>
<p>Type the problem number(s) next to the table elements</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ul>
<li>Row 5: 5, 9</li>
<li>Row 2: 11, 7</li>
<li>Column C: 8</li>
<li>Column E: 6, 5</li>
<li>Column L: 3</li>
</ul>
</div>
</div>
</div>
</div>
<div id="clean-data-tables-make-life-easier-6-min-teaching" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="clean-data-tables-make-life-easier-6-min-teaching" class="callout-inner">
<h3 class="callout-title">Clean data tables make life easier (6 min teaching)</h3>
<div class="callout-content">
<p>Let’s check the cleaned version of the previous <a href="../files/04-better-metadata.xlsx">file</a>.</p>
<p>Look how easy it is to calculate the average biomas, and if you have
Office365, see how easy it is to get average values per genotype.</p>
<p>Do you think it would take more effort to record data in a cleaned
way to start with, in comparison to the ‘bad example’ presented
before?</p>
<p>How long do you think it took to “clean” the original, problematic
data?</p>
</div>
</div>
</div>
<div id="exercise-3-outsmarted-by-excel-3-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-3-outsmarted-by-excel-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 3: Outsmarted by Excel (3 min)</h3>
<div class="callout-content">
<p>Excel has built in autoformatting functions which can make
biologist’s life more difficult</p>
<div class="section level2">
<h2 id="finding-the-right-date">Finding the right date<a class="anchor" aria-label="anchor" href="#finding-the-right-date"></a>
</h2>
<p>Open Excel and type following values into the cells:</p>
<table class="table">
<thead><tr class="header">
<th>A</th>
<th>B</th>
<th>C</th>
<th>D</th>
<th>E</th>
<th>F</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Gene</td>
<td>SEPT2</td>
<td>Sample</td>
<td>0013</td>
<td>Record</td>
<td>12/5/4</td>
</tr>
<tr class="even">
<td>Mar1</td>
<td>1 March</td>
<td>Mar-1</td>
<td>1-3</td>
<td>14/3/20</td>
<td>43904</td>
</tr>
</tbody>
</table>
<ul>
<li>Is what you see what you typed?</li>
<li>Can you force Excel to keep your formatting?</li>
<li>Do you know which year the dates represent?</li>
</ul>
</div>
</div>
</div>
</div>
<p>(9 min teaching)</p>
<p>A team of Australian researchers analyzed nearly 3,600 genetics
papers <a href="https://doi.org/10.1186/s13059-016-1044-7" class="external-link">Ziemann
2016</a>. As is common practice in the field, these papers all came with
supplementary files containing lists of genes used in research.</p>
<p>The Australian researchers found that roughly 1 in 5 of these papers
included errors in their gene lists that were due to Excel automatically
converting gene names to things like calendar dates or random
numbers.</p>
<p>Those errors actually forced the renaming of 27 gene symbols
including SEPT4 (now SEPTIN4) and MARCH1 (now MARCHF1).</p>
<p>Storing and handling dates is generally problematic even in
programming languages, as we tend to use dates in various formats and
those formats are region specific.</p>
<p>Have you ever got confused by a meeting date from an American
collaborator?</p>
<p>If the dates are “real” data, for example sample collection from a
patient or field measurement, preventing misinterpretations is crucial.
There are two safe options:</p>
<ol style="list-style-type: decimal">
<li>Store dates in 3 columns for year, month and day:</li>
</ol>
<table class="table">
<thead><tr class="header">
<th>Year</th>
<th>Month</th>
<th>Day</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>2021</td>
<td>3</td>
<td>16</td>
</tr>
<tr class="even">
<td>2021</td>
<td>10</td>
<td>1</td>
</tr>
</tbody>
</table>
<ol start="2" style="list-style-type: decimal">
<li>Store the date as an <a href="https://en.wikipedia.org/wiki/ISO_8601" class="external-link">ISO</a> string:
<code>YYYYMMDD</code> e.g. <code>20210316</code>
</li>
</ol>
<p>Even though, normally <code>YYYY-MM-DD</code> format is preferred, it
will be sadly reformatted by Excel according to your locale!</p>
<p>When using text files (.csv, .tsv), You should always document what
format you are using to represent dates.</p>
<div id="to-use-or-not-to-use-excel" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="to-use-or-not-to-use-excel" class="callout-inner">
<h3 class="callout-title">To use or not to use Excel</h3>
<div class="callout-content">
<p>The Excel file format <code>.xlsx</code> is now open, widely used and
supported by external libraries and thus it could be considered
interoperable. Nowadays it is admissible as being FAIR.</p>
<p>However, plain text files like comma or tab separated values (.csv,
.tsv) can be accessed without any special software. Data in a CSV file
can also easily be imported into other formats and environments, such as
SQLite and R. We are not tied to a certain version of a certain
expensive program when we work with CSV files, so it is a good format to
work with for maximum portability, interoperability and endurance. If
such files are handled only with text editors or programmatically (R,
Python) then they are a safer option as they prevent the autoformatting
issues described before.</p>
<p>If you analyse your data with R or Python, or you know that your data
are meant to be processed that way you should be using text formats
whenever possible, and as soon as you capture your data.</p>
<p>However, if you only use Excel and so does your community, just keep
using it. Just be aware of the possible pitfalls discussed, especially
when working with gene or protein names and accession numbers.</p>
</div>
</div>
</div>
<div id="cleaning-data-with-open-refine" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="cleaning-data-with-open-refine" class="callout-inner">
<h3 class="callout-title">Cleaning data with Open Refine</h3>
<div class="callout-content">
<p>There are tools that help you to clean and reorganize existing
data.</p>
<p><a href="https://openrefine.org/" class="external-link">OpenRefine</a> is a powerful tool
for working with messy data: cleaning it; transforming it from one
format into another; and extending it with web services and external
data.</p>
<p>With OpenRefine you can find and merge synonyms like:
<code>E. Coli</code>, <code>EColi</code>, <code>Escherichia coli</code>
into one, or split values in the <code>Name</code> field into
<code>FirstName</code> and <code>LastName</code>.</p>
<p>There is a carpentry course available: <a href="https://datacarpentry.org/OpenRefine-ecology-lesson/" class="external-link">Data
Cleaning with OpenRefine for Ecologists</a></p>
</div>
</div>
</div>
<div id="exercise-4-data-tables-and-fair-3-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-data-tables-and-fair-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 4: Data tables and FAIR (3 min)</h3>
<div class="callout-content">
<p>Which of the following statements is true/false (T or F):</p>
<ul>
<li>Do’s and don’ts help in programmatic re-use:</li>
<li>Avoiding multiple tabs improves interoperability:</li>
<li>Having accompanying README file with a table description is not
FAIR:</li>
<li>No ‘spaces’ in columns headers improve readability:</li>
<li>2022-07-15 is ISO date format:</li>
<li>20220715 date format is better for excel than 2022-07-15:</li>
<li>“No data” is better than leaving cell “blank” for missing data:</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<ul>
<li>Do’s and don’ts help in programmatic re-use: T</li>
<li>Avoiding multiple tabs improves interoperability: T</li>
<li>Having accompanying README file with a table description is not
FAIR: F</li>
<li>No ‘spaces’ in columns headers improve readability: F</li>
<li>2022-07-15 is ISO date format: T</li>
<li>20220715 date format is better for excel than 2022-07-15: T</li>
<li>“No data” is better than leaving cell “blank” for missing data:
F</li>
</ul>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Never use formatting to encode information</li>
<li>Include only one piece of information in a cell</li>
<li>It is easier to store data in the correct form than to clean data
for reuse</li>
</ul>
</div>
</div>
</div>
</div>
</section></section><section id="aio-08-laboratory-records"><p>Content from <a href="08-laboratory-records.html">Laboratory records</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/08-laboratory-records.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 70 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How and why to keep good quality records for experiments?</li>
<li>How to streamline and simplify record keeping and its maintenance
with electronic lab notebooks and online protocols?</li>
<li>How to keep records FAIR?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Use Benchling to create an electronic lab notebook entry with
electronic protocols</li>
<li>Reuse and modify an existing lab notebook entry, and export the
modified version</li>
<li>Use an existing protocol on protocols.io and adapt it to your own
needs</li>
<li>Find out where to track the history of a record in Benchling and
protocols.io</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="before-we-start">Before we start…<a class="anchor" aria-label="anchor" href="#before-we-start"></a>
</h2>
<hr class="half-width">
<p>Before we start this session on good record keeping, it might be a
good idea to make ourselves cup of tea. Here’s a peer-reviewed protocol
for making tea:
<img src="../../fig/06-handwritten-tea-protocol.jpg" alt="handwritten-tea-protocol" width="600" class="figure"><em>Figure credits: Ines Boehm and Ben Thomas</em></p>
</section><section><h2 class="section-heading" id="differences-between-analog-and-record-keeping">Differences between analog and record keeping<a class="anchor" aria-label="anchor" href="#differences-between-analog-and-record-keeping"></a>
</h2>
<hr class="half-width">
<p>How did you find making your cup of tea from the above protocol? As
you could see, one scientist put milk in before boiling water, another
one put boiling water in before milk. Another couldn’t find which type
of milk was used. Whilst the steps can be followed to produce a cup of
tea, the teas can vary wildly. Slight variations are acceptable, but
such as in an experiment, it is <em>more important that steps can be
repeated correctly</em> so that others can do the same.</p>
<p>Here is the same protocol typed out on <a href="https://dx.doi.org/10.17504/protocols.io.buhknt4w" class="external-link">protocol.io</a>.
Which do you find easier to follow?</p>
<p>Although digital protocols are the better of the two, analogue
protocols certainly have their place. For example, it is often easier to
make notes on analogue protocols which you can return to later.
Additionally, your working environment may make it too risky to bring
expensive technology (just in case), whereas pen and paper can be fine.
In cases like these, a hybrid system is often best - where analogue
notes are digitized. This can be the best of both worlds.</p>
<div id="exercise-1-differences-between-analog-and-digital-record-keeping-4-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-differences-between-analog-and-digital-record-keeping-4-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Differences between analog and digital record keeping (4 + 5 min)</h3>
<div class="callout-content">
<p>Compare the <a href="https://www.protocols.io/view/how-to-make-a-cup-of-tea-buhknt4w" class="external-link">electronic
version of the tea protocol</a> with the paper one from the photo:
<img src="../../fig/06-handwritten-tea-protocol.jpg" alt="handwritten-tea-protocol" width="400" class="figure"><em>Figure credits: Ines Boehm and Ben Thomas</em></p>
<p>What are advantages and disadvantages of traditional analog records
vs. digital records?</p>
<p>Try to find at least a handful of advantages and disadvantages for
each.</p>
<p>With all of these, which system do you think is most
advantageous?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p><strong>Advantages of traditional analog records</strong></p>
<ul>
<li>Ability to directly draw on your records</li>
<li>works regardless of internet/power access</li>
</ul>
<p><strong>Disadvantages of traditional analog records</strong></p>
<ul>
<li>can be lost and/or damaged (not Findable or Accessible)</li>
<li>only in one location at any time (not Findable or Accessible)</li>
<li>handwriting can make it less intelligible</li>
<li>harder to edit/move elements around smoothly (not
Interoperable)</li>
<li>can’t store most data types (e.g. imaging data) in a useable way
(not Reusable)</li>
</ul>
<p><strong>Advantages of digital records</strong></p>
<ul>
<li>Intelligible: can smoothly and easily move elements around to edit
it</li>
<li>Findable and Accessible: can be shared instantly anywhere around the
world, with anyone</li>
<li>Interoperable: can be easily commented on by anyone anywhere</li>
<li>doesn’t take up physical space (no record rooms/folders)</li>
<li>regular backups mean it won’t be lost</li>
<li>Reusable: version controls mean changes can easily be tracked</li>
<li>Reusable: can store protocols directly with other supporting data
types (e.g. video explanations)</li>
<li>can you think of more?</li>
</ul>
<p><strong>Disadvantages of digital records</strong></p>
<ul>
<li>dependent on internet access and power (not Accessible)</li>
<li>some digital record keeping services charge a fee</li>
<li>risk of corruption if data is not backed up (either yourself or by
the service used - not Reusable)</li>
</ul>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="why-do-we-need-to-keep-good-quality-records">Why do we need to keep good quality records?<a class="anchor" aria-label="anchor" href="#why-do-we-need-to-keep-good-quality-records"></a>
</h2>
<hr class="half-width">
<p>(19 min teaching)</p>
<p>Good scientific practice includes good record keeping, which ensures
not only transparency and reproducibility, but also accountability. One
prime example of why this is necessary is the recent data scandal
surrounding <a href="https://www.biopharmadive.com/news/novartis-gene-therapy-application-contained-manipulated-data-fda-says/560346/" class="external-link">Novartis’
FDA approved gene therapy</a> Zolgensma, for the fatal childhood motor
neuron disease Spinal Muscular Atrophy (the most expensive treatment
ever approved). Novartis submitted manipulated data, showing that the
comparison of two versions of Zolgensma in Phase 1 and Phase 3 testing
had similar therapeutic activity. How can we prevent the occurrence of
data manipulation such as this in the future, or how can we as a
research community implement practices to make it easier to find
manipulated records? FAIR record keeping, for example version control
can help, as it shows what changes have been made when in electronic
laboratory notebooks (ELNs), which will make it difficult to manipulate
results such as this without leaving a trace.</p>
<p>In order to avoid data mismanagement and such unexplained
discrepancies, it is imperative to keep dated, accurate, complete and
intelligible records of our experiments and the protocols we use. This
means they should include enough detail for others to reproduce under
ideally the same conditions. You are, legally (!!) the one responsible
for your records, not your colleague, or your PI.</p>
<div id="what-designates-a-good-record" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="what-designates-a-good-record" class="callout-inner">
<h3 class="callout-title">What designates a good record?</h3>
<div class="callout-content">
<p>Both protocols and laboratory records need to be detailed and kept
accurate and complete. They should be accessible (physically and/or
electronically) to others both short and long term. Regular back-ups on
a cloud and physical hard-drive are necessary to ensure appropriate
archiving. All your records should be kept in compliance with
departmental, institutional, and other regulatory requirements, with
special care given to human and animal research records. A few common
guidelines of good record keeping for protocols and laboratory notebooks
are the following:</p>
<p><strong>Protocols</strong></p>
<ul>
<li>who thought of the protocol if not you</li>
<li>complete and detailed instructions describing why and how to do an
experiment</li>
<li>what special materials and instruments are being used and where they
were obtained</li>
<li>health and safety advice and how to dispose of waste</li>
<li>allow repetition of your procedures and studies by yourself and
others</li>
</ul>
<p><strong>Laboratory Notebooks</strong></p>
<ul>
<li>contain all relevant details (what, when why and how you did
it)</li>
<li>who you are (the person creating the record)</li>
<li>what project(s) is the record part of</li>
<li>information on LOT numbers and batch numbers of cells and antibodies
used</li>
<li>what happened and what did not happen (data, including images)</li>
<li>how you manipulated and analysed the results</li>
<li>your interpretation (and the interpretations of others if important)
and next steps in the project based on these results</li>
<li>should be well organised for ease of navigation (indexed, labelled,
catalogued)</li>
<li>accurate and complete: include all irginal data and important study
details (metadata) and successful and unsuccessful studies</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="why-do-we-want-to-keep-fair-records">Why do we want to keep FAIR records?<a class="anchor" aria-label="anchor" href="#why-do-we-want-to-keep-fair-records"></a>
</h2>
<hr class="half-width">
<p>Focusing on FAIR this keeps our records Findable, Accessible,
Interoperable/Intelligible as well as Re-usable. Accessibility that
allows better reuse increases our citations and visibility in the field,
digital record keeping increases the legibility of notes and provenance
(tracking of dates and origins of work) allow for better reproducibility
which we have discussed in the previous lesson. Additionally, greater
accessibility affords accountability to the original creator of the
work. We will now show you how easy it is to share records once they are
online, and address some benefits that new repositories such as
electronic lab notebooks (ELNs) or online protocols have. There are
multiple repo’s for ELNs and online protocols. We will discuss two free
options that are easy to use: <a href="https://www.benchling.com/" class="external-link">Benchling</a> and <a href="https://www.protocols.io/welcome" class="external-link">procols.io</a> If you have not
created accounts yet for both of them, please do so now as you will need
them for the following exercises.</p>
</section><section><h2 class="section-heading" id="how-records-become-fair">How records become FAIR<a class="anchor" aria-label="anchor" href="#how-records-become-fair"></a>
</h2>
<hr class="half-width">
<div class="section level3">
<h3 id="electronic-lab-notebooks">Electronic lab notebooks<a class="anchor" aria-label="anchor" href="#electronic-lab-notebooks"></a>
</h3>
<p>By now you should be familiar with the concept of why digital record
keeping is important, as this is ultimately more reproducible. It is
self-explanatory that handwritten notes and laboratory notebook entries
take more time to search when looking for an entry. Electronic lab
notebooks allow organisation by project, and the search function, or
filters, can quickly find what we are looking for. We will now show how
to reuse a lab entry that has already been created by someone else.</p>
<div id="exercise-2-re-using-a-published-lab-entry-6-minutes" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-2-re-using-a-published-lab-entry-6-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 2: Re-using a published lab entry (6 minutes):</h3>
<div class="callout-content">
<p>To highlight how easy it is to reuse protocols someone else has used
in their lab entry, integrate these into your electronic lab notebooks,
and export these for e.g. printing we will be looking at the following
<a href="https://benchling.com/s/etr-SY8fi7L8ZIDSMCLCf92o" class="external-link">Benchling lab
entry, making Breakfast</a>.</p>
<div class="section level3">
<h3 id="re-use-a-published-lab-entry">Re-use a published lab entry<a class="anchor" aria-label="anchor" href="#re-use-a-published-lab-entry"></a>
</h3>
<ol style="list-style-type: decimal">
<li>First within your own workspace click the big ‘+’ (Create Project)
right next to Projects in your Benchling workspace</li>
<li>Call the project ‘Breakfast’, and add an appropriate description,
click ‘Create project’</li>
<li>Click on the above ‘Benchling lab entry’ link bringing you to the
public lab entry ‘Eggs Florentine in Portobello Mushrooms’.</li>
<li>Select the clock symbol on the right hand side underneath Share: Now
you can see the history of the entry and changes that have previously
been made to the document with a timestamp. If someone had tried to
‘manipulate’ data, you would be able to see this here. You also see the
owner of the document.</li>
<li>Click ‘Clone from version’</li>
<li>Select the ‘Breakfast’ folder to clone it to</li>
</ol>
</div>
</div>
</div>
</div>
<p>(2 min teaching)</p>
<p>With your newly set up lab entry, play around with it to explore the
interface. Add or remove some text, use the tool to embed a picture etc…
You can add text beside the image to ‘annotate’ this appropriately for
example. Explore the various things you can interact with to get an
understanding of the interface on Benchling.</p>
<div id="exercise-3-adapting-a-protocol-to-your-needs-4-minutes" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-3-adapting-a-protocol-to-your-needs-4-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 3: Adapting a protocol to your needs (4 minutes)</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>You have now accessed a digital record and want to reuse it to make
your own breakfast. To show how reusable digital recors are we will
first navigate through the cloned file you made in your project.</li>
<li>Navigate to your Project ‘Breakfast’, you can tell you are in your
Project, if your initials show in a red circle next to entries in the
side bar. You should see the lab entry ‘Eggs Florentine in Portobello
Mushrooms’, and the top bar above the title and toolbar should read
‘Tea’, ‘Portobello Mushrooms and Spinach’, ‘Poached Egg and Hollandaise
Sauce’, ‘Add Protocol’, ‘Notes’, and ‘Metadata’.</li>
<li>Click through those tabs and you will see that in your notes you
have your lab entry describing how breakfast was made with embedded
graphics and a shopping list and current prices. The other three tabs
describe the protocols that were used, and you can add additional
protocols with the ‘add protocol’ tab. We want you to adapt the ‘Tea’
protocol to suit your ingredients and methods.</li>
<li>Once you have made appropriate changes in the Tea protocol, you
should consider changing the order in which the breakfast and tea are
made.</li>
<li>Once you have made all suggested change have a look at the history
of the record (clock button), you can see the changes you have recently
made, and you can see it still relates to the original document. It
tells you what record it has been cloned from and when.</li>
<li>Click the link to the original record. As you can see digital record
keeping allows provenance, crediting the original author, but also
allowing you to keep track of your sources.</li>
<li>Navigate back to your lab entry in your project (your initials are a
sign that you are in the right place).</li>
</ol>
</div>
</div>
</div>
<p>You hopefully should have set your tea making protocol up so that
it’s just the way you like it. Moreover, it should have been easy to
make the changes to the protocol. Looking at the history of the record
allows you to see the original protocol from which you adapted, and
linked your adaptation.</p>
<div id="exercise-4-sharing-your-record-4-minutes" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-4-sharing-your-record-4-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 4: Sharing your record (4 minutes)</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Click the info icon on the right hand side underneath the clock
symbol you used previously and select ‘Export entry’</li>
<li>Your export is now running, you will receive an email when the
export is complete</li>
<li>Click the link in the email to download your protocol as a .zip</li>
<li>Unzip the file and in your own time, print the protocol if you want
to use the recipe in the kitchen, or share it with friends.</li>
<li>You can share .pdf versions, or click Share and generate a Share
link of your lab entry. This makes your record interoperable as many
users across many platforms across the world can access your entry if
you make it public and share it on for example social media. If there is
no IT access present, you always have the option to print the .pdf
copy.</li>
</ol>
</div>
</div>
</div>
<p>(2 min teaching)</p>
<p>Now that you have your export you can easily share it with others and
use it yourself elsewhere, while your digital record (and links to the
original record) are maintained online. You can share the PDF that you
have exported with others (or print it and store it the lab/workspace),
or link them to your digital record, allowing them to make further
changes if you wish.</p>
<p>Our Benchling tutorial: <a href="https://www.wiki.ed.ac.uk/display/RDMS/Benchling+%28quick%29+tutorial" class="external-link">BioRDM
Wiki</a></p>
<p><strong>Further resources and tutorials from Benchling:</strong></p>
<ul>
<li><a href="https://help.benchling.com/en/" class="external-link">Main help page, with access
to several tutorials</a></li>
<li><a href="https://help.benchling.com/en/collections/69523-molecular-biology" class="external-link">More
molecular biology features</a></li>
<li><a href="https://help.benchling.com/en/collections/1608962-benchling-training-kit-for-academics" class="external-link">Benchling
training kit for academics</a></li>
</ul>
</div>
<div class="section level3">
<h3 id="electronic-protocols">Electronic Protocols<a class="anchor" aria-label="anchor" href="#electronic-protocols"></a>
</h3>
<p>As you could see, Benchling has an integrated platform for protocols.
But there are other repositories, such as protocols.io that we briefly
mentioned beforehand, which have been developed to help make protocols
FAIR. One can publish protocols with a DOI which has many benefits that
we discussed in the previous lesson. Another strength is, that you can
create a protocol, fork the original protocol if your lab started
implementing small changes, and retain <strong>both</strong> versions
that you can cite in your publications or share with your
collaborators.</p>
<div id="exercise-5-adapt-a-public-protocol-and-retain-its-provenance-6-minutes" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-5-adapt-a-public-protocol-and-retain-its-provenance-6-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 5: Adapt a public protocol and retain its provenance (6 minutes)</h3>
<div class="callout-content">
<p>To copy protocols from protocols.io, edit them and export them as a
.pdf there are a few simple steps to follow. We will be using the <a href="https://dx.doi.org/10.17504/protocols.io.buhknt4w" class="external-link">making a cup of
tea protocol</a>.</p>
<div class="section level3">
<h3 id="fork-the-protocol-preserving-the-original-for-crediting">Fork the protocol, preserving the original for crediting<a class="anchor" aria-label="anchor" href="#fork-the-protocol-preserving-the-original-for-crediting"></a>
</h3>
<ol style="list-style-type: decimal">
<li>Open the link to the above protocol, as you can see we have assigned
it its own DOI</li>
<li>First click on Metrics: Because we are FAIR, this shows you how how
many views over time this protocol has had, how many exports, how many
individual steps it involves and how many times it has been forked.</li>
<li>Now click on the downwards arrow next to the title</li>
<li>Select ‘Copy/Fork’ and click ‘make a fork’</li>
<li>Select the Folder you want the protocol to be forked to and click
‘continue’</li>
<li>Your fork of “How to make a cup of tea” is ready now, click ‘edit
new fork’</li>
<li>On the right hand tool bar, the clock icon, shows you the history of
the protocol (as before in Benchling). Currently you should see no
history as you have not made changes.</li>
</ol>
</div>
<div class="section level3">
<h3 id="edit-the-forked-protocol">Edit the forked protocol<a class="anchor" aria-label="anchor" href="#edit-the-forked-protocol"></a>
</h3>
<ol style="list-style-type: decimal">
<li>Go to ‘Materials’ in the top tool-bar: add or edit materials
according to your preferences, e.g. change full-fat milk to oat-milk, or
add honey, lemon etc</li>
<li>Go to ‘Steps’ in the top tool-bar: edit the protocol according to
your preferences</li>
<li>You can edit the ‘Description’ and ‘Guidelines &amp; Warnings’ if
you would like to</li>
<li>As soon as you change anything, the timestamp and where in the
protocol this change was made appears in the history.</li>
<li>Click ‘View’, you will now see the reader view of your protocol. It
clearly states underneath the title ‘Forked from How to make a cup of
tea’ and the original protocol is linked. This allows clear
identification of your source.</li>
<li>Click ‘Edit’</li>
</ol>
</div>
<div class="section level3">
<h3 id="optional-export-the-forked-protocol">Optional: Export the forked protocol<a class="anchor" aria-label="anchor" href="#optional-export-the-forked-protocol"></a>
</h3>
<ol style="list-style-type: decimal">
<li>Click ‘More’ in the top tool-bar, select ‘Export’ &gt; ‘PDF’ &gt;
‘To your computer’ and click export (leave selections blank)</li>
</ol>
</div>
</div>
</div>
</div>
<p>(3 min teaching)</p>
<p>Now, if you go back to the original protocols.io protocol that you
have forked from, and click on metrics, you will see how that views have
increased. Additionally, if you click on ‘Forks’, it will show you that
there are a multitude of private forks and if you click on one of these
forks, e.g. ‘Cup of tea (polish way)’ you will be taken to the protocol.
A particular perk of this is accessibility and accompanying
accountability to the original creator of the work.</p>
</div>
</section><section><h2 class="section-heading" id="how-to-choose-the-right-platform">How to choose the right platform?<a class="anchor" aria-label="anchor" href="#how-to-choose-the-right-platform"></a>
</h2>
<hr class="half-width">
<p>There are more than a 100 platforms that provide services to host
electronic lab notebooks or protocols, therefore it can seem quite
daunting trying to find the right platform. There is some advice we can
offer when looking for the right service.</p>
<ul>
<li>Make sure they are in compliance with departmental, institutional,
and other regulatory and legal requirements (including where data is
geographically located, and which types of data can and cannot be
stored).</li>
<li>You want an acceptable pricing model (is it free?), check if your
institution has a subscription to any.</li>
<li>What is the longevity of the ELN (is it a brand new ELN, or has it
been established for a while now, i.e. is there a risk of the ELN
folding &amp; data being lost?).</li>
<li>What is the ability to share &amp; export entries, experience of
colleagues, availability of support, integration with other relevant
platforms (e.g. dropbox), and the potential for use with mobile devices
if required?</li>
<li>What is the user interface like? Does it feel intuitive or does it
take you days to find what you are looking for.</li>
<li>Check for operating system compatibility and real time
collaboration.</li>
<li>If you plan to maintain it locally it should be Open Source.</li>
</ul>
<p>The BioRDM team has put together a comprehensive summary of <a href="https://www.wiki.ed.ac.uk/pages/viewpage.action?pageId=463750271" class="external-link">ELNs
on the University of Edinburgh Wiki</a> where they test-ran a handful of
ELNs for you so you can make a more informed choice.</p>
<p>It is always best to use a free version or trial version to test an
ELN first and see which features you are missing and which ones you
prefer. Some often used ones are:</p>
<ul>
<li>Scinote (free version available)</li>
<li>Rspace (trial possible)</li>
<li>Benchling (free version available)</li>
<li>Labarchives (trial possible contacting the company)</li>
<li>WikiBench (works on top of Confluence)</li>
</ul>
<div id="exercise-6-eln-challenge-4-minutes" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-6-eln-challenge-4-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 6: ELN challenge (4 minutes)</h3>
<div class="callout-content">
<p>Do you use an ELNs? Which one? What features do you like?</p>
</div>
</div>
</div>
<p>(2 min teaching)</p>
</section><section><h2 class="section-heading" id="how-does-good-record-keeping-help-us-get-fair-ready">How does good record keeping help us get FAIR ready?<a class="anchor" aria-label="anchor" href="#how-does-good-record-keeping-help-us-get-fair-ready"></a>
</h2>
<hr class="half-width">
<p>Accessibility of your protocols online allows to share them with
collaborators who need them for a publication and a DOI of said
protocols allows to cite them with credibility. Electronic lab records
allow for easier re-usability and access of your data across multiple
platforms. Changes in your records can be traced back, therefore giving
accountability.</p>
<div id="fair-record-keeping---quiz-3-min-into-break" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="fair-record-keeping---quiz-3-min-into-break" class="callout-inner">
<h3 class="callout-title">FAIR record keeping - Quiz (3 min + into break)</h3>
<div class="callout-content">
<p>Which of the following statement are true (T)/false (F).</p>
<ul>
<li>Good record keeping ensures transparency and reproducibility.</li>
<li>There are no advantages to using analog record keeping when compared
to digital record keeping.</li>
<li>Digital records help people view a protocol simultaneously.</li>
<li>Digitally kept records can be quickly and easily edited.</li>
<li>On balance, digital record keeping is more advantageous than analog
record keeping.</li>
<li>Digital records are easier to search (for and within) than analog
records.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ul>
<li>Good record keeping ensures transparency and reproducibility.
(T)</li>
<li>There are no advantages to using analog record keeping when compared
to digital record keeping. (F)</li>
<li>Digital records help people view a protocol simultaneously. (T)</li>
<li>Digitally kept records can be quickly and easily edited. (T)</li>
<li>On balance, digital record keeping is more advantageous than analog
record keeping. (T)</li>
<li>Digital records are easier to search than analog records. (T)</li>
</ul>
</div>
</div>
</div>
</div>
<div id="further-reading" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="further-reading" class="callout-inner">
<h3 class="callout-title">Further Reading</h3>
<div class="callout-content">
<p><a href="https://www.scinote.net/electronic-lab-notebook-guide/" class="external-link">ELN
Guide</a><br><a href="https://www.wiki.ed.ac.uk/pages/viewpage.action?pageId=463750271" class="external-link">BioRDM
- ELN resources</a></p>
</div>
</div>
</div>
<div id="for-instructors-advanced-teaching" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="for-instructors-advanced-teaching" class="callout-inner">
<h3 class="callout-title">For instructors: Advanced teaching</h3>
<div class="callout-content">
<p>There is also so-called “Laboratory Information Management Systems”
(LIMS) which allow to register barcoded samples in laboratories
(e.g. covid tests). This allows easier management of workflows, result
input and tracking of samples. These systems provide enhanced levels of
record keeping and researchers might encounter them in particular in
clinical research settings.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Good record keeping ensures transparency and reproducibility.</li>
<li>Record keeping is an integral part of data FAIRification.</li>
<li>Record keeping is key to good data management practices.</li>
<li>Having experimental details already in electronic form makes it
easier to include them in READMEs or repository records</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-09-files-organization"><p>Content from <a href="09-files-organization.html">Working with files</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/09-files-organization.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 60 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How should I name my files?</li>
<li>How does folder organisation help me</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Understand elements of good naming strategy</li>
<li>Evaluate pros and cons of different project organizations</li>
<li>Explain how file management helps in being FAIR</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="project-organization-planning-file-names-and-folder-structure">Project organization: planning file names and folder structure<a class="anchor" aria-label="anchor" href="#project-organization-planning-file-names-and-folder-structure"></a>
</h2>
<hr class="half-width">
<p>(3 min teaching) Before you even start collecting or working with
data you should decide how to structure and name files and folders. This
will:</p>
<ul>
<li>allow for standardized data collection and analysis by many team
members.</li>
<li>make it easier for the researcher to determine where files should be
saved.</li>
<li>avoid file duplication.</li>
<li>help make retrieval and archiving more efficient.</li>
</ul>
<p><img src="../fig/07-intro_folder_structure.png" alt="Intro to folder structure" class="figure"><em>Figure credits: Andrés
Romanowski</em></p>
<p>Consistent naming and organization of files in folders has two main
goals:</p>
<ul>
<li>quick to find files</li>
<li>ability to tell the file content without opening it</li>
</ul></section><section><h2 class="section-heading" id="naming-your-files-and-folders">Naming your files (and folders)<a class="anchor" aria-label="anchor" href="#naming-your-files-and-folders"></a>
</h2>
<hr class="half-width">
<p>One important and often overlooked practice of organizing, sharing,
and keeping track of data files is standardized naming.<br>
It is important to develop naming conventions which permit the encoding
of experimental factors which are important to the project.</p>
<p>File (folder) names should be consistent, meaningful to you and your
collaborators, allow you to easily find what you are looking for, give
you a sense of the content without opening the file, and allow you to
easily identify if something is missing.</p>
<blockquote>
<p>Example</p>
<p><code>LD_phyA_off_t04_2020-08-12.norm.xlsx</code></p>
<p>could be a file that contains:</p>
<ul>
<li>normalized data (norm),</li>
<li>from experiment in long day (LD)</li>
<li>for genotype phyA</li>
<li>with media without sucrose (off)</li>
<li>at timepoint 4 (t04)</li>
</ul>
</blockquote>
<div id="exercise-1-naming-and-sorting-12-min---3-min-explanation" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-naming-and-sorting-12-min---3-min-explanation" class="callout-inner">
<h3 class="callout-title">Exercise 1: Naming and sorting (12 min - 3 min explanation)</h3>
<div class="callout-content">
<p>Have a look at the example files from a project, similar to the one
from the previous metadata episode.</p>
<p>All the files have been sorted by name and demonstrate consequences
of different naming strategies.</p>
<p>For your information, to encode experimental details the following
conventions were taken:</p>
<ul>
<li>phyB/phyA are sample genotypes</li>
<li>sXX is the sample number</li>
<li>LD/SD are different light conditions (long or short day)</li>
<li>on/off are different media (on sucrose, off sucrose)</li>
<li>measurement date</li>
<li>other details are timepoint and raw or normalized data</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<pre><code>2020-07-14_s12_phyB_on_SD_t04.raw.xlsx
2020-07-14_s1_phyA_on_LD_t05.raw.xlsx
2020-07-14_s2_phyB_on_SD_t11.raw.xlsx
2020-08-12_s03_phyA_on_LD_t03.raw.xlsx
2020-08-12_s12_phyB_on_LD_t01.raw.xlsx
2020-08-13_s01_phyB_on_SD_t02.raw.xlsx
2020-7-12_s2_phyB_on_SD_t01.raw.xlsx
AUG-13_phyB_on_LD_s1_t11.raw.xlsx
JUL-31_phyB_on_LD_s1_t03.raw.xlsx
LD_phyA_off_t04_2020-08-12.norm.xlsx
LD_phyA_on_t04_2020-07-14.norm.xlsx
LD_phyB_off_t04_2020-08-12.norm.xlsx
LD_phyB_on_t04_2020-07-14.norm.xlsx
SD_phyB_off_t04_2020-08-13.norm.xlsx
SD_phyB_on_t04_2020-07-12.norm.xlsx
SD_phya_off_t04_2020-08-13.norm.xlsx
SD_phya_ons_t04_2020-07-12.norm.xlsx
ld_phyA_ons_t04_2020-08-12.norm.xlsx  </code></pre>
</div>
</div>
</div>
</div>
<div id="exercise-1-naming-and-sorting-12-min---3-min-explanation" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-1-naming-and-sorting-12-min---3-min-explanation" class="callout-inner">
<h3 class="callout-title">Exercise 1: Naming and sorting (12 min - 3 min
explanation)<em> (continued)</em>
</h3>
<div class="callout-content">
<ul>
<li>What are the problems with having the date first?</li>
<li>How do different date formats behave once sorted?</li>
<li>Can you tell the importance of a leading 0 (zeros)?</li>
<li>Is it equally easy to find all data from LD conditions as ON
media?</li>
<li>Can you spot the problem when using different cases
(upper/lower)?</li>
<li>Do you see benefits of keeping consistent lengths of the naming
conventions?</li>
<li>Do you see what happens when you mix conventions?</li>
</ul>
<blockquote>
<h2 id="solution">Solution</h2>
<ul>
<li>Using dates up front makes it difficult to quickly find data for
particular conditions or genotypes. It also masks the “logical” order of
samples or timepoints.</li>
<li>Named months break the “expected” sorting, same as dates without
leading 0</li>
<li>Without leading zeros, ‘s12’ appear before s1 and s2</li>
<li>the first (and second) part of the name are easiest to spot</li>
<li>the last file is also from LD conditions, but appears after SD, same
with ‘phya’ genotypes</li>
<li>the last 3 file names are easiest to read as all parts appear on top
of each other due to the same 3 letter-length codes ons and off</li>
<li>The lack of consistency makes it very difficult to get data from
related samples/conditions.</li>
</ul>
</blockquote>
</div>
</div>
</div>
<p><br></p>
<p>(4 min teaching)</p>
<p>Some things to take into account to decide on your naming convention
are:</p>
<ul>
<li>Does your convention make your files easy to sort and find (e.g. by
important features)?</li>
<li>Include parameters that are as descriptive as possible (i.e.:
project, experiment, researcher, sample, organism, date/range, data
type, method).</li>
<li>Defined a standard vocabulary (shortcuts) for parameters and
document any abbreviation.</li>
<li>Decide which elements go in which order.</li>
<li>Decide the convention when to use symbols, capitals, hyphens (e.g
kebab-case, CamelCase, or snake_case).</li>
<li>Define a maximum name length. Aim for filenames no longer than ~30
characters.</li>
</ul>
<p><strong>Do’s:</strong></p>
<ul>
<li>for dates use the YYYY-MM-DD standard and place at the end of the
file UNLESS you need to organize your files chronologically</li>
<li>include version number (if applicable), use leading zeroes (i.e.:
v005 instead of v5).</li>
<li>make sure the 3-letter file format extension is present at the end
of the name (e.g. .doc, .xls, .mov, .tif)</li>
<li>add a <strong>PROJECT_STRUCTURE</strong> (README) file in your top
directory which details your naming convention, directory structure and
abbreviations</li>
</ul>
<p><strong>Don’ts:</strong></p>
<ul>
<li>avoid using spaces (use _ or - instead)</li>
<li>avoid dots, commas and special characters (e.g. ~ ! @ # $ % ^ &amp;
* ( ) ` ; &lt; &gt; ? , [ ] { } ’ “)</li>
<li>avoid using language specific characters (e.g óężé), unfortunately
they still cause problems with most software or between operating
systems (OS)</li>
<li>avoid long names</li>
<li>avoid repetition, e.g if directory name is
<em>Electron_Microscopy_Images</em>, and file
<em>ELN_MI_IMG_20200101.img</em> then ELN_MI_IMG is redundant</li>
<li>avoid deep paths with long names (i.e. deeply nested folders with
long names) as archiving or moving between OS may fail</li>
</ul>
<div id="exercise-2-a-good-name-3-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-a-good-name-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: A good name (3 min)</h3>
<div class="callout-content">
<p>Select which file options adhere the best to the presented
recommendations:</p>
<ol style="list-style-type: decimal"><li>
</li></ol>
<ol style="list-style-type: lower-alpha">
<li>analysis-20210906.xlsx<br>
</li>
<li>rna-levels-by-site.v002.xlsx<br>
</li>
<li>analysis of rna levels from 5Aug2021.xlsx</li>
</ol>
<ol start="2" style="list-style-type: decimal"><li>
</li></ol>
<ol style="list-style-type: lower-alpha">
<li>20210906-birds-count-EDI.csv<br>
</li>
<li>birds.csv<br>
</li>
<li>birds-count&amp;diversity EDI 2021-09-06.csv</li>
</ol>
<ol start="3" style="list-style-type: decimal"><li>
</li></ol>
<ol style="list-style-type: lower-alpha">
<li>2020-7-12_s2_phyB_+_SD_t01.raw.xlsx<br>
</li>
<li>ld_phyA_on_s02-t01_2020-07-12.norm.xlsx<br>
</li>
<li>ld_phya_ons_02-01_2020-07-12.norm.xlsx</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ul>
<li>1 b)</li>
<li>2 a)</li>
<li>3 b)</li>
</ul>
</div>
</div>
</div>
</div>
<p>If adding all the relevant details to file names makes them too long,
it is often a signal that you should use folders to organize the files
and capture some of those parameters.</p>
<div id="exercise-3-folders-vs-files-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-folders-vs-files-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 3: Folders vs Files (5 min)</h3>
<div class="callout-content">
<p>Have a look as these two different organization strategies:</p>
<ol style="list-style-type: decimal">
<li><p>|– Project<br>
|– |– arab_LD_phyA_off_t04_2020-08-12.metab.xlsx</p></li>
<li><p>|– Project<br>
|– |– arabidopsis<br>
|– |– |– long_day<br>
|– |– |– |– phyA<br>
|– |– |– |– |– off_sucrose_2020-08-12<br>
|– |– |– |– |– |– t04.metab.xlsx</p></li>
</ol>
<p>Can you think of scenarios in which one is better suited than other?
Hint: think of other files that could be present as well.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>The first strategies, can work very well if the project has only few
files, so all of them can quickly be accessed (no need to change
folders) and the different parameters are easily visible. For example a
couple of conditions, couple of genotypes or species</p>
<p>|– Project<br>
|– |– arab_LD_phyA_off_t04_2020-08-12.metab.xlsx<br>
|– |– arab_LD_WILD_off_t03_2020-08-11.metab.xlsx<br>
|– |– arab_SD_phyA_off_t01_2020-05-12.metab.xlsx<br>
|– |– arab_SD_WILD_off_t02_2020-05-11.metab.xlsx<br>
|– |– rice_LD_phyA_off_t05_2020-05-02.metab.xlsx<br>
|– |– rice_LD_WILD_off_t06_2020-05-02.metab.xlsx<br>
|– |– rice_SD_phyA_off_t07_2020-06-02.metab.xlsx<br>
|– |– rice_SD_WILD_off_t08_2020-06-02.metab.xlsx</p>
<p>The second strategy works better if we have a lot of individual files
for each parameter. For example, imagine the metabolites are measured
hourly throughout the day, and there are ten different genotypes, two
species and 4 light conditions. You would not want to have all the 2000
files in one folder.</p>
<p>|– Project<br>
|– |– arabidopsis<br>
|– |– |– long_day<br>
|– |– |– |– phyA<br>
|– |– |– |– |– off_sucrose_2020-08-12<br>
|– |– |– |– |– |– t01.metab.xlsx<br>
|– |– |– |– |– |– t02.metab.xlsx<br>
|– |– |– |– |– |– t03.metab.xlsx<br>
|– |– |– |– |– |– …<br>
|– |– |– |– |– |– t23.metab.xlsx<br>
|– |– |– |– |– |– t24.metab.xlsx<br>
|– |– rice<br>
|– |– |– long_day<br>
|– |– |– |– phyA<br>
|– |– |– |– |– off_sucrose_2020-06-03<br>
|– |– |– |– |– |– t01.metab.xlsx<br>
|– |– |– |– |– |– …<br>
|– |– |– |– |– |– t24.metab.xlsx</p>
</div>
</div>
</div>
</div>
<div id="must-do-document-your-strategy" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="must-do-document-your-strategy" class="callout-inner">
<h3 class="callout-title">Must do: Document your strategy</h3>
<div class="callout-content">
<p>Regardless of whether you are using long filenames or incorporating
some of the variables within the folder structure, document it!<br>
Always include a PROJECT_STRUCTURE (or README) file describing your file
naming and folder organisation conventions.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="strategies-to-set-up-a-clear-folder-structure">Strategies to set up a clear folder structure<a class="anchor" aria-label="anchor" href="#strategies-to-set-up-a-clear-folder-structure"></a>
</h2>
<hr class="half-width">
<p>(3 min teaching) Establishing a system that allows you to access your
files, avoid duplication and ensure that your data can be easily found
needs planning.</p>
<p>You can start by developing a logical folder structure. To do so, you
need to take into account the following suggestions:</p>
<ul>
<li>Use folders to group related files. A single folder will make it
easy to locate them.</li>
<li>Name folders appropriately: use descriptive names after the areas of
work to which they relate.</li>
<li>Structure folders hierarchically: use broader topics for your main
folders and increase in specificity as you go down the hierarchy.</li>
<li>Be consistent: agree on a naming convention from the outset of your
research project.</li>
</ul>
<div id="exercise-4-typical-folder-organizations-7-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-typical-folder-organizations-7-min" class="callout-inner">
<h3 class="callout-title">Exercise 4: Typical folder organizations (7 min)</h3>
<div class="callout-content">
<p>Have a look at the four different folder structures.
<img src="../../fig/07-file_organisation.png" alt="file-organisation-strategies" width="600" class="figure"><em>Figure credits: Ines Boehm</em></p>
<p>The first two: a) b) are recommended for computing, the other two: c)
d) are for more wet/biological projects.</p>
<ul>
<li>Which one is the most similar to your project structure?</li>
<li>When/why would you use a) and when/why b)</li>
<li>When/why would you use c) and when/why d)</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>Firstly, the root directory contains a README file that provides an
overview of the project as a whole, a CITATION file that explains how to
reference it and a LICENSE, all three make it
<strong>REUSABLE</strong>.</p>
<p>The <strong>a)</strong> structure is recommended by the <a href="https://doi.org/10.1371/journal.pcbi.1005510" class="external-link">Good enough
practices in scientific computing</a> paper.<br>
This project structure clearly separates the inputs (the raw data) from
the outputs (the results) and the analysis procedure (python code).
Following the same convention (like src folder for code) makes it easy
to find interesting elements, for example the raw data or particular
plotting procedure. Good for sharing analysis project, also for
pipelines where one set of inputs generated the set of outputs in the
step by step manner.</p>
<p>The <strong>b)</strong> structure is called “Organized by analysis”
or “by figure”. As the name suggest it may be recommended to share data
underling a publication. In that way each paper figure is represented by
its raw data, processing scripts and the final results and figure plots.
It is also well suited if each analysis deals with different data type
or different aspect of it. When compared to ‘a)’ it makes easier to
match the required inputs with the computational procedures.</p>
<p>The structure similar to c) is recommended for Brain Imaging Data
Structure <a href="https://doi.org/10.1038/sdata.2016.44" class="external-link">BIDS</a>, as
it is organized by “patient” (in this case patient was replaced by pig
:) ) and type of scans. Here the focus is on individual subject /
samples, for which various data was obtained.</p>
<p>Structured <strong>d)</strong> is useful when we are interested in
outcomes of experimental conditions (here drug treatments). The same set
of samples/subjects/genotypes are exposed to different experimental
variables/conditions and the data are probably compared between all the
samples at the same conditions.</p>
</div>
</div>
</div>
</div>
<div id="optional-good-enough-practices-for-scientific-computing-recommendations" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="optional-good-enough-practices-for-scientific-computing-recommendations" class="callout-inner">
<h3 class="callout-title">(Optional) Good enough practices for scientific computing recommendations</h3>
<div class="callout-content">
<p>The <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510#sec009" class="external-link">Good
enough practices in scientific computing</a> paper makes the following
simple recommendations:</p>
<ul>
<li>Put each project in its own directory, which is named after the
project</li>
<li>Put text documents associated with the project in the ‘doc’
directory</li>
<li>Put raw data and metadata in a ‘data’ directory</li>
<li>Put files generated during cleanup and analysis in a ‘results’
directory</li>
<li>Put project source code in the ‘src’ directory</li>
<li>Put compiled programs in the ‘bin’ directory</li>
<li>Name all files to reflect their content or function:</li>
<li>Use names such as ‘bird_count_table.csv’, ‘notebook.md’, or
‘summarized_results.csv’.</li>
<li>Do not use sequential numbers (e.g., result1.csv, result2.csv) or a
location in a final manuscript (e.g., fig_3_a.png), since those numbers
will almost certainly change as the project evolves.</li>
</ul>
</div>
</div>
</div>
<!---
Replaced by folders comparison as it was not working
> ## Organization for computing (3 minutes)
> Take a look at the folder structure recommended by the
> [Good enough practices in scientific computing](https://doi.org/10.1371/journal.pcbi.1005510) paper.
>
> Why do you think this layout is *recommended* and
> suited for a computing project?
> >
> > .
> > |-- CITATION
> > |-- README
> > |-- LICENSE
> > |-- requirements.txt
> > |
> > |-- data
> > |   |-- birds_count_table.csv
> > |
> > |-- doc
> > |   |-- notebook.md
> > |   |-- manuscript.md
> > |   |-- changelog.txt
> > |
> > |-- results
> > |   |-- summarized_results.csv
> > |
> > |-- src
> > |   |-- sightings_analysis.py
> > |   |-- runall.py
> > |
> >
> {: .source}
> > ## Solution
> > This project structure clearly separates the inputs (the raw data)
> > from the outputs (the results) and the analysis procedure (python code).
> > Following the same convention (like src folder for code) makes it easy
> > to find interesting elements, for example the raw data or particular plotting procedure.
> >
> > The root directory contains a README file that provides an overview of the project as a whole,
> > a CITATION file that explains how to reference it and a LICENSE, all three make it **REUSABLE**.
> > The src directory contains a controller script runall.py that loads the data and triggers the whole analysis.
> {: .solution}
{: .challenge}
-->
<p>(5 min teaching)</p>
<div id="after-you-have-a-plan" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="after-you-have-a-plan" class="callout-inner">
<h3 class="callout-title">After you have a plan</h3>
<div class="callout-content">
<p>Your naming conventions might need some adjustments as the project
progresses. Don’t despair, just document it!</p>
<p>If you change the strategy document it in PROJECT_STRUCTURE (or
README) stating why you made the change and when. Update the locations
and names of files which followed the old convention</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="backing-up-your-project-files-and-folders">Backing up your project files and folders<a class="anchor" aria-label="anchor" href="#backing-up-your-project-files-and-folders"></a>
</h2>
<hr class="half-width">
<ul>
<li>Back up (almost) everything created by a human being or recorded by
a machine as soon as it is created.</li>
<li>Always backup your files in 3 places, at least one should be
off-site.</li>
<li>USB sticks are a failure-prone option and are not a valid solution
for backup of scientific data</li>
<li>A robust backup cannot be achieved manually</li>
</ul>
<p>Do you know how and where to keep 3 copies of your data which are
always up to date?</p>
<p>Secure data preservation is very difficult to achieve without
institutional support and know-how. One option is cloud storage, but not
all data may be put in a public cloud.</p>
<p>You should always check your institutional guidelines and what
solutions are available in your organisation.</p>
</section><section><h2 class="section-heading" id="project-file-organization-and-fair-guidelines">Project file organization and FAIR guidelines<a class="anchor" aria-label="anchor" href="#project-file-organization-and-fair-guidelines"></a>
</h2>
<hr class="half-width">
<div id="exercise-5-fair-files-3-minutes" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-5-fair-files-3-minutes" class="callout-inner">
<h3 class="callout-title">Exercise 5: FAIR Files (3 minutes)</h3>
<div class="callout-content">
<p>Choose 3 main benefits of a good strategy for folder organisation and
naming conventions.</p>
<ul>
<li>Makes data more findable</li>
<li>Aids in making data more reproducible - projects can be copied
easily</li>
<li>Raw data can be reanalysed multiple times</li>
<li>Naming conventions can be read automatically</li>
<li>Easy to understand content by name, less misunderstandings</li>
<li>Easier to find and share data with others</li>
<li>Easy inspection of the project progress (present files)</li>
<li>Fewer meetings required when sharing data</li>
<li>Time saving</li>
</ul>
</div>
</div>
</div>
<p>Have you realised that the above suggestions mean including valuable
metadata as part of your folder structure and file names?</p>
<div id="where-to-next" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="where-to-next" class="callout-inner">
<h3 class="callout-title">Where to next</h3>
<div class="callout-content">
<p>Bulk renaming of files can be done with the software such as Ant
Renamer, RenameIT or Rename4Mac.</p>
<p><a href="https://doi.org/10.1371/journal.pcbi.1005510" class="external-link">Good enough
practices in scientific computing (Wilson et al., 2017)</a></p>
</div>
</div>
</div>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was created using the following references as
inspiration:</p>
<ul>
<li><a href="https://doi.org/10.1371/journal.pcbi.1005510" class="external-link">Good enough
practices in scientific computing (Wilson et al., 2017)</a></li>
<li><a href="https://www.data.cam.ac.uk/data-management-guide/organising-your-data" class="external-link">Organising
your data</a></li>
<li><a href="https://www.wur.nl/en/Value-Creation-Cooperation/Collaborating-with-WUR-1/Organising-files-and-folders.htm" class="external-link">Organising
files and folders</a></li>
<li><a href="https://libguides.princeton.edu/c.php?g=102546&amp;p=930626" class="external-link">File
naming</a></li>
<li><a href="https://librarycarpentry.org/lc-fair-research/05-reusable/index.html" class="external-link">Library
Caprenty FAIR Data</a></li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>A good file name suggests the file content</li>
<li>Good project organization saves you time</li>
<li>Describe your files organization in PROJECT_STRUCTURE or README
including naming convention</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-10-reusable-analysis"><p>Content from <a href="10-reusable-analysis.html">Reusable analysis</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/10-reusable-analysis.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 90 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How keep track of your data analysis procedure?</li>
<li>How to make reproducible plots?</li>
<li>What are Jupyter notebooks</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Recognise benefits of notebooks for analysis</li>
<li>Use a jupyter notebook and modify it</li>
<li>Understand notebooks role in being FAIR</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="reusable-analysis">Reusable Analysis<a class="anchor" aria-label="anchor" href="#reusable-analysis"></a>
</h2>
<hr class="half-width">
<p>(~50 min teaching)</p>
<p>Data analysis and research are an iterative process. Experimental
measurements and their analysis will guide next steps within research,
e.g. identification of new targets for drug discovery, the invalidation
of a target would likely make us switch to a different target for target
validation. Or in large data, or omics experiments data is initially
explored, analysis method will be adjusted and modified, and so will
experimental procedures if adequate results have not been achieved.</p>
<p>This so called ad-hoc analysis will only after several iterations of
data exploration, retries, adjustments, and modifications to
experimental procedures lead to selection of suitable analysis methods
and parameters. One would then conduct statistical validation of results
and generate final results and graphs for publications. This process can
be difficult to track using conventional methods such as pen + paper and
Excel.</p>
</section><section><h2 class="section-heading" id="reusable-computing">Reusable Computing<a class="anchor" aria-label="anchor" href="#reusable-computing"></a>
</h2>
<hr class="half-width">
<p>We have previously discussed electronic lab notebooks and their
benefit in being FAIR. If you are working with large datasets,
mathematical models, complex visualisations of your data then you might
already be frustrated by having to copy and paste your figures into your
electronic lab notebook after each iteration of new code (or generation
in Excel). You might also have lost track of which code corresponded to
which figure and why you changed your code last time.</p>
<p>There is a simple solution to this: <strong>Computational
notebooks</strong></p>
</section><section><h2 class="section-heading" id="computational-notebooks---jupyter-notebook-for-fair-practices">Computational Notebooks - Jupyter Notebook for FAIR practices<a class="anchor" aria-label="anchor" href="#computational-notebooks---jupyter-notebook-for-fair-practices"></a>
</h2>
<hr class="half-width">
<p>Computational notebooks are essentially laboratory notebooks for
scientific computing. Instead of pasting DNA gels alongside lab
protocols, researchers embed code, data and text to document their
computational methods. A free and popular tool that is being used across
disciplines is <strong>the Jupyter notebook</strong>.</p>
<p><a href="../www.jupyter.org">Jupyter</a> Notebooks are interactive web
applications which allow you to type and edit lines of code in the
programming languages July (Ju), Python (Py), or R, hence it’s name
Jupyter, and view the output (e.g. graphs or calculations) immediately.
The huge benefit of such notebooks is that source code is mixed with
documentation, thus explaining experimental setup, analysis, results of
tables and plots throughout.</p>
<p>What you need to do is write the code that will analyse our data, and
explain the narrative behind the experiment and interpretation of the
results. For data scientists, this format can drive exploration via
interactive computing. This is an environment in which users execute
code, see what happens, modify and repeat in an iterative process
between researcher and data, in an ad-hoc way. As a basic principle
Jupyter Notebooks run on a ‘kernel’ which is responsible in execution of
the code. Generally this ‘kernel’ can run on your computer or you can
use external servers to store and analyse your data.</p>
<p>Notebooks can also be exported as .pdf and .html files which allow
easy sharing of outputs, and other tools such as <a href="https://nbviewer.org/" class="external-link">nbviewer</a>, an open-source service that
allows users to render their Jupyter notebooks on GitHub in a web
browser without having to install the software or any programming
libraries.</p>
<p>To show you how easy it is to work with Jupyter Notebooks, we have
created an exercise for you where we will work on “real-life” data from
an experiment looking at circadian influence on plants following
short-day and long-day light exposure. We will create some graphs and
test whether there are differences between our genotypes of arabidopsis
using the language R as an example.</p>
<div id="testimonial1" class="callout testimonial">
<div class="callout-square">
<i class="callout-icon" data-feather="heart"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Testimonial</h3>
<div class="callout-content">
<p>The example notebook and data files can be found in instructors
folders.<br>
The most recent version are available via a separated github
project:<br><a href="https://github.com/BioRDM/fair-jupyter" class="external-link">fair-jupyter</a></p>
</div>
</div>
</div>
<div id="exercise-1-basics-of-jupyter-notebooks-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-basics-of-jupyter-notebooks-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: basics of Jupyter notebooks (5 min)</h3>
<div class="callout-content">
<p>Navigate to the jupyter server, we will first show you how to
duplicate a notebook and save it and how to run the code:</p>
<ol style="list-style-type: decimal">
<li><p>Select the notebook titled
‘student_notebook_light_conditions.ipynb’ as depicted below and click
‘Duplicate’. Confirm with Duplicate when you are asked if you are
certain that you want to duplicate the notebook. <img src="../fig/10-01-duplicate_notebook.png" alt="Figure 1. Duplicate a Jupyter notebook" class="figure"><em>Figure 1. Duplicate a
Jupyter notebook</em></p></li>
<li><p>A copy of the notebook has appeared with the suffix ‘-Copy’ and a
number (<strong>Figure 2a</strong>), select this notebook. Have a look
around the notebook and explore its anatomy (<strong>Figure 2</strong>),
you should see experimental details, an image, and code. If you click on
separate parts of the notebook you can see that it is divided into
individual cells (<strong>Figure 2 e-g</strong>) which are of varying
type (Code, R in this case, or Markdown - <strong>Figure 2d</strong>).
Hashtags are comments within the code and shall help you to interpret
what individual bits of code do. <img src="../fig/10-02-jupyter_anatomy.png" alt="Figure 2. Anatomy of a Jupyter notebook: (a) depicts the name of thenotebook, (b, c) are toolbars, (c) contains the most commonly usedtools, (d) shows of what type - Markdown, Code etc... - the currentlyselected cell is, and (e-g) are examples of cells, where (e) shows thecurrently selected cell." class="figure"><em>Figure 2. Anatomy of a Jupyter notebook: (a) depicts the name of the
notebook, (b, c) are toolbars, (c) contains the most commonly used
tools, (d) shows of what type - Markdown, Code etc… - the currently
selected cell is, and (e-g) are examples of cells, where (e) shows the
currently selected cell.</em></p></li>
<li><p>Change the title of the notebook (<strong>a</strong>) to your
initials e.g. “student_notebook_light_conditions_IB”</p></li>
<li><p>To Save the notebook click on the disk symbol in the toolbar
(<strong>c</strong>).</p></li>
<li><p>To run the code select the top cell of the notebook
(<strong>e</strong>) - this is likely pre-selected already - and click
“Run” in the tool bar (<strong>c</strong>). The selected cell jumps one
cell down (<strong>f</strong>).</p></li>
<li><p>To avoid having to press Run repeatedly until we are at the end
of the code (try clicking Run two more times and see what happens), we
will show you how to run all the code. In the top tool bar
(<strong>b</strong>) click “Cell” and select “Run All”. (“Cell &gt; Run
All”.)</p></li>
<li><p>If you have another look at the notebook you can see that a
table, graph and outputs of statistical testing have been
generated.</p></li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>If you followed all steps correctly you should have reproduced the
table, a graph and statistical testing. Apart from the pre-filled
markdown text the rendered values of the code should look like this:
<img src="../fig/10-03-jupyter_dataframe.png" alt="Figure 3. Rendering of data frame" class="figure"><em>Figure 3. Rendering of
data frame</em> <img src="../fig/10-04-jupyter_plot.png" alt="Figure 4. Rendering of plot" class="figure"><em>Figure 4. Rendering of
plot</em></p>
</div>
</div>
</div>
</div>
<div id="exercise-2-how-to-add-and-remove-content-3-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-2-how-to-add-and-remove-content-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: how to add and remove content (3 min)</h3>
<div class="callout-content">
<p>In your previously saved notebook, we will now show you how to add
text and remove cells within your notebook. Additionally we will show
you how to change code:</p>
<ol style="list-style-type: decimal">
<li>We want to change the author name to your name: Double click on the
cell containing the author name. You can see how the layout of the cell
changes into Markdown, and you can simply change the name.</li>
<li>Press the Run button in the toolbar and the Markdown of the cell
will be rendered again and you can see the text shows as
previously.</li>
<li>We now want to add some details about the “light_results.txt” file
which is being loaded. To add a cell above the code, click the cell
currently above the first lines of code and click the “+” in the
toolbar. This creates a cell below selected cell.</li>
<li>Now add a note about the file which is being loaded and the purpose
of the subsequent analysis: e.g. Loading of results following short- and
long-day light exposure on arabidopsis, followed by visualisation of
differences in chlorophyll/biomas etc… content between genotypes on
short-days and long-days. To show the cell as Markdown and not code,
make sure “Markdown” is selected in the toolbar.</li>
<li>To remove a cell, select it and click on the scissors icon in the
toolbar (This can be undone under Edit &gt; Undo Delete Cells).</li>
<li>To change the output of your graph click on the cell containing the
code below the “Visualise data” title. We want you to change the colours
of the box-plots. You can do this where the comment “# change colour of
groups” is written. Either you can use three different HEX codes (a
6-symbol code representing colours from white to black), or colours
simply by their name - e.g. dark blue, orange…</li>
<li>To save the graph under a different name add your initials to the
file name under which the image is being saved. Press Run again. Your
image should be visible in the overall file hierarchy.</li>
</ol>
</div>
</div>
</div>
<p>(3 min teaching)</p>
</section><section><h2 class="section-heading" id="jupyter-notebooks-are-easy-to-use-and-great-for-collaborative-efforts">Jupyter Notebooks are easy to use and great for collaborative
efforts<a class="anchor" aria-label="anchor" href="#jupyter-notebooks-are-easy-to-use-and-great-for-collaborative-efforts"></a>
</h2>
<hr class="half-width">
<p>Imagine your collaborators have shared this experiment measuring
biomas of <em>Arabidopsis</em> with you and you were able to duplicate
and follow their entire analysis from start to finish, including
interpretation of data. Now you are interested in analysing what biomas
look like on long-days. You are in luck!</p>
<p>Because Jupyter Notebooks are so easy to handle you simply need to
copy and paste the already existing code your collaborators shared and
adapt variables such as short-day to long-day and change the names under
which figures are being saved to avoid duplication.</p>
<div id="exercise-3-add-another-analysis-step-10-min-including-walkthrough" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-add-another-analysis-step-10-min-including-walkthrough" class="callout-inner">
<h3 class="callout-title">Exercise 3: add another analysis step (10 min including walkthrough)</h3>
<div class="callout-content">
<p>We have shown you how to manipulate text and code in Jupyter
notebooks, now we want you to add data visualisation (a graph) and stats
for long- day light condition:</p>
<ol style="list-style-type: decimal">
<li>Add additional cells including</li>
</ol>
<ul>
<li>titles</li>
<li>edited code to depict graph from long-days and not short-days</li>
<li>Figure legend</li>
<li>statistical testing of difference between genotypes on
long-days</li>
<li>interpretation of results of statistical testing.</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>The following code will result in your new graph:</p>
<pre class="source"><code><span><span class="fu">ggplot</span><span class="op">(</span><span class="fu">subset</span><span class="op">(</span><span class="va">df</span>, <span class="va">light_condition</span> <span class="op">%in%</span> <span class="st">"LD"</span><span class="op">)</span>,</span>
<span><span class="co"># subset only SD from &gt;&gt;light condition column for plotting</span></span>
<span>      mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">biomas</span>, fill <span class="op">=</span> <span class="va">genotype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="co"># x-axis shows genotype, y-axis shows biomas</span></span>
<span>   <span class="fu">geom_boxplot</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.3</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Biomas per Genotype on long days"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Genotype"</span>, <span class="co"># Title of x-axis</span></span>
<span>       y <span class="op">=</span> <span class="st">"Biomas (g)"</span><span class="op">)</span> <span class="op">+</span> <span class="co"># Title of y-axis</span></span>
<span>   <span class="co"># change colour of groups</span></span>
<span>   <span class="fu">scale_fill_manual</span><span class="op">(</span>values<span class="op">=</span><span class="fu">c</span><span class="op">(</span><span class="st">"#999999"</span>, <span class="st">"#E69F00"</span>, <span class="st">"#56B4E9"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu">theme</span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"none"</span><span class="op">)</span></span></code></pre>
<p>The following code will result in testing of biomas between genotypes
in long-days - we assign a new variable to separate both analysis.</p>
<pre class="source"><code><span><span class="va">res.aov.LD</span> <span class="op">&lt;-</span> <span class="fu">aov</span><span class="op">(</span><span class="va">biomas</span> <span class="op">~</span> <span class="va">genotype</span>, data <span class="op">=</span> <span class="fu">subset</span><span class="op">(</span><span class="va">df</span>, <span class="va">light_condition</span> <span class="op">%in%</span> <span class="st">"LD"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Summary of the analysis</span></span>
<span><span class="fu">summary</span><span class="op">(</span><span class="va">res.aov.LD</span><span class="op">)</span></span></code></pre>
<p>The following code will result in Tukey multiple pairwise-comparison
testing.</p>
<pre class="source"><code><span><span class="co"># conduct Tukey multiple pairwise-comparison</span></span>
<span><span class="fu">TukeyHSD</span><span class="op">(</span><span class="va">res.aov.LD</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="easy-sharing-of-your-notebook">Easy sharing of your notebook<a class="anchor" aria-label="anchor" href="#easy-sharing-of-your-notebook"></a>
</h2>
<hr class="half-width">
<p>We have now managed to not only reproduce code, but we were able to
add to the analysis and interpretation of overall results. To show your
PI and colleagues your results, you want to save the notebook in
readable format.</p>
<div id="exercise-4-sharing-of-your-jupyter-notebook-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-sharing-of-your-jupyter-notebook-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 4: Sharing of your Jupyter Notebook (5 min)</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Download your Notebook (ensure all code has been run) as .html</li>
<li>View the documents and think about why it is important to run all
code before download (try Cell &gt; All Output &gt; Clear and download
your Notebook then and compare)</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>It is important all code is run before the notebook is downloaded, as
during download only the text and graphs are saved that are currently
visible in your notebook.</p>
</div>
</div>
</div>
</div>
<p>The example notebook is extremely well documented to show the good
practices when coding. Without such level of comments and explanations
it may not be possible for new-commers to re-use this notebook.</p>
</section><section><h2 class="section-heading" id="plotting-in-r-or-python">Plotting in R or Python<a class="anchor" aria-label="anchor" href="#plotting-in-r-or-python"></a>
</h2>
<hr class="half-width">
<p>Plotting in R or Python is often a natural starting point to learn
programming and allows to create more professional scientific plots than
those available in Excel. Additionally, these plots are easier to
recreate, and easier to adjust for specific dimensions and journal
formatting guidelines. Using code it is very easy to prepare series of
figures that follow the same formatting of all their elements.</p>
</section><section><h2 class="section-heading" id="ad-hoc-analysis-with-jupyter-notebooks">Ad-hoc analysis with Jupyter Notebooks<a class="anchor" aria-label="anchor" href="#ad-hoc-analysis-with-jupyter-notebooks"></a>
</h2>
<hr class="half-width">
<p>Notebook can document entire ad-hoc analysis. It can capture the
motivations and decisions which lead us to the final results (as the
markdown-cells).<br>
It contains information about the input data uses, the actual parameters
and functions called.<br>
It can include intermediate results, adjustment made.<br>
It captures all the steps that lead to the final result which can be
accompanied with the conclusions.</p>
<p>One thing to keep in mind is that your reusable analysis with Jupyter
is only ever as good as your self discipline. Things you want to keep in
mind are:</p>
<ul>
<li>document the entire decision process and motivations behind it</li>
<li>document input data</li>
<li>document parameters and their significance</li>
<li>decide what data you want to retain and clean - annotate why</li>
<li>comment your code where necessary</li>
<li>follow coding good practices (in naming variables, functions, in
code formatting)</li>
<li>Notebook has to be shipped with all file inputs and description of
runtime environment</li>
</ul>
<p>Notebooks are very well suited to:</p>
<ul>
<li>orchestrate „short”, step by step operations in R, python, shell
(notebooks can use all 3 at the same time)</li>
<li>capture parameters</li>
<li>add interpretations</li>
<li>act as a „flexible” user interface (user can change the runtime
parameters in the notebook following the embedded instructions)</li>
</ul>
<p>However, notebooks should not be used a replacement of integrated
development environment (IDE) and writing modularized, split into
packages code. They are not suitable to write a long programs, long
executable code should be compiled into stand alone modules.</p>
<div id="exercise-5-accessibility-of-jupyter-notebooks-5-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-5-accessibility-of-jupyter-notebooks-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 5: Accessibility of Jupyter Notebooks (5 min)</h3>
<div class="callout-content">
<p>On a scale from -2 to 2, how do you feel about the following
statements (R is interchangeable with Python), where -2 (strongly
disagree), 0 no opinion to +2 strongly agree:</p>
<ul>
<li>making graphs for a subset of data is easier in R than in Excel</li>
<li>it is easier to filter for data in R than in Excel</li>
<li>it is easier to apply formulas or calculations in R than in
Excel</li>
<li>it is easier to generate a series of plots with similar layout in R
than Excel</li>
<li>it is easier to do large scale data processing in R than in
Excel</li>
<li>using notebooks does not require any programming knowledge</li>
<li>notebooks give you a better overview of your data processing than
Excel</li>
<li>Jupyter is free, whilst a Microsoft Office (+Excel) suite costs
$149.99, this alone is an incentive to use Jupyter notebooks as not all
individuals have Microsoft Office supported by their employers</li>
<li>you need to learn R to do any data processing in notebooks</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="pipelines">Pipelines<a class="anchor" aria-label="anchor" href="#pipelines"></a>
</h2>
<hr class="half-width">
<p>Computational pipelines differ from ad-hoc analysis like a protocol
differs from lab record. Pipelines tend to represent fixed and testes
step by step analysis / processing. There is no need for
experimentation, but, the processing should be fast and reproducible.
Pipelines usually take few configuration parameters and input files. The
pipeline computations will be repeated multiple times on different input
data and the life-span of a established pipeline can be in months or
years.</p>
<p>For that reason the computational pipelines should offer:</p>
<ul>
<li>robustness, for example restarting job from the middle of the
analysis without a need to recreate the first steps</li>
<li>support for parallel computing</li>
<li>easy to maintain, ie change inputs, update dependencies, add new
step</li>
<li>be methodologically sound</li>
<li>have well specified dependencies</li>
</ul>
<p>It is typically achieved by using one of the systems for managing
computational workflows, like for example:</p>
<ul>
<li>Nextflow</li>
<li>Snakemake</li>
<li>Galaxy</li>
</ul>
<p>They all offer well defined:</p>
<ul>
<li>ways of defining inputs and parameters</li>
<li>behaviour, like transition to next stages</li>
<li>error handling</li>
<li>restart options</li>
</ul></section><section><h2 class="section-heading" id="computing-in-r-and-python">Computing in R and Python<a class="anchor" aria-label="anchor" href="#computing-in-r-and-python"></a>
</h2>
<hr class="half-width">
<p>Both languages are flexible and easy to start. They permit quickly
achieve the desired effect, have things done with minimal setup and
lines of codes. However, because of their flexibility it is easy to pick
up bad habits and write code without much understanding of good
practices of software engineering.</p>
<p>For the journey into scientific computing we recommend:</p>
<ul>
<li>Firstly, learn how to make simple plots, clean/reorganize files and
data tables</li>
<li>Secondly, learn basics of software engineering and good programming
practices</li>
<li>Then start coding advanced analysis and processing, construct
pipelines with workflows</li>
</ul>
<p>One of the pitfalls of R and Python is that their behaviour depends
on libraries installed on your machine, not only R o Python libraries
but also the system one. It often creates situation in which the code
works on one machine but does not work on another.</p>
<p>There are ways of dealing with those issues, for example, for Python,
there is a package management system called Conda. Conda installs, runs,
and updates packages and their dependencies. Conda can switch between
project environments, so each project can have its own set of installed
libraries and their versions.<br>
Conda support many languages (Python, R, Ruby, Lua, Scala, Java,
JavaScript, C, C++, FORTRAN) but it is particularly well suited for
Python.</p>
<p>In our experience, for R renv works better than conda.</p>
<p>As a rule, despite the package managment system used, you should try
to keep track of what libraries you installed during your code
development and in what versions.</p>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was adopted after</p>
<ul>
<li><a href="https://doi.org/10.1038/d41586-018-07196-1" class="external-link">Why Jupyter is
a data scientists’ computational notebook of choice</a></li>
</ul>
</div>
</div>
</div>
<div id="further-reading" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="further-reading" class="callout-inner">
<h3 class="callout-title">Further Reading</h3>
<div class="callout-content">
<ul>
<li><a href="https://reproducible-analysis-workshop.readthedocs.io/en/latest/4.Jupyter-Notebook.html" class="external-link">Reproducible
analysis and Research Transparency</a></li>
<li><a href="https://www.browserstack.com/guide/coding-standards-best-practices" class="external-link">Coding
Standards best practices</a></li>
</ul>
</div>
</div>
</div>
<div id="for-instructors-advanced-teaching" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="for-instructors-advanced-teaching" class="callout-inner">
<h3 class="callout-title">For instructors: Advanced teaching</h3>
<div class="callout-content">
<p>For experienced Notebook users who normally use Python:</p>
<ul>
<li>Create your own notebook using Python instead of R, replicate the
above code-output</li>
</ul>
<p>For experienced R users:</p>
<ul>
<li>Plot both short- and long-day light conditions as a grouped
boxplot</li>
</ul>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<pre><code><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">df</span>, mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">genotype</span>, y <span class="op">=</span> <span class="va">biomas</span>, fill<span class="op">=</span><span class="va">light_condition</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Biomas per Genotype"</span>,</span>
<span>        x <span class="op">=</span> <span class="st">"Genotype"</span>,</span>
<span>        y <span class="op">=</span> <span class="st">"Biomas (g)"</span>,</span>
<span>        fill <span class="op">=</span> <span class="st">"Light condition"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span></span></code></pre>
<p><img src="../fig/10-05-advanced_plot.png" alt="Figure 5. Advanced Plot" class="figure"><em>Figure 5. Short- and long-day light
conditions depicted as a grouped boxplot</em></p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Jupyter Notebooks are useful tools to share analysis with
non-programmers</li>
<li>One single document can visualise background, results, formulae/code
and metadata</li>
<li>One single document helps to make your work more understandable,
repeatable and shareable</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-11-version-control"><p>Content from <a href="11-version-control.html">Version control</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/11-version-control.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 74 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>How not to worry about making changes to a project?</li>
<li>What is version control?</li>
<li>What is git and github?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>What are the problems with introducing changes to files</li>
<li>Making a good file change description</li>
<li>Why use version control systems</li>
<li>Use of semantic versioning</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(1 min teaching)</p>
<p><img src="../fig/ew-versions.png" alt="Figure 1. Versioning Hell" class="figure"><br><em>from: Wit and wisdom from Jorge Cham (<a href="https://phdcomics.com/" class="external-link">http://phdcomics.com/</a>)</em></p>
<p>Our projects are not static we:</p>
<ul>
<li>add new files</li>
<li>add data entries to files</li>
<li>modify manuscripts / methods</li>
<li>reformat / reorganize data tables</li>
<li>reanalyse data / update figures</li>
<li>experiment with data cleaning / processing / visualization</li>
<li>share and co-edit data or text files</li>
</ul>
<p>All those actions introduce change to files of our project.</p>
<div id="exercise-1-problems-with-change-5-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-1-problems-with-change-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Problems with change (5 min)</h3>
<div class="callout-content">
<p>Which of these issues can you relate to?</p>
<ul>
<li>I have fifteen versions of this file - which one do I use?</li>
<li>I can’t remake this figure from last year</li>
<li>I slightly modified my code in one place, everything stopped
working</li>
<li>I have several copies of the same directory because I’m worried
about breaking something</li>
<li>Somebody added erroneous records in a shared file with samples,
cannot find who and why</li>
<li>You remember seeing a data file but cannot find it anymore: is it
deleted? Moved away?</li>
<li>I tried multiple analysis and I don’t remember which one I chose to
generate my output data</li>
<li>I have to merge changes to our manuscript from 10 different emails
with collaborators</li>
<li>I accidentally deleted a part of my work</li>
<li>I came to an old project and forgot where I left/saved it</li>
<li>I reorganized my data files, but broke the analysis pipeline and
don’t know how to fix it</li>
<li>I have trouble to find the source of a mistake in an experiment</li>
<li>My directory is polluted with a lot of unused/temporary/old folders
because I’m afraid of losing something important</li>
<li>I made a lot of changes to my paper but want to bring back one
paragraph</li>
</ul>
</div>
</div>
</div>
<p>(6 min teaching)</p>
<p>When introducing changes to files and their content, we:</p>
<ul>
<li>overwrite old content or loose whole files (hoping they won’t be
needed)</li>
<li>may introduce side effects, e.g. renaming files may break analysis
pipelines</li>
<li>introduce multiple changes to multiple files which should be treated
as one change, e.g. new version of a figure and a paragraph that
describes it</li>
<li>need to collate concurrent changes from collaborators e.g. “resolve
conflicts”</li>
</ul>
<p>Addressing those issues by tracking changes is called <strong>version
control</strong>. Version control decouples the process of making
changes from the physical change itself. It identifies (tags) a
collection of changes to files and references them, potentially allowing
to come back to the state before or after that change to retrieve
details.</p>
<p>With version control file names do not reflect their versions.
Information about versions and changes are kept separate from the
files.</p>
<p>It is possible to achieve version control manually by a systematic
approach for managing changes.</p>
<section><h2 class="section-heading" id="manual-versioning">Manual Versioning<a class="anchor" aria-label="anchor" href="#manual-versioning"></a>
</h2>
<hr class="half-width">
<p>Our first suggested approach in which everything is done by hand has
two main parts:</p>
<ol style="list-style-type: decimal">
<li>
<strong><em>Add a file called <code>CHANGELOG.txt</code> to the
project’s <code>docs</code> subfolder</em></strong>, and make dated
notes about changes to the project in this file in reverse chronological
order (i.e., most recent first). This file is the equivalent of a lab
notebook, and should contain entries like those shown below.</li>
</ol>
<pre><code>## 2016-04-08

* Switched to cubic interpolation as default.
* Moved question about family's TB history to end of questionnaire.

## 2016-04-06

* Added option for cubic interpolation.
* Removed question about staph exposure (can be inferred from blood test results).</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>
<strong><em>Copy the entire project whenever a significant change
has been made</em></strong> (i.e. one that materially affects the
results) and store that copy in a sub-folder whose name reflects the
date in the area that is being synchronized. This approach results in
projects being organized as shown below:</li>
</ol>
<pre><code>.
|-- project_name
|   -- current
|       -- ...project content as described earlier...
|   -- 2016-03-01
|       -- ...content of 'current' on Mar 1, 2016
|   -- 2016-02-19
|       -- ...content of 'current' on Feb 19, 2016</code></pre>
<p>Here the <code>project_name</code> folder is mapped to external
storage (such as Dropbox), <code>current</code> is where the current
work is done, and other folders within <code>project_name</code> are old
versions.</p>
<blockquote>
<p><strong>Data is Cheap, Time is Expensive</strong></p>
<p>Copying everything like this may seem wasteful since many files won’t
have changed, but consider: a terabyte hard drive costs about $50
retail, which means that 50 GByte costs less than $5. Provided large
data files are kept out of the backed-up area (discussed below), this
approach costs less than the time it would take to select files by hand
for copying.</p>
</blockquote>
<ol start="3" style="list-style-type: decimal">
<li><p><strong><em>Keep changes small.</em></strong><br>
Each change should not be too large as to make the change tracking
irrelevant. For example, a single change called “Data for paper” adding
or changing several different type of data sets, and their analysis
scripts and results is likely too large, as this doesn’t allow to
distinguish between changes to different components of the analysis.
Similarly, changes should not be broken up into pieces that are too
small.<br>
As a rule of thumb a good size for a single change is a group of edits
that you could imagine wanting to undo in one step.</p></li>
<li><p><strong><em>Store each project in a folder that is mirrored off
the researcher’s working machine</em></strong><br>
Use a system such as Dropbox or or Google Drive and synchronize the
folder at least daily. It may take a few minutes, but that time is
repaid the moment a laptop is stolen or its hard drive fails.</p></li>
<li><p><strong><em>Share changes frequently.</em></strong><br>
Everyone working on the project should share and incorporate changes
from others on a regular basis. Do not allow individual investigator’s
versions of the project repository to drift apart, as the effort
required to merge differences goes up faster than the size of the
difference.</p></li>
</ol>
<div id="exercise-2-manual-versioning-5-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-2-manual-versioning-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Manual versioning (5 min)</h3>
<div class="callout-content">
<p>Which of these issues does manual versioning help to tackle?</p>
<ul>
<li>I have fifteen versions of this file and I don’t know which one to
use</li>
<li>I can’t remake this figure from last year</li>
<li>I slightly modified my code in one place, everything stopped
working</li>
<li>I have several copies of the same directory because I’m worried
about breaking something</li>
<li>Somebody added erroneous records in a shared file with samples, I
cannot find who and why</li>
<li>You remember seeing a data file but cannot find it anymore: is it
deleted? Moved away?</li>
<li>I tried multiple analysis and I don’t remember which one I chose to
generate my output data</li>
<li>I have to merge changes to our manuscript from 10 different emails
from collaborators</li>
<li>I made a lot of changes to my paper but want to bring back one
paragraph</li>
</ul>
</div>
</div>
</div>
<div id="how-to-document-a-change-6-min-teaching" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="how-to-document-a-change-6-min-teaching" class="callout-inner">
<h3 class="callout-title">How to document a change (6 min teaching)</h3>
<div class="callout-content">
<p>A good entry that documents changes should contain:</p>
<ul>
<li>date of the change</li>
<li>author of the change</li>
<li>list of affected files</li>
<li>a short description of the nature of the introduced changes AND/OR
motivation behind the change</li>
</ul>
<p>Examples of the descriptions are:</p>
<ul>
<li>added flow cytometry data for control and starvation stressed
samples</li>
<li>updated matplot library to version 3.4.3 and regenerated
figures</li>
<li>added panel with protein localisation to Figure 3 and its discussion
in the text</li>
<li>reverted to the previous version of the abstract text as the
manuscript reached word limits</li>
<li>cleaned the strain inventory<br>
Recent freezer cleaning and ordering indicated a lot of problem with the
strains data. The missing physical samples were removed from the table,
duplicated ids are marked for checking with PCR. The antibiotic
resistence were moved from phenotype description to its own column.</li>
<li>new regulation heatmap<br>
As suggested by Will I used the normalization and variance stabilization
procedure from Hafemeister et al prior to clustering and heatmap
generation</li>
</ul>
<p>The larger the project (measured either in: collaborators, file
numbers, or workflow complexity) the more detailed the change
description should be.<br>
While your personal project can contain brief descriptions, larger
projects should always contain information about the motivation behind
the change and the consequences.</p>
</div>
</div>
</div>
<p>This manual procedure satisfies the requirements of version tracking
without needing new tools. If multiple researchers are working on the
same project they will need to coordinate so that only a single person
works on specific files at any time. In particular, they may wish to
create one change log file per contributor and to merge those files
whenever a backup copy is made.</p>
<p>What the manual process described above requires most is
self-discipline. The version control tools that underpin our second
approach don’t just accelerate the manual process: they also automate
some steps while enforcing others and thereby deliver more reliable
results.</p>
<div id="exercise-3-changelog-in-action-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-changelog-in-action-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 3: Changelog in action (5 min)</h3>
<div class="callout-content">
<p>Have a look at one of the example Github repositories and how they
track changes:</p>
<ul>
<li><a href="https://github.com/ewallace/pseudonuclease_evolution_2020/commits/master" class="external-link">data
from E.R. Ballou et al. 2020</a></li>
<li><a href="https://github.com/BioRDM/nmj-pig/commits/main" class="external-link">data from
I. Boehm et al. 2020</a></li>
</ul>
<p>Give examples of:</p>
<ul>
<li>what makes them a good changelog</li>
<li>what could be improved</li>
</ul>
<p>Think which are the most difficult features to replicate with manual
version control?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>The good things:</p>
<ul>
<li>all log entries contain date and author</li>
<li>all log entries contain list of files that have been modified</li>
<li>for text files the actual change can be visible</li>
<li>the description text gives an idea of the change</li>
</ul>
<p>What could be improved</p>
<ul>
<li>The pig files should probably be recorded in smaller chunks
(commits). The raw data and cleaned data could be added separetely
unless they all were captured at the same time.</li>
<li>Rather than general “Readme update” a more specific description
could be provided “Reformatted headers and list”</li>
<li>some of the Ballous changes could do with more detailed
descriptions, for example why the change took place in case of IQ_TREE
entries</li>
</ul>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="version-control-systems">Version Control Systems<a class="anchor" aria-label="anchor" href="#version-control-systems"></a>
</h2>
<hr class="half-width">
<p>(13 min teaching)</p>
<div id="how-version-control-systems-work" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="how-version-control-systems-work" class="callout-inner">
<h3 class="callout-title">How Version Control Systems Work</h3>
<div class="callout-content">
<p>A version control system is a tool that keeps track of file changes
for us and helps us version and merge our files. It allows you to decide
which changes make up the next version, called a commit, and keeps
useful metadata about them. The complete history of commits for a
particular project and their metadata make up a repository.</p>
<p>A version control system stores snapshots of a project’s files in a
repository. Users modify their working copy of the project and then save
changes to the repository when they wish to make a permanent record
and/or share their work with colleagues. The version control system
automatically records when the change was made and by whom along with
the changes themselves.</p>
<p>Repositories can be kept in sync across different computers
facilitating collaboration among different people.</p>
<p>Crucially, if several people have edited files simultaneously, the
version control system will detect the collision and require them to
resolve any conflicts.</p>
</div>
</div>
</div>
<p><img src="../fig/vc-play-changes.svg" alt="Figure 2. How VCS works" class="figure"><br><img src="../fig/vc-play-changes2.svg" class="figure"><br><img src="../fig/vc-merges.svg" class="figure"><br><em>from: Version control with git (<a href="https://carpentries-incubator.github.io/git-novice-branch-pr/01-basics/" class="external-link uri">https://carpentries-incubator.github.io/git-novice-branch-pr/01-basics/</a>)</em></p>
<p>Tool-based version control has several benefits over manual version
control:</p>
<ul>
<li><p>Instead of requiring users to backup copies of the whole project,
version control safely stores just enough information to allow old
versions of files to be re-created on demand.</p></li>
<li><p>Instead of relying on users to choose sensible names for backup
copies the version control system timestamps all saved changes
automatically.</p></li>
<li><p>Instead of requiring users to be disciplined about completing the
changelog version control systems prompt them every time a change is
saved. They also keep a 100% accurate record of what was
<em>actually</em> changed, as opposed to what the user <em>thought</em>
they changed, which can be invaluable when problems crop up
later.</p></li>
<li><p>Instead of simply copying files to remote storage, version
control checks to see whether doing that would overwrite anyones work.
If so, they identify conflicts and merge changes.</p></li>
</ul>
<p>We believe that the best tools for tracking changes are the version
control systems that are used in software development such as Git. Git
keeps track of what was changed in a file when and by whom and it can
synchronize changes to a central server (GitHub) so that many users can
manage changes to the same set of files.</p>
<p>Simplifying Git is a local program (like Word) that runs on your
machine and knows how to track changes. Github is an online system that
can communicate with Git and propagate your changes to other computers
(like OneDrive can do for Word documents).</p>
<p>Keeping track of changes that you or your collaborators make to data
and software is a critical part of research. Being able to reference or
retrieve a specific version of the entire project aids in
reproducibility for you leading up to publication, when responding to
reviewer comments, and when providing supporting information for
reviewers, editors, and readers.</p>
<div id="exercise-4-using-version-control-system-5-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-4-using-version-control-system-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 4: Using Version Control System (5 min)</h3>
<div class="callout-content">
<p>Five reasons to use a version control system in research</p>
<ul>
<li>Tell the story: The history of your commit messages will describe
your project progress.</li>
<li>Travel back in time: a version control system makes it easy to
compare different time points of your project smoothly. If you want to
compare the stage of your project a year ago from now, it only takes one
command-line of code.</li>
<li>Experiment with changes: if you want to make changes in a script you
can first make a “snapshot” of the project status before experimenting
with changes.</li>
<li>Backup your work: by being able to link your local repository
(folder) to a distant online host (GitHub), a version control system
backs up your precious work instantly.</li>
<li>Collaborate easily on projects: having a web-hosted synchronised
version of your project will encourage collaboration with other
researchers. Think about a colleague of yours being able to add a script
to make a figure for your publication for instance. The repository will
track their contribution and evidence their participation in the
project.</li>
</ul>
<p>Which two are the most important for you.</p>
</div>
</div>
</div>
<div id="version-control-system-and-fair" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="version-control-system-and-fair" class="callout-inner">
<h3 class="callout-title">Version Control System and FAIR</h3>
<div class="callout-content">
<p>Looking at the reasons to use a version control system (VCS) in
research, how does using VCS help in being FAIR?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<ul>
<li>Online respository is a Findable and Accessible resource (also
Trackable).</li>
<li>Online repositories are also interoperable since they can be
accessed from anywhere (with internet) and are not restricted to your
local hard drive.</li>
<li>The ‘telling story’ and being able to travel back in time helps in
reuse and reproducibility.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="learning-git-10-min-teaching" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="learning-git-10-min-teaching" class="callout-inner">
<h3 class="callout-title">Learning git (10 min teaching)</h3>
<div class="callout-content">
<p>If you do any form of scientific computing: Python, R, Matlap or even
bash script you simply must learn git and use it on a daily basis (no
excuses).</p>
<p><img src="../fig/using-git.png" alt="Figure 3. Using Git" class="figure"><br><em>from: xkcd (<a href="https://xkcd.com/1597/" class="external-link uri">https://xkcd.com/1597/</a>)</em></p>
<p>For lab-bench biologists git can have a steep learning curve. It was
designed and developed for and by Linux developers so … user experience
has never been its top priority. We encourage biologist to follow the
best practice of manual versioning and then try to see git in actions.
We are sure that after manual experience you apreciate the benefits of
git, and accept the learning overhead.</p>
<p>Some git resources:</p>
<ul>
<li>Software Carpentry git workshop <a href="https://swcarpentry.github.io/git-novice/" class="external-link uri">https://swcarpentry.github.io/git-novice/</a>
</li>
<li>Edinburgh Carpentries runs courses: <a href="https://edcarp.github.io/" class="external-link uri">https://edcarp.github.io/</a>
</li>
<li><a href="https://ourcodingclub.github.io/tutorials/git/" class="external-link uri">https://ourcodingclub.github.io/tutorials/git/</a></li>
<li>Crash course on youtube: <a href="https://youtu.be/SWYqp7iY_Tc" class="external-link uri">https://youtu.be/SWYqp7iY_Tc</a>
</li>
<li>Learn git branching: <a href="https://learngitbranching.js.org/" class="external-link uri">https://learngitbranching.js.org/</a>
</li>
<li>Learn git-game: <a href="https://github.com/git-game/git-game" class="external-link uri">https://github.com/git-game/git-game</a>
</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="semantic-versioning-optional">Semantic versioning (Optional)<a class="anchor" aria-label="anchor" href="#semantic-versioning-optional"></a>
</h2>
<hr class="half-width">
<p>(10 min teaching) Although we recommend to record small changes often
a particular point in time (a set of changes) has its logical
meaning.</p>
<p>Probably all of us made folders: submission, revision 1, revision 2
when publishing an article. Each containing the complete set of
files.</p>
<p>In software engineering terminology, such as the “official” point in
time (especially if it is meant to be consumed by others) is called
<strong><em>release</em></strong>.</p>
<p>If you are a user of a software/library, unlike when accessing
manuscripts, you may not always want to use the latest version of a
software. For example, it may not support your old Windows version, it
may change how it processes the data, or the output format. Updating
software may introduce a lot of side effects to your pipeline.</p>
<p>Modern software is built using a lot of external software libraries,
the potential issue when using the “newest version” of a library is well
known and called “dependency hell”. To help deal with this problem,
<strong><em>semantic versioning</em></strong> has been proposed (<a href="https://semver.org/%5C)%5Bhttps://semver.org/%5D" class="external-link uri">https://semver.org/\)[https://semver.org/]</a>.</p>
<p>In brief, it is a way for numbering software releases (naming
particular versions).</p>
<p><img src="../fig/semver-def.png" alt="Figure 4. Semantic versioning" class="figure"><br><em>from: Semantic versioning, Parikshit Hooda (<a href="https://www.geeksforgeeks.org/introduction-semantic-versioning/" class="external-link uri">https://www.geeksforgeeks.org/introduction-semantic-versioning/</a>)</em></p>
<p>Semantic Versioning is a 3-component number in the format of X.Y.Z,
where:</p>
<ul>
<li><p>The leftmost number X denotes a major version. Major versions
introduce “breaking changes”, i.e. changes will affect other software
that interacts with it. For example, if the file format for inputs or
outputs changes it is a breaking change as the existing files cannot be
read (in case of input change) or the output files cannot be read.
Another example are renaming program functions or changing their call
parameters.</p></li>
<li><p>Y stands for a minor version. It is used for the release of new
functionality in the system. When you increase the minor version you
communicate that the software can be used exactly as the old version but
it also offers some new features. For example, your plot library can now
plot in colors not only black and white or the new version of software
can output the results to both Excel and CSV files rather than just
CSV.</p></li>
<li><p>Z stands for a Patch Versions: Versions for patches are used for
bug fixes and security updates. There are no functionality changes in
the patch version upgrades.</p></li>
</ul>
<p><img src="../fig/semver-upgrade2.png" alt="Figure 5. Semantic versioning" class="figure"><br><em>from: Semantic versioning, Parikshit Hooda (<a href="https://www.geeksforgeeks.org/introduction-semantic-versioning/" class="external-link uri">https://www.geeksforgeeks.org/introduction-semantic-versioning/</a>)</em></p>
<p>When you increase the Major version you zero the remaining ones, when
you increase Minor you zero the patch number.</p>
<p>Having “.” between major, minor and patch assures that when lexically
sorted, major version will “dominate” minor, and minor dominates the
patch number, see the sorted releases:</p>
<pre><code>1.91.12
1.91.8
2.13.3-alpha
2.14-alpha
2.15-beta
3.0.923</code></pre>
<p>Semantic versioning permits dependency software to automatically
choose the most recent version of a library, or automatically update
libraries to a patched version while avoiding problems with changing to
new major release.</p>
<p>While semantic versioning is mainly used for software, it is also
adopted for datasets or collaborative documents, as it communicates the
<em>severity of changes</em>. For example if you as the co-author read:
<code>manuscript.2.1.5.docx</code> and now there is
<code>manuscript.2.1.10.docx</code> you probably don’t need to do
anything, while <code>manuscript.3.0.10.docx</code> should capture your
attention.</p>
<p>For data and documents often only major.minor numbers are used.</p>
<div id="semantic-versioning-quiz-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="semantic-versioning-quiz-5-min" class="callout-inner">
<h3 class="callout-title">Semantic versioning quiz (5 min)</h3>
<div class="callout-content">
<ol style="list-style-type: decimal">
<li>Which of the library version is the latest?</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>0.12.4</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>1.1.0</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>1.12.3-alpha</li>
</ol></li>
<li><ol start="4" style="list-style-type: lower-alpha">
<li>1.12.2</li>
</ol></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>You use library fair.2.3.1 to automatically upload your data after
analysis. You found out that there is avaialble new version of the
library fair.4.3.1. Which situation is the most probable?</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>fair.4.3.1 uploads data twice as fast as fair.2.3.1</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>you need to reserve time to test the new version and adapt your
analysis pipeline to work with it</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>both versions offer the same features and bug fixes as they have the
same minor/path version 3.1</li>
</ol></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>You developed code that helps to plot detrended and normalized data,
the last release has version 1.2.1. You added Z-score to the available
normalization methods and fixed the spelling errors in the user
messages. Which version number should you give to the new realease:</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>2.0.0</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>1.3.0</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>1.3.2</li>
</ol></li>
<li><ol start="4" style="list-style-type: lower-alpha">
<li>1.2.2</li>
</ol></li>
<li><ol start="5" style="list-style-type: lower-alpha">
<li>2.2.1</li>
</ol></li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>Your group releases data about studied cancer patients following
semantic versioning. The last data sets are released as 1.11, you added
data from a new patient,<br>
what version should you use:</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>1.12</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>2.11</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>2.0</li>
</ol></li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>You were asked to better anonymize the above cancer data (v1.11) in
future releases. Instead of data and time of patient visit you release
only the year, the hospital names have been encoded and patient age has
been obscured by random noise.<br>
What version should you use:</li>
</ol>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>1.12</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>2.11</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>2.0</li>
</ol></li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>c</li>
<li>b</li>
<li>b</li>
<li>a</li>
<li>c</li>
</ol>
</div>
</div>
</div>
</div>
<p><strong><em>Remember</em></strong> semantic versioning is only a
convention, there is no guarantee that updating to next patch version
won’t break your pipeline.</p>
<div id="semantic-versioning-and-fair" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="semantic-versioning-and-fair" class="callout-inner">
<h3 class="callout-title">Semantic versioning and FAIR</h3>
<div class="callout-content">
<p>How semantic versioning helps to be FAIR</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>Release version helps to be findable, the semantic notations help in
achieving interoperability, reuse and reproducibility.</p>
</div>
</div>
</div>
</div>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<ul>
<li>Majority of the content of this episode was adopted after Wilson et
al. <a href="https://github.com/swcarpentry/good-enough-practices-in-scientific-computing" class="external-link">Good
Enough Practices for Scientific Computing</a>.</li>
<li>Semantic annotations after (Parikshit Hooda)[<a href="https://www.geeksforgeeks.org/introduction-semantic-versioning/" class="external-link uri">https://www.geeksforgeeks.org/introduction-semantic-versioning/</a>]</li>
<li><a href="https://semver.org/" class="external-link">Semantic Versioning 2.0.0</a></li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Version control system helps maintaining good practices</li>
<li>Version control system keeps your work safe</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-12-templates"><p>Content from <a href="12-templates.html">Templates for consistency</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/12-templates.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 58 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What are the benefits of having templates?</li>
<li>How do templates make your research FAIR?</li>
<li>
</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>How to achieve consistency with templates</li>
<li>How do templates help with FAIR</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(13 min teaching)</p>
<section><h2 class="section-heading" id="what-are-templates">What are Templates<a class="anchor" aria-label="anchor" href="#what-are-templates"></a>
</h2>
<hr class="half-width">
<p>Templates are documents with a pre-determined layout and style which
allow for reproducibility of new documents based on the same
layout/pattern/style/guidelines. For example a useful template are label
templates that contain all necessary experimental information. Thinking
back to our arabidopsis experiment, a tube label template for an
experiment could contain: date, genotype, if short- or light-day.</p>
<div id="benefits-of-templates" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="benefits-of-templates" class="callout-inner">
<h3 class="callout-title">Benefits of Templates</h3>
<div class="callout-content">
<p>Use of such templates is incredibly convenient. If we were to print
labels for tubes, using a template we could print an entire batch at
once instead of having to do it label by label or hand write each
individual one. Overall templates:</p>
<ul>
<li>save time</li>
<li>enforce best practices (e.g. naming standards, using PIDs)</li>
<li>provide guidelines for other users of the template</li>
<li>assure consistency across sample collection, experiments, and data
analysis</li>
<li>allow automatisation of searching/processing/analysis</li>
<li>ensure reproducibility! (FAIR)</li>
</ul>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="types-of-templates">Types of templates<a class="anchor" aria-label="anchor" href="#types-of-templates"></a>
</h2>
<hr class="half-width">
<p>Templates could be anything with a pre-determined layout and purpose
to be used for <strong>repeating activities</strong>. For example
templates could be:</p>
<ul>
<li>in an ELN for different experiment types (e.g. Benchling)</li>
<li>for sample recording and registration (e.g. in Clinical Trial
Management)</li>
<li>for project structure</li>
<li>for a readme file</li>
<li>for a data management plan</li>
</ul>
<div id="exercise-1-your-template-30-min-at-end-of-day-15-min-next-day-discussing-templates" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-1-your-template-30-min-at-end-of-day-15-min-next-day-discussing-templates" class="callout-inner">
<h3 class="callout-title">Exercise 1: Your template (30 min at end of day + 15 min next day discussing templates)</h3>
<div class="callout-content">
<p>Write your own group template for either</p>
<ul>
<li>a measurement (PCR on robot etc…)</li>
<li>experiment (e.g. gene levels in response to stress)</li>
</ul>
<p>In Excel or as a Document (txt/word) or Benchling (if you fancy using
it!), also provide some example data.</p>
<p>Think about:</p>
<ul>
<li>what you will need as the producer</li>
<li>what you will need as the consumer</li>
<li>how you could limit the values (controlled vocabularies)</li>
<li>what PID or ontology terms you could use to annotate</li>
</ul>
<p>Drop the templates into the folder shared on the Notepad. Make sure
to exchange emails in case Zoom dies.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Templates save time</li>
<li>Templates enforce best practices and ensure consistency</li>
<li>Templates allow for automatisation of processes</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-13-data-repositories"><p>Content from <a href="13-data-repositories.html">Public repositories</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/13-data-repositories.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 65 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>Where can I deposit datasets?</li>
<li>What are general data repositories?</li>
<li>How to find a repository?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>See the benefits of using research data repositories.</li>
<li>Differentiate between general and specific repositories.</li>
<li>Find a suitable repository.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<section><h2 class="section-heading" id="what-are-research-data-repositories">What are research data repositories?<a class="anchor" aria-label="anchor" href="#what-are-research-data-repositories"></a>
</h2>
<hr class="half-width">
<p>(13 min teaching) Research data repositories are online repositories
that enable the preservation, curation and publication of research
‘products’. These repositories are mainly used to deposit research
‘data’. However, the scope of the repositories is broader as we can also
deposit/publish ‘code’ or ‘protocols’ (as we saw with protocols.io).</p>
<p>There are general “data agnostic” repositories, for example:</p>
<ul>
<li>
<a href="https://datadryad.org" class="external-link">Dryad</a>,</li>
<li>
<a href="https://zenodo.org" class="external-link">Zenodo</a>,</li>
<li>
<a href="https://figshare.com" class="external-link">FigShare</a>,</li>
<li>
<a href="https://thedata.org" class="external-link">Dataverse</a>.</li>
</ul>
<p>Or domain specific, for example:</p>
<ul>
<li>
<a href="https://www.uniprot.org/" class="external-link">UniProt</a> protein data,</li>
<li>
<a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a> sequence
data,</li>
<li>
<a href="https://www.ebi.ac.uk/metabolights/" class="external-link">MetaboLights</a>
metabolomics data</li>
<li>
<a href="https://github.com/" class="external-link">GitHub</a> for code.</li>
</ul>
<p>Research outputs should be submitted to discipline/domain-specific
repositories whenever it is possible. When such a resource does not
exist, data should be submitted to a ‘general’ repository. Research data
repositories are a key resource to help in data FAIRification as they
assure Findability and Accessibility.</p>
<div id="exercise-1-public-general-record-8-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-public-general-record-8-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Public general record (8 min)</h3>
<div class="callout-content">
<p>Have a look at the following record for data set in Zenodo
repository: <a href="https://doi.org/10.5281/zenodo.5045374" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.5045374.svg" alt="DOI" class="figure"></a><br>
What elements make it FAIR?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>The elements that make this deposit FAIR are:</p>
<p>Findable (persistent identifiers, easy to find data and
metadata):</p>
<ul>
<li>F1. (Meta)data are assigned a globally unique and persistent
identifier - YES</li>
<li>F2. Data are described with rich metadata (defined by R1 below)-
YES</li>
<li>F3. Metadata clearly and explicitly include the identifier of the
data they describe - YES</li>
<li>F4. (Meta)data are registered or indexed in a searchable resource -
YES</li>
</ul>
<p>Accessible (The (meta)data retrievable by their identifier using a
standard web protocols):</p>
<ul>
<li>A1. (Meta)data are retrievable by their identifier using a
standardised communications protocol - YES</li>
<li>A2. Metadata are accessible, even when the data are no longer
available - YES</li>
</ul>
<p>Interoperable (The format of the data should be open and
interpretable for various tools):</p>
<ul>
<li>I1. (Meta)data use a formal, accessible, shared, and broadly
applicable language for knowledge representation. - YES</li>
<li>I2. (Meta)data use vocabularies that follow FAIR principles -
PARTIALLY</li>
<li>I3. (Meta)data include qualified references to other (meta)data -
YES</li>
</ul>
<p>Reusable (data should be well-described so that they can be
replicated and/or combined in different settings, reuse states with a
clear licence):</p>
<ul>
<li>R1. (Meta)data are richly described with a plurality of accurate and
relevant attributes - YES
<ul>
<li>R1.1. (Meta)data are released with a clear and accessible data usage
license - YES</li>
<li>R1.2. (Meta)data are associated with detailed provenance - YES</li>
<li>R1.3. (Meta)data meet domain-relevant community standards -
YES/PARTIALLY</li>
</ul>
</li>
</ul>
</div>
</div>
</div>
</div>
<p>(4 min discussion)</p>
<div id="exercise-2-datasets-discovery-5-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-datasets-discovery-5-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Datasets discovery (5 min)</h3>
<div class="callout-content">
<p>Try to find either:</p>
<ul>
<li>similar data sets in Zenodo</li>
<li>data sets of interest for you</li>
</ul>
<p>Hint 1: Verify the completeness/richness of the associated metadata.
Is it complete? Are some bits missing?<br>
Hint 2: Does the dataset include a ReadMe.txt file?</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>Zenodo is a good place to keep your data separate from paper. It
gives access to all files, allowing you to cite the data as well (or
instead of) the paper.<br>
However, it is not good for disovery, and does not enforce most
metadata!</p>
</div>
</div>
</div>
</div>
<p><br> (3 min teaching) &gt; ## Minimal data set &gt; Minimal data set
to consist of the data required to replicate all study findings reported
&gt; in the article, as well as related metadata and methods. &gt; &gt;
* The values behind the means, standard deviations and other measures
reported; &gt; * The values used to build graphs; &gt; * The points
extracted from images for analysis. &gt; &gt; (no need for raw data if
the standard in the field is to share data that have been processed)
&gt; &gt; <a href="https://journals.plos.org/plosbiology/s/data-availability" class="external-link">PLOS</a><br>
&gt;<br>
{: .callout}</p>
<div id="challenge-3.-domain-specific-repositories-7-min." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="challenge-3.-domain-specific-repositories-7-min." class="callout-inner">
<h3 class="callout-title">Challenge 3. Domain specific repositories (7 min).</h3>
<div class="callout-content">
<p>Select one of the following repositories based on your
expertise/interests:</p>
<ul>
<li>Have a look at mRNAseq accession ‘E-MTAB-7933’ in <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-7933/" class="external-link">ArrayExpress</a>
</li>
<li>Have a look at microscopy ‘project-1101’ in <a href="https://idr.openmicroscopy.org/webclient/?show=project-1101" class="external-link">IDR</a>
</li>
<li>Have a look at the synthethic part record
‘SubtilinReceiver_spaRK_separated’ within the ‘bsu’ collection in <a href="https://synbiohub.org/public/bsu/SubtilinReceiver_spaRK_separated/1" class="external-link">SynBioHub</a>
</li>
<li>Have a look at the proteomics record ‘PXD013039’ in <a href="https://www.ebi.ac.uk/pride/archive/projects/PXD013039" class="external-link">PRIDE</a>
</li>
<li>Have a look at the metabolomics record ‘MTBLS2289’ in <a href="https://www.ebi.ac.uk/metabolights/MTBLS2289/descriptors" class="external-link">Metabolights</a>
</li>
<li>Have a look at the scripts deposit ‘RNA-Seq-validation’ in <a href="https://github.com/PauloFlores/RNA-Seq-validation" class="external-link">GitHub</a>
</li>
</ul>
<p>Report to the group, what advantages can you see in using a specific
repository over a generalist repository like Zenodo.</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>Some advantages are:</p>
<ul>
<li>The repository is more relevant to your discipline than a generalist
one.</li>
<li>Higher exposure (people looking for those specific types of data
will usually first look at the specific repository).</li>
<li>Higher number of citations (see above).</li>
</ul>
</div>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="how-do-we-choose-a-research-data-repository">How do we choose a research data repository?<a class="anchor" aria-label="anchor" href="#how-do-we-choose-a-research-data-repository"></a>
</h2>
<hr class="half-width">
<p>(3 min teaching) As a general rule, your research needs to be
deposited in discipline/data specific repository. If no specific
repository can be found, then you can use a generalist repository.
Having said this, there are tons of data repositories to choose from.
Choosing one can be time consuming and challenging as well. So how do
you go about finding a repository:</p>
<ul>
<li>
<p>Check the publisher’s / funder’ recommended list of repositories,
some of which can be found below:</p>
<ul>
<li><a href="https://www.springernature.com/gp/authors/research-data-policy/recommended-repositories" class="external-link">BioMed
Central / Springer Nature</a></li>
<li><a href="https://submit.elifesciences.org/html/elife_author_instructions.html#policies" class="external-link">eLife</a></li>
<li><a href="https://www.elsevier.com/about/policies/research-data" class="external-link">Elsevier</a></li>
<li><a href="https://www.embopress.org/page/journal/14602075/authorguide#datadeposition" class="external-link">EMBO
Press</a></li>
<li><a href="https://f1000research.com/for-authors/data-guidelines" class="external-link">F1000
Research</a></li>
<li><a href="https://academic.oup.com/gigascience/pages/instructions_to_authors" class="external-link">GIGAscience
- OUP</a></li>
<li><a href="https://journals.plos.org/plosbiology/s/recommended-repositories" class="external-link">PLoS</a></li>
<li><a href="https://www.nature.com/sdata/policies/repositories" class="external-link">Scientific
Data - Nature</a></li>
<li><a href="https://authorservices.taylorandfrancis.com/data-sharing-policies/repositories/" class="external-link">Taylor
and Francis</a></li>
<li><a href="https://bbsrc.ukri.org/research/resources/" class="external-link">BBSRC</a></li>
<li><a href="https://nerc.ukri.org/research/sites/environmental-data-service-eds/policy/" class="external-link">NERC</a></li>
<li><a href="https://royalsociety.org/journals/ethics-policies/data-sharing-mining/" class="external-link">Royal
Society</a></li>
<li><a href="https://wellcomeopenresearch.org/for-authors/data-guidelines" class="external-link">Wellcome
Open Research</a></li>
</ul>
</li>
<li>
<p>Check <a href="https://fairsharing.org/recommendations/?q=" class="external-link">Fairsharing
recommendations</a></p>
<ul>
<li>alternatively, check the <a href="https://www.re3data.org/" class="external-link">Registry
of research data repositories - re3data</a>
</li>
</ul>
</li>
</ul>
<div id="exercise-4-finding-a-repository-5-min-4-min-discussion." class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-4-finding-a-repository-5-min-4-min-discussion." class="callout-inner">
<h3 class="callout-title">Exercise 4: Finding a repository (5 min + 4 min discussion).</h3>
<div class="callout-content">
<ol style="list-style-type: lower-alpha">
<li>Find a repo for genomics data.<br>
</li>
<li>Find a repo for microscopy data.<br>
Note to instructor: Fairsharing gives few options, people may give
different answer follow up why they selected particular ones.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4"> Show me the solution </h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<ol style="list-style-type: lower-alpha">
<li>GEO/SRA and ENA/ArrayExpress are good examples. Interestingly these
repositories do not issue a DOI.<br>
</li>
<li>IDR is good examples.</li>
</ol>
</div>
</div>
</div>
</div>
<p>(6 min teaching)</p>
<p>A list of UoE BioRDM’s recommended data repositories can be found <a href="https://www.wiki.ed.ac.uk/display/RDMS/Suggested+data+repositories" class="external-link">here</a>.</p>
<div id="what-comes-first-the-repository-or-the-metadata" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="what-comes-first-the-repository-or-the-metadata" class="callout-inner">
<h3 class="callout-title">What comes first? the repository or the metadata?</h3>
<div class="callout-content">
<p>Finding a repository first may help in deciding what metadata to
collect and how!</p>
</div>
</div>
</div>
<div id="extra-features" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="extra-features" class="callout-inner">
<h3 class="callout-title">Extra features</h3>
<div class="callout-content">
<p>It is also worth considering that some repositories offer extra
features, such as running simulations or providing visualisation. For
example, <a href="https://fairdomhub.org/" class="external-link">FAIRDOMhub</a> can run model
simulations and has project structures. Do not forget to take this into
account when choosing your repository. Extra features might come in
handy.</p>
</div>
</div>
</div>
<div id="can-github-be-cited" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="can-github-be-cited" class="callout-inner">
<h3 class="callout-title">Can GitHub be cited?</h3>
<div class="callout-content">
<p>To make your code repositories easier to reference in academic
literature, you can create persistent identifiers for them.
Particularly, you can use the data archiving tool in Zenodo to archive a
GitHub repository and issue a DOI for it.</p>
</div>
</div>
</div>
</section><section><h2 class="section-heading" id="evaluating-a-research-data-repository">Evaluating a research data repository<a class="anchor" aria-label="anchor" href="#evaluating-a-research-data-repository"></a>
</h2>
<hr class="half-width">
<p>You can evaluate the repositories by following this criteria:</p>
<ul>
<li>who is behind it, what is its funding</li>
<li>quality of interaction: is the interaction for purposes of data
deposit or reuse efficient, effective and satisfactory for you?</li>
<li>take-up and impact: what can I put in it? Is anyone else using it?
Will others be able to find stuff deposited in it? Is the repository
linked to other data repositories so I don’t have to search tehre as
well? Can anyone reuse the data? Can others cite the data, and will
depositing boost citations to related papers?</li>
<li>policy and process: does it help you meet community standards of
good practice and comply with policies stipulating data deposit?</li>
</ul>
<p>An interesting take can be found at Peter Murray-Rust’s blog post <a href="https://blogs.ch.cam.ac.uk/pmr/2011/08/19/criteria-for-successful-repositories/" class="external-link">Criteria
for succesful repositories</a>.</p>
<div id="exercise-5-wrap-up-discussion-7-min---optional." class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-5-wrap-up-discussion-7-min---optional." class="callout-inner">
<h3 class="callout-title">Exercise 5: Wrap up discussion (7 min - OPTIONAL).</h3>
<div class="callout-content">
<p>Discuss the following questions:</p>
<ul>
<li>Why is choosing a domain specific repositories over zenodo more
FAIR?</li>
<li>How can selecting a repository for your data as soon as you do an
experiment (or even before!) benefit your research and help your data
become FAIR?</li>
<li>What is your favourite research data repository? Why?</li>
</ul>
</div>
</div>
</div>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was adapted or inspired by:.</p>
<ul>
<li><a href="https://www.go-fair.org/fair-principles/" class="external-link">FAIR
principles</a></li>
<li><a href="https://www.wiki.ed.ac.uk/display/RDMS/Suggested+data+repositories" class="external-link">BioRDM
suggested data repositories</a></li>
<li><a href="https://www.dcc.ac.uk/news/how-can-we-evaluate-data-repositories-pointers-dryaduk" class="external-link">DCC
- How can we evaluate data repositories?</a></li>
<li><a href="https://blogs.ch.cam.ac.uk/pmr/2011/08/19/criteria-for-successful-repositories/" class="external-link">Criteria
for succesful repositories</a></li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Repositories are the main means for sharing research data.</li>
<li>You should use data-type specific repository whenever possible.</li>
<li>Repositories are the key players in data reuse.</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-14-data_management_planning"><p>Content from <a href="14-data_management_planning.html">It's all about planning</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/14-data_management_planning.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 78 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What is the research data life cycle?</li>
<li>How to plan for FAIR sharing?</li>
<li>What is a Data Management Plan (DMP)?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Learn what actions can help in achieving FAIR.</li>
<li>Learn to plan for different stages/steps of the data life
cycle.</li>
<li>Draft a simple DMP for your project.</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>(5 min teaching)</p>
<section><h2 class="section-heading" id="the-research-data-life-cycle">The research data life cycle<a class="anchor" aria-label="anchor" href="#the-research-data-life-cycle"></a>
</h2>
<hr class="half-width">
<p>The Research Data Life Cycle is composed of a sequential series of
stages/steps in which data is produced, processed and shared. The arrows
between circles represent the transitions that occur in research as work
is finished and passed onto the next stage/step. The re-use is the
driving force of the cycle, because if you (or others) were not going to
use the data you would not need to store it or even process it.</p>
<p><img src="../fig/05-research_data_life_cycle.png" alt="Figure 5.1. The Research Data Life Cycle" class="figure"><em>Figure credits:
Tomasz Zieliński</em></p>
</section><section><h2 class="section-heading" id="data-management-is-a-continuous-process">Data management is a continuous process<a class="anchor" aria-label="anchor" href="#data-management-is-a-continuous-process"></a>
</h2>
<hr class="half-width">
<p>Data management should be done throughout the duration of your
project. If you wait till the end, it will take a massive effort on your
side and will be more of a burden than a benefit.</p>
<p>There are many actions/steps which you can take during your research
workflow which would make it easier to share your data in a
<strong>Findable, Accessible, Interoperable and Reusable</strong> way,
while, helping you in your day to day activities.</p>
<div id="exercise-1-action-plan-challenge-7-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-action-plan-challenge-7-min" class="callout-inner">
<h3 class="callout-title">Exercise 1: Action plan challenge (7 min)</h3>
<div class="callout-content">
<p>Where would you say the following actions belong in the Research Data
Life Cycle? How do they help in achieving FAIR principles?</p>
<ul>
<li>clarify usage rights</li>
<li>give credit through citations</li>
<li>use open source software</li>
<li>attach PIDs to your data</li>
<li>attach descriptive metadata</li>
<li>produce standardised metadata</li>
<li>backup your data</li>
<li>create figures and plots in python/R</li>
<li>organize your files in folders</li>
<li>select data repositories</li>
<li>add open licence</li>
<li>link publications, data and methods</li>
<li>create a template for assay descriptions</li>
<li>use institutional repositories</li>
<li>use controlled vocabularies</li>
<li>convert numerical data to csv</li>
<li>track versions of files</li>
<li>perform statistical analysis</li>
<li>deposit datasets to Zenodo/Dryad</li>
<li>record experiment details in an Electronic Lab Notebook</li>
<li>use github for your code</li>
<li>ask someone to revise your project structure</li>
<li>reformat and clean data tables</li>
<li>use a Minimal Information Standard</li>
<li>use PID in data description</li>
<li>download a dataset</li>
<li>link to UniProt or GenBank records</li>
</ul>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<p>Actually many of the above steps can be taken throughout the data
life cycle.</p>
</div>
</div>
</div>
</div>
<p>(13 min teaching)</p>
</section><section><h2 class="section-heading" id="plan-ahead-data-management-plans-dmps">Plan ahead: data management plans (DMPs)<a class="anchor" aria-label="anchor" href="#plan-ahead-data-management-plans-dmps"></a>
</h2>
<hr class="half-width">
<p>A good data management is about PLANning!</p>
<p>The (Data Management Plan) DMP’s purpose is to make you think about
your data before you even start the experiments. It should show that you
are thinking about what will happen with your data during and after the
project. Which of the actions mentioned above you will take and how you
will execute them. Finally, how your project is going to be FAIR.</p>
<p><img src="../fig/05-planning.png" alt="Figure 5.3. Planning ahead" class="figure"><em>Figure credits: Tomasz Zieliński and Andrés Romanowski</em></p>
<p>You should think about:</p>
<ul>
<li>How you will store the data</li>
<li>How you will organize and describe your data</li>
<li>How you will grant access to your data</li>
<li>How you will share your data</li>
<li>How you will preserve your data</li>
<li>How others can use your data</li>
<li>How much it will all cost</li>
</ul>
<p>Most funders require a DMP with grant applications. Some institutions
even ask their PhD students to prepare a DMP for their PhD project.</p>
<p>You should think about how you are going to manage your data (or
outputs, in general) for each of your projects (or even individual assay
types). For individual project the main focus should be on: <strong>what
data will be produced, how will they be stored and organized, how you
are going to describe them and track them</strong>. For example what
file formats will be generated, how you are going to name your files,
and how you will link it to your laboratory notes.</p>
<p>For grant applications, DMPs tend to be less technical, for example
no need to discuss folder structures, but, they should emphasize the
<strong>data safety (as preservation and access), data longevity,
sharing, discovery and re-use</strong>.</p>
<div id="dmp-online" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="dmp-online" class="callout-inner">
<h3 class="callout-title">DMP Online</h3>
<div class="callout-content">
<p><a href="https://dmponline.dcc.ac.uk/" class="external-link">DMP Online</a> is a UK tool
that is available via subscription to many UK Universities and
Institutions. It contains DMP templates for the different funders
requirements and information on how to fill each section. Whenever you
need to write DMP for a grant application, check if this resource is
available to you.</p>
<p><a href="https://dmptool.org" class="external-link">DMPTool</a> is similar tool tailored to
USA.</p>
<p>Additionally, your own Institutions may have resources to help with
your DMP.</p>
<p>Authors of this course, have created ready to use paragraphs for
writing DMPs and a list of recommended repositories at <a href="https://www.wiki.ed.ac.uk/display/RDMS/Data+Management+Plans" class="external-link">BioRDM
wiki</a></p>
</div>
</div>
</div>
<div id="exercise-2-challenge-30-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2-challenge-30-min" class="callout-inner">
<h3 class="callout-title">Exercise 2: Challenge (30 min)</h3>
<div class="callout-content">
<p>Working in pairs, think of your last paper (or project). Pretend that
you have a joined project that combines the outputs of both your
papers/projects.</p>
<p>Write a short DMP for this <strong>joined project</strong>.</p>
<p>Your DMP should contain the following three sections:</p>
<ol style="list-style-type: decimal">
<li><p>What data you will acquire during the project. Please describe
the type of data you will generate (for example ‘flow cytometry data’)
as well as file formats and data volume. These data will be stored under
(include the meta data as well). Estimate the size of your
data.</p></li>
<li><p>How you will store the data Please describe how you will store
and organize your data, what metadata will you capture and in a what
form. Tell how you will document the data during the duration of the
project</p></li>
<li><p>How you will share the data Please describe the strategies for
data sharing, licensing and access information.</p></li>
</ol>
<p><strong>Remember it is a joined project</strong></p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2"> Show me the solution </h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<p>An example DMP can look like:</p>
<ol style="list-style-type: decimal">
<li>The project will generate a combination of qualitative and
quantitative data derived from phenotyping, LC/MS metabolomics, and
general molecular biology techniques.</li>
</ol>
<p>The main data types and their formats are:</p>
<ul>
<li>Phenotyping images (tiff)</li>
<li>Time-series / Numerical data (Excel)</li>
<li>MS Metabolomics (mzML)</li>
<li>processing scripts (Python, R)</li>
</ul>
<p>The project will generate a total of 4 Tb data.</p>
<ol start="2" style="list-style-type: decimal">
<li>The instrument specific raw data will be converted into the open
formats mentioned above. Daily experimental work will be recorded using
an electronic lab notebook (Benchling). We will use ISA templates from
MetaboLights for MS data.</li>
</ol>
<p>All research data will be stored using the University file system.
This is a high quality storage with guaranteed backup. Scripts/codes
will be stored under version control using GitHub.</p>
<ol start="3" style="list-style-type: decimal">
<li>Metabolomics data will be made available through the MetaboLights
(<a href="https://www.ebi.ac.uk/metabolights/" class="external-link uri">https://www.ebi.ac.uk/metabolights/</a>) repository. The
remaining datasets will be made available through Zenodo. The data will
be released no later than a year after the project ends. All data will
be made available under CC-BY and the code under MIT licensing.</li>
</ol>
</div>
</div>
</div>
</div>
<div id="exercise-2-part-2-explore-each-others-dmps-10-min" class="callout discussion">
<div class="callout-square">
<i class="callout-icon" data-feather="message-circle"></i>
</div>
<div id="exercise-2-part-2-explore-each-others-dmps-10-min" class="callout-inner">
<h3 class="callout-title">Exercise 2, part 2: Explore each others DMPs (10 min)</h3>
<div class="callout-content">
<p>Now, take a look at the other group’s DMP and make
comments/suggestions on how to improve it (at the end of the DMP)</p>
</div>
</div>
</div>
<div id="exercise-3-quiz-3-min" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3-quiz-3-min" class="callout-inner">
<h3 class="callout-title">Exercise 3: Quiz (3 min)</h3>
<div class="callout-content">
<p>Answer the following questions with true or false (T or F):</p>
<ol style="list-style-type: decimal">
<li>The best time to do data management is at the end of a project, when
you’ve collected all the data you’re managing.</li>
<li>Data management plans (DMPs) detail what will happen to data before
collection begins.</li>
<li>The best storage method for data is multiple backups to USBs.</li>
<li>There is a single best way to manage, organise, and share data.</li>
<li>For grant applications, DMPs should mention data preservation,
longevity, sharing, discover, and reuse.</li>
<li>Your metadata should be standardised and descriptive.</li>
<li>Taking the time to plan out what’s needed in metadata and your DMP
will save you time in the long run and make your data more FAIR.</li>
<li>DMP online is a tool which constructs DMPs for researchers.</li>
<li>Data addressed in a DMP can be freely shared regardless of
confidentiality.</li>
<li>Data can be given creative commons licenses to dictate how others
can and cannot use it.</li>
</ol>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3"> Show me the solution </h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<ol style="list-style-type: decimal">
<li>The best time to do data management is at the end of a project, when
you’ve collected all the data you’re managing. (F)</li>
<li>Data management plans (DMPs) detail what will happen to data before
collection begins. (T)</li>
<li>The best storage method for data is multiple backups to USBs.
(F)</li>
<li>There is a single best way to manage, organise, and share data.
(F)</li>
<li>For grant applications, DMPs should mention data preservation,
longevity, sharing, discover, and reuse. (T)</li>
<li>Your metadata should be standardised and descriptive. (T)</li>
<li>Taking the time to plan out what’s needed in metadata and your DMP
will save you time in the long run and make your data more FAIR.
(T)</li>
<li>DMP online is a tool which constructs DMPs for researchers. (F)</li>
<li>Data addressed in a DMP can be freely shared regardless of
confidentiality. (F)</li>
<li>Data can be given creative commons licenses to dictate how others
can and cannot use it. (T)</li>
</ol>
</div>
</div>
</div>
</div>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was adapted after:</p>
<ul>
<li>
<a href="https://www.wiki.ed.ac.uk/display/RDMS/Data+Management+Plans" class="external-link">Data
Management Plans - UoE BioRDM wiki</a>.</li>
<li>
<a href="https://www.ed.ac.uk/information-services/research-support/research-data-service/before/benefits-of-writing-a-dmp" class="external-link">Benefits
of writing a DMP - UoE RDS</a>.</li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>Data within a project undergo a set of steps known as the research
data life cycle.</li>
<li>Planning can help make your data FAIR.</li>
<li>Data management is a continuous process during a project.</li>
<li>A DMP is the best way to prepare for a new project.</li>
</ul>
</div>
</div>
</div>
</section></section><section id="aio-15-putting-all-together"><p>Content from <a href="15-putting-all-together.html">Putting it all together</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/15-putting-all-together.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 0 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul>
<li>What will your journey to be FAIR productive entail?</li>
</ul>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul>
<li>Discuss steps and changes in your habits you will take after this
course</li>
<li>List of resources that will help you achieve these goals</li>
</ul>
</div>
</div>
</div>
</div>
</div>
<p>For the past three days we have discussed how you can embark on a
journey to be FAIR, we also showed you tools you can use in your daily
life and throughout the research data life cycle to make your data more:
Findable, Accessible, Interoperable and Reproducible</p>
<p>Simple things like thinking about the metadata that can accompany
your data, use of templates and preparation of your data for future
sharing can help in achieving Open Research and better FAIR practices.
We now want to hear from you what you will change after this course to
be more FAIR:</p>
<div id="exercise-1-your-journey-to-be-fairproductive" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1-your-journey-to-be-fairproductive" class="callout-inner">
<h3 class="callout-title">Exercise 1: Your journey to be FAIRproductive</h3>
<div class="callout-content">
<p>What are the things you are going to change in your work habits, or
actions you are going to take after this course?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" aria-labelledby="headingSolution1" data-bs-parent="#accordionSolution1">
<div class="accordion-body">
<ul>
<li>For each paper I publish I will make a Zenodo deposit with all the
data files</li>
<li>I will use GitHub for all my code instead of using folders</li>
<li>I will start using ELNs</li>
<li>Write a data management plan for my project</li>
<li>Reorganise my folder structure</li>
<li>Use templates</li>
<li>Start using readme files (make a readme template)</li>
<li>Write protocols on protocols.io</li>
<li>Back up my data accordingly</li>
<li>Think as a user of my data instead of just a creator</li>
<li>Start using Jupyter notebooks, or RMarkdown to describe analysis
workflows</li>
<li>Write intelligible metadata</li>
</ul>
</div>
</div>
</div>
</div>
<p>As you can see, anything counts. Even simple things like
reorganisation of your folder structure for a better overview or to stop
using USB sticks as backup material are steps towards FAIR. We have
collated here the list of resources throughout the course for a better
overview which will hopefully help you to guide future practices.</p>
<div id="resources" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="resources" class="callout-inner">
<h3 class="callout-title">Resources</h3>
<div class="callout-content">
<p>The BioRDM team has a lot of information about the here taught course
material on their <a href="https://www.wiki.ed.ac.uk/display/RDMS/" class="external-link">BioRDM wiki</a>.</p>
<ul>
<li>Details of funding bodies and their involvement and requirements can
be found at <a href="https://www.coalition-s.org/plan-s-funders-implementation/" class="external-link">Plan
S/cOAlition S</a>.</li>
<li>
<a href="https://www.coalition-s.org/blog/unboxing-the-journal-checker-tool/" class="external-link">cOAlition
S journal checker tool</a> to assess compliance being developed.</li>
<li>The <a href="https://doaj.org/" class="external-link">Directory of Open Access Journals
(DOAJ)</a> is a tool to find which journals are Open Access</li>
<li><a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000246" class="external-link">Challenges
&amp; benefits of OS</a></li>
<li><a href="https://www.cos.io/" class="external-link">Centre for Open Science</a></li>
<li><a href="https://www.youtube.com/watch?v=c-bemNZ-IqA" class="external-link">Ted talk
supporting OS</a></li>
<li>Wiki <a href="https://en.wikipedia.org/wiki/Open_science" class="external-link">Open
Science</a>
</li>
<li><a href="https://www.eosc-hub.eu/open-science-info" class="external-link">European Open
Science Cloud</a></li>
<li><a href="https://portlandpress.com/biochemist/article/42/3/58/225220/Science-is-necessarily-collaborative" class="external-link">Science
is necessarily collaborative - The Biochemist article</a></li>
<li>Digital Object Identifiers <a href="https://www.doi.org/" class="external-link">(DOI)</a>
</li>
<li>
<a href="https://orcid.org/" class="external-link">ORCID</a> is a free, unique, persistent
identifier that you own and control—forever</li>
<li>Minimum Information for Biological and Biomedical Investigations <a href="https://fairsharing.org/collection/MIBBI" class="external-link">(MIBBI)</a> is the
collection of the most known standards</li>
<li>
<a href="https://fairsharing.org/standards/" class="external-link">FAIRSharing</a> offers
excellent search service for finding standards</li>
<li><a href="https://howtofair.dk/how-to-fair/metadata/" class="external-link">Metadata - How
to FAIR</a></li>
<li><a href="https://doi.org/10.1371/journal.pcbi.1005510" class="external-link">Good enough
practices in scientific computing (Wilson et al., 2017)</a></li>
<li><a href="https://github.com/swcarpentry/good-enough-practices-in-scientific-computing" class="external-link">Good
Enough Practices for Scientific Computing</a></li>
<li><a href="https://www.data.cam.ac.uk/data-management-guide/organising-your-data" class="external-link">Organising
your data</a></li>
<li><a href="https://www.wur.nl/en/Value-Creation-Cooperation/Collaborating-with-WUR-1/Organising-files-and-folders.htm" class="external-link">Organising
files and folders</a></li>
<li><a href="https://libguides.princeton.edu/c.php?g=102546&amp;p=930626" class="external-link">File
naming</a></li>
<li><a href="https://librarycarpentry.org/lc-fair-research/05-reusable/index.html" class="external-link">Library
Carpentry FAIR Data</a></li>
<li><a href="https://doi.org/10.1038/d41586-018-07196-1" class="external-link">Why Jupyter is
a data scientists’ computational notebook of choice</a></li>
<li><a href="https://www.browserstack.com/guide/coding-standards-best-practices" class="external-link">Coding
Standards best practices</a></li>
<li><a href="https://reproducible-analysis-workshop.readthedocs.io/en/latest/4.Jupyter-Notebook.html" class="external-link">Reproducible
analysis and Research Transparency</a></li>
<li>
<a href="https://dmponline.dcc.ac.uk/" class="external-link">DMP Online</a> is a UK tool
that is available via subscription to many UK Universities and
Institutions</li>
<li><a href="https://www.wiki.ed.ac.uk/display/RDMS/Data+Management+Plans" class="external-link">BioRDM
wiki DMPs</a></li>
<li><a href="https://www.ed.ac.uk/information-services/research-support/research-data-service/before/benefits-of-writing-a-dmp" class="external-link">Benefits
of writing a DMP - UoE RDS</a></li>
</ul>
</div>
</div>
</div>
<div id="repositories-and-ontologies" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="repositories-and-ontologies" class="callout-inner">
<h3 class="callout-title">Repositories and Ontologies</h3>
<div class="callout-content">
<div class="section level4">
<h4 id="data-agnostic-repositories">Data agnostic repositories<a class="anchor" aria-label="anchor" href="#data-agnostic-repositories"></a>
</h4>
<p>There are general “data agnostic” repositories, for example:</p>
<ul>
<li>
<a href="https://datadryad.org" class="external-link">Dryad</a>,</li>
<li>
<a href="https://zenodo.org" class="external-link">Zenodo</a>,</li>
<li>
<a href="https://figshare.com" class="external-link">FigShare</a>,</li>
<li>
<a href="https://thedata.org" class="external-link">Dataverse</a>.</li>
</ul>
</div>
<div class="section level4">
<h4 id="or-domain-specific-for-example">Or domain specific, for example:<a class="anchor" aria-label="anchor" href="#or-domain-specific-for-example"></a>
</h4>
<ul>
<li>
<a href="https://www.uniprot.org/" class="external-link">UniProt</a> protein data,</li>
<li>
<a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a> sequence
data,</li>
<li>
<a href="https://www.ebi.ac.uk/metabolights/" class="external-link">MetaboLights</a>
metabolomics data</li>
<li>
<a href="https://github.com/" class="external-link">GitHub</a> for code.</li>
</ul>
<p>Similar to ORCID, there are other registries that can be used to
identify many of biological concepts and entities:</p>
<ul>
<li>species e.g. <a href="https://www.ncbi.nlm.nih.gov/Taxonomy" class="external-link">NCBI
taxonomy</a>
</li>
<li>chemicals e.g. <a href="https://www.ebi.ac.uk/chebi" class="external-link">ChEBI</a>
</li>
<li>proteins e.g. <a href="https://www.uniprot.org/" class="external-link">UniProt</a>
</li>
<li>genes e.g. <a href="https://www.ncbi.nlm.nih.gov/genbank/" class="external-link">GenBank</a>
</li>
<li>metabolic reactions, enzymes e.g <a href="https://www.genome.jp/kegg/" class="external-link">KEGG</a>
</li>
</ul>
<p><a href="https://bioportal.bioontology.org/" class="external-link">BioPortal</a> or <a href="https://www.ncbi.nlm.nih.gov/" class="external-link">NCBI</a><br>
are good places to start searching for a registry or a term.</p>
</div>
<div class="section level4">
<h4 id="here-some-links-to-help-you-find-the-right-ontologies-for-your-research">Here some links to help you find the right ontologies for your
research:<a class="anchor" aria-label="anchor" href="#here-some-links-to-help-you-find-the-right-ontologies-for-your-research"></a>
</h4>
<ol style="list-style-type: decimal">
<li>Finding ontologies: <a href="https://bioportal.bioontology.org/" class="external-link uri">https://bioportal.bioontology.org/</a>
</li>
<li>List of recommended ontologies: <a href="https://www.obofoundry.org/" class="external-link">http://www.obofoundry.org/</a>
</li>
</ol>
</div>
</div>
</div>
</div>
<div id="licences-we-recommend-using" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="licences-we-recommend-using" class="callout-inner">
<h3 class="callout-title">Licences we recommend using</h3>
<div class="callout-content">
<p>Licenses explicitly declare conditions and terms by which data and
software can be re-used. Here, we recommend:</p>
<ul>
<li>for data <a href="https://creativecommons.org/licenses/by/4.0/" class="external-link">Creative Commons
Attribution (CC BY)</a> license</li>
<li>for code a permissive open source license such as the <a href="https://opensource.org/licenses/MIT" class="external-link">MIT</a>, <a href="https://opensource.org/licenses/BSD-2-Clause" class="external-link">BSD</a>, or <a href="https://www.apache.org/licenses/" class="external-link">Apache license</a>.</li>
</ul>
</div>
</div>
</div>
<div id="list-of-elns-and-online-protocols" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="list-of-elns-and-online-protocols" class="callout-inner">
<h3 class="callout-title">List of ELNs and online protocols</h3>
<div class="callout-content">
<ul>
<li><a href="https://www.benchling.com/" class="external-link">Benchling</a></li>
<li><a href="https://www.protocols.io/" class="external-link">Procols.io</a></li>
<li>
<a href="https://www.colabra.app/" class="external-link">Colabra</a> (free version
available)</li>
<li>
<a href="https://www.scinote.net/" class="external-link">SciNote</a> (free version
available)</li>
<li>
<a href="https://www.researchspace.com/" class="external-link">Rspace</a> (trial
possible)</li>
<li>
<a href="https://www.labarchives.com/" class="external-link">Labarchives</a> (trial
possible contacting the company)</li>
<li>
<a href="https://www.ed.ac.uk/information-services/research-support/research-data-service/during/open-research-tools/wikibench" class="external-link">WikiBench</a>
(works on top of Confluence)</li>
</ul>
</div>
</div>
</div>
<div id="eln-resources" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="eln-resources" class="callout-inner">
<h3 class="callout-title">ELN Resources</h3>
<div class="callout-content">
<ul>
<li>Comprehensive summary of <a href="https://www.wiki.ed.ac.uk/pages/viewpage.action?pageId=463750271" class="external-link">ELNs
on the University of Edinburgh Wiki</a> by BioRDM</li>
<li>Our Benchling tutorial: <a href="https://www.wiki.ed.ac.uk/display/RDMS/Benchling+%28quick%29+tutorial" class="external-link">BioRDM
Wiki</a>
</li>
<li><a href="https://help.benchling.com/en/collections/1608962-benchling-training-kit-for-academics" class="external-link">Benchling’s
training kit for academics</a></li>
<li><a href="https://www.scinote.net/electronic-lab-notebook-guide/" class="external-link">ELN
Guide</a></li>
<li><a href="https://www.colabra.app/course/" class="external-link">Colabra’s lab management
course</a></li>
</ul>
</div>
</div>
</div>
<div id="other-useful-courses" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="other-useful-courses" class="callout-inner">
<h3 class="callout-title">Other useful courses</h3>
<div class="callout-content">
<p>You can explore more courses developed by the <a href="https://github.com/orgs/carpentries-incubator/repositories" class="external-link">Carpentries
Incubator</a></p>
<ul>
<li><a href="https://datacarpentry.org/OpenRefine-ecology-lesson/" class="external-link">Data
Cleaning with OpenRefine for Ecologists</a></li>
<li><a href="https://carpentries-incubator.github.io/open-science-with-r/" class="external-link">Open
Science with R</a></li>
</ul>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<ul>
<li>There are simple steps to help make your data more FAIR throughout
the research data lifecycle</li>
<li>Implement these steps throughout to keep track of your data and
changes</li>
</ul>
</div>
</div>
</div></section><section id="aio-episode_template"><p>Content from <a href="episode_template.html">Template</a></p>
<hr>
<p>Last updated on 2025-01-27 |

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/episodes/episode_template.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
<p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 0 minutes</p>
<div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>
<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<hr>
</div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<hr>
</div>
</div>
</div>
</div>
</div>
<p>I am episode intro</p>
<section><h2 class="section-heading" id="i-am-a-section">I am a section<a class="anchor" aria-label="anchor" href="#i-am-a-section"></a>
</h2>
<hr class="half-width">
<p>With a text.</p>
<figure><img src="../fig/figure_file.jpg" alt="Figure 1. I am some figure" class="figure mx-auto d-block"></figure><p><em>After <a href="https://www.figure.link/" class="external-link">Figure
source</a></em></p>
<div id="i-am-a-yellow-info" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="i-am-a-yellow-info" class="callout-inner">
<h3 class="callout-title">I am a yellow info</h3>
<div class="callout-content">
<p>And my text.</p>
</div>
</div>
</div>
<pre class="source"><code>I am code</code></pre>
<div id="i-am-a-problem" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="i-am-a-problem" class="callout-inner">
<h3 class="callout-title">I am a problem</h3>
<div class="callout-content">
<p>Defined here.</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1"> Show me the solution </h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<ul>
<li>I am an answer.</li>
<li>So am I.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="attribution" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="attribution" class="callout-inner">
<h3 class="callout-title">Attribution</h3>
<div class="callout-content">
<p>Content of this episode was adopted after XXX et al. <a href="https://biodare2.ed.ac.uk" class="external-link">YYY</a>.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points</h3>
<div class="callout-content">
<hr>
</div>
</div>
</div>
</section></section>
</div>
    </main>
</div>
<!-- END  : inst/pkgdown/templates/content-extra.html -->

      </div>
<!--/div.row-->
      		<footer class="row footer mx-md-3"><hr>
<div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>

        <a href="https://github.com/carpentries-incubator/fair-bio-practice/edit/main/README.md" class="external-link">Edit on GitHub</a>

	
        | <a href="https://github.com/carpentries-incubator/fair-bio-practice/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/carpentries-incubator/fair-bio-practice/" class="external-link">Source</a></p>
				<p><a href="https://github.com/carpentries-incubator/fair-bio-practice/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:bio_rdm@ed.ac.uk">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">

        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>

        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.11" class="external-link">sandpaper (0.16.11)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.7" class="external-link">pegboard (0.7.7)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.5" class="external-link">varnish (1.0.5)</a></p>
			</div>
		</footer>
</div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://carpentries-incubator.github.io/fair-bio-practice/instructor/aio.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "All in One View",
  "creativeWorkStatus": "active",
  "url": "https://carpentries-incubator.github.io/fair-bio-practice/instructor/aio.html",
  "identifier": "https://carpentries-incubator.github.io/fair-bio-practice/instructor/aio.html",
  "dateCreated": "2025-01-27",
  "dateModified": "2025-01-27",
  "datePublished": "2025-01-27"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo --><script>
          var _paq = window._paq = window._paq || [];
          /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
          _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
          _paq.push(["setDomains", ["*.lessons.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
          _paq.push(["setDoNotTrack", true]);
          _paq.push(["disableCookies"]);
          _paq.push(["trackPageView"]);
          _paq.push(["enableLinkTracking"]);
          (function() {
              var u="https://matomo.carpentries.org/";
              _paq.push(["setTrackerUrl", u+"matomo.php"]);
              _paq.push(["setSiteId", "1"]);
              var d=document, g=d.createElement("script"), s=d.getElementsByTagName("script")[0];
              g.async=true; g.src="https://matomo.carpentries.org/matomo.js"; s.parentNode.insertBefore(g,s);
          })();
        </script><!-- End Matomo Code -->
</body>
</html><!-- END:   inst/pkgdown/templates/layout.html-->

